{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "# requires dammit env\n",
    "# source activate dammit\n",
    "from dammit.fileio.gff3 import GFF3Parser"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make list of genes being quantified\n",
    "\n",
    "Salmon was run with the de novo transcriptome assemblies, with the original Trinity ID in the header for each fasta contig record, like this (example salmon output file):\n",
    "\n",
    "```\n",
    "Name\tLength\tEffectiveLength\tTPM\tNumReads\n",
    "TRINITY_DN2202_c0_g1_i1\t210\t39.818\t2.683835\t2.000000\n",
    "TRINITY_DN2270_c0_g1_i1\t213\t41.064\t0.000000\t0.000000\n",
    "TRINITY_DN2201_c0_g1_i1\t266\t69.681\t0.766816\t1.000000\n",
    "TRINITY_DN2222_c0_g1_i1\t243\t55.794\t2.873014\t3.000000\n",
    "TRINITY_DN2291_c0_g1_i1\t245\t56.916\t0.000000\t0.000000\n",
    "TRINITY_DN2269_c0_g1_i1\t294\t89.251\t0.000000\t0.000000\n",
    "TRINITY_DN2269_c1_g1_i1\t246\t57.479\t0.000000\t0.000000\n",
    "TRINITY_DN2279_c0_g1_i1\t426\t207.443\t0.000000\t0.000000\n",
    "TRINITY_DN2262_c0_g1_i1\t500\t280.803\t0.190459\t1.000912\n",
    "TRINITY_DN2253_c0_g1_i1\t1523\t1303.116\t0.164015\t4.000000\n",
    "TRINITY_DN2287_c0_g1_i1\t467\t247.962\t0.000000\t0.000000\n",
    "TRINITY_DN2287_c1_g1_i1\t325\t113.826\t0.469425\t1.000000\n",
    "TRINITY_DN2237_c0_g1_i1\t306\t98.441\t0.542788\t1.000000\n",
    "TRINITY_DN2237_c0_g2_i1\t307\t99.229\t0.000000\t0.000000\n",
    "TRINITY_DN2250_c0_g1_i1\t368\t151.832\t0.000000\t0.000000\n",
    "TRINITY_DN2250_c1_g1_i1\t271\t72.988\t0.000000\t0.000000\n",
    "TRINITY_DN2208_c0_g1_i1\t379\t162.080\t1.978014\t6.000000\n",
    "TRINITY_DN2277_c0_g1_i1\t269\t71.657\t0.745677\t1.000000\n",
    "TRINITY_DN2231_c0_g1_i1\t209\t39.409\t0.000000\t0.000000\n",
    "TRINITY_DN2231_c1_g1_i1\t334\t121.411\t0.000000\t0.000000\n",
    "TRINITY_DN2204_c0_g1_i1\t287\t84.121\t0.000000\t0.000000\n",
    "```\n",
    "\n",
    "Takes each salmon output .quant file (like the one above), parse the \"i\" (isoform) off of the ID to get a list of genes and isoforms identified by Trinity."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L_goodei_transfer_2\n",
      "Contigs: (385474, 5)\n",
      "Trinity 'genes': 266317\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/L_goodei_contig_gene.csv\n",
      "F_notti_FW_2\n",
      "Contigs: (159768, 5)\n",
      "Trinity 'genes': 123565\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_nottii_contig_gene.csv\n",
      "L_parva_FW_3\n",
      "Contigs: (409536, 5)\n",
      "Trinity 'genes': 275950\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/L_parva_contig_gene.csv\n",
      "F_similis_transfer_1\n",
      "Contigs: (520315, 5)\n",
      "Trinity 'genes': 392860\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_similis_contig_gene.csv\n",
      "F_olivaceous_FW_2\n",
      "Contigs: (350264, 5)\n",
      "Trinity 'genes': 248630\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_olivaceus_contig_gene.csv\n",
      "F_notatus_FW_2\n",
      "Contigs: (416296, 5)\n",
      "Trinity 'genes': 286353\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_notatus_contig_gene.csv\n",
      "F_heteroclitusMDPP_FW_3\n",
      "Contigs: (668484, 5)\n",
      "Trinity 'genes': 496133\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_heteroclitusMDPP_contig_gene.csv\n",
      "F_heteroclitusMDPL_transfer_3\n",
      "Contigs: (592415, 5)\n",
      "Trinity 'genes': 392026\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_heteroclitusMDPL_contig_gene.csv\n",
      "F_parvapinis_FW_1\n",
      "Contigs: (352342, 5)\n",
      "Trinity 'genes': 279009\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_parvapinis_contig_gene.csv\n",
      "F_diaphanus_BW_1\n",
      "Contigs: (384216, 5)\n",
      "Trinity 'genes': 288318\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_diaphanus_contig_gene.csv\n",
      "F_catanatus_BW_2\n",
      "Contigs: (405863, 5)\n",
      "Trinity 'genes': 283212\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_catanatus_contig_gene.csv\n",
      "F_zebrinus_FW_2\n",
      "Contigs: (266977, 5)\n",
      "Trinity 'genes': 194115\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_zebrinus_contig_gene.csv\n",
      "F_sciadicus_transfer_1\n",
      "Contigs: (241279, 5)\n",
      "Trinity 'genes': 178995\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_sciadicus_contig_gene.csv\n",
      "F_grandis_transfer_3\n",
      "Contigs: (809057, 5)\n",
      "Trinity 'genes': 608505\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_grandis_contig_gene.csv\n",
      "F_rathbuni_BW_2\n",
      "Contigs: (501215, 5)\n",
      "Trinity 'genes': 364231\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_rathbuni_contig_gene.csv\n",
      "F_chrysotus_BW_2\n",
      "Contigs: (396398, 5)\n",
      "Trinity 'genes': 269787\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/F_chrysotus_contig_gene.csv\n",
      "A_xenica_FW_2\n",
      "Contigs: (362782, 5)\n",
      "Trinity 'genes': 248903\n",
      "File written: /Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/A_xenica_contig_gene.csv\n"
     ]
    }
   ],
   "source": [
    "species_dirs = \"../salmon_denovo_by_species\"\n",
    "out_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gene_name_counts_23June2018/\"\n",
    "dirs = os.listdir(species_dirs)\n",
    "for species in dirs:\n",
    "    if species != \".DS_Store\":\n",
    "        count = 0\n",
    "        species_gene_file = out_dir + species + \"_contig_gene.csv\"\n",
    "        files = os.listdir(species_dirs + \"/\" + species)\n",
    "        for quant_dir in files:\n",
    "            if quant_dir != \".DS_Store\":\n",
    "                while count == 0:\n",
    "                    replicate = quant_dir.split(\".\")[0]\n",
    "                    print(replicate)\n",
    "                    quant_files = os.listdir(species_dirs + \"/\" + species + \"/\" + quant_dir)\n",
    "                    for file in quant_files:\n",
    "                        if file.endswith(\".sf\"):\n",
    "                            table = pd.read_csv(species_dirs + \"/\" + species + \"/\" + quant_dir + \"/\" + file,sep=\"\\t\")\n",
    "                            print(\"Contigs:\",table.shape)\n",
    "                            contig_names = pd.DataFrame(table.Name.str.split(\"_\",5).tolist(),columns = ['Trinity','Contig','Component','Gene','Isoform'])\n",
    "                            contig_names2 = pd.DataFrame()\n",
    "                            contig_names2['Name'] = contig_names[['Trinity','Contig','Component','Gene','Isoform']].apply(lambda x: '_'.join(x),axis=1)\n",
    "                            contig_names2['Gene'] = contig_names[['Trinity','Contig','Component','Gene']].apply(lambda x: '_'.join(x),axis=1)\n",
    "                            print(\"Trinity 'genes':\",contig_names2['Gene'].nunique())\n",
    "                            contig_names2.to_csv(species_gene_file)\n",
    "                            print(\"File written:\",species_gene_file)\n",
    "                            count += 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Run tximport in R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Merge annotations for each species, connecting Trinity contigs/genes to annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "species_dirs = \"../salmon_denovo_by_species\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(385474, 5)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/RNAseq_15killifish/salmon_denovo_by_species/L_goodei/L_goodei_transfer_2.quant/quant.sf\",sep=\"\\t\")\n",
    "table.head()\n",
    "table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TRINITY_DN2271_c0_g1_i1</th>\n",
       "      <th>TRINITY_DN2271_c0_g1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN2222_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN2222_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN2222_c1_g1_i1</td>\n",
       "      <td>TRINITY_DN2222_c1_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN2209_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN2209_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN2281_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN2281_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN2240_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN2240_c0_g1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   TRINITY_DN2271_c0_g1_i1  TRINITY_DN2271_c0_g1\n",
       "0  TRINITY_DN2222_c0_g1_i1  TRINITY_DN2222_c0_g1\n",
       "1  TRINITY_DN2222_c1_g1_i1  TRINITY_DN2222_c1_g1\n",
       "2  TRINITY_DN2209_c0_g1_i1  TRINITY_DN2209_c0_g1\n",
       "3  TRINITY_DN2281_c0_g1_i1  TRINITY_DN2281_c0_g1\n",
       "4  TRINITY_DN2240_c0_g1_i1  TRINITY_DN2240_c0_g1"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table = pd.read_csv(\"../salmon_denovo_by_speciesout_dir\" + \"F_heteroclitusMDPL_trinity_contig_gene_table.txt\",sep=\"\\t\")\n",
    "table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using dammit's `GFF3Parser` function \n",
    "1. Digests the gff3 file for each species (output from dammit, downloaded from the farm cluster)\n",
    "2. Sorts each contig by E-value score\n",
    "3. Assigns the lowest E-value score for each contig \n",
    "4. Separately, for each contig, saves gene names from the NCBI annotated F. heteroclitus genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F_notatus\n",
      "F_notatus.trinity_out.Trinity.fasta.dammit.gff3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/johnsolk/anaconda3/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(167061, 8)\n",
      "Number of contigs:\n",
      "(416296, 2)\n",
      "Number of unique gene names:\n",
      "37751\n",
      "Number of Fhet gene names:\n",
      "22530\n",
      "F_rathbuni\n",
      "F_rathbuni.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(176367, 8)\n",
      "Number of contigs:\n",
      "(501215, 2)\n",
      "Number of unique gene names:\n",
      "40216\n",
      "Number of Fhet gene names:\n",
      "22718\n",
      "F_diaphanus\n",
      "F_diaphanus.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(128323, 8)\n",
      "Number of contigs:\n",
      "(384216, 2)\n",
      "Number of unique gene names:\n",
      "36687\n",
      "Number of Fhet gene names:\n",
      "21624\n",
      "F_chrysotus\n",
      "F_chrysotus.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(155206, 8)\n",
      "Number of contigs:\n",
      "(396398, 2)\n",
      "Number of unique gene names:\n",
      "38576\n",
      "Number of Fhet gene names:\n",
      "22455\n",
      "L_goodei\n",
      "L_goodei.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(138485, 8)\n",
      "Number of contigs:\n",
      "(385474, 2)\n",
      "Number of unique gene names:\n",
      "35129\n",
      "Number of Fhet gene names:\n",
      "22631\n",
      "F_heteroclitusMDPL\n",
      "F_heteroclitusMDPL.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(188268, 8)\n",
      "Number of contigs:\n",
      "(592415, 2)\n",
      "Number of unique gene names:\n",
      "39563\n",
      "Number of Fhet gene names:\n",
      "22849\n",
      "F_heteroclitusMDPP\n",
      "F_heteroclitusMDPP.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(186798, 8)\n",
      "Number of contigs:\n",
      "(668484, 2)\n",
      "Number of unique gene names:\n",
      "39072\n",
      "Number of Fhet gene names:\n",
      "22994\n",
      "F_similis\n",
      "F_similis.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(154675, 8)\n",
      "Number of contigs:\n",
      "(520315, 2)\n",
      "Number of unique gene names:\n",
      "37209\n",
      "Number of Fhet gene names:\n",
      "22338\n",
      "L_parva\n",
      "L_parva.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(145529, 8)\n",
      "Number of contigs:\n",
      "(409536, 2)\n",
      "Number of unique gene names:\n",
      "37684\n",
      "Number of Fhet gene names:\n",
      "22601\n",
      "A_xenica\n",
      "A_xenica.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(148980, 8)\n",
      "Number of contigs:\n",
      "(362782, 2)\n",
      "Number of unique gene names:\n",
      "37921\n",
      "Number of Fhet gene names:\n",
      "21806\n",
      "F_catanatus\n",
      "F_catanatus.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(164596, 8)\n",
      "Number of contigs:\n",
      "(405863, 2)\n",
      "Number of unique gene names:\n",
      "41933\n",
      "Number of Fhet gene names:\n",
      "22435\n",
      "F_zebrinus\n",
      "F_zebrinus.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(102046, 8)\n",
      "Number of contigs:\n",
      "(266977, 2)\n",
      "Number of unique gene names:\n",
      "33694\n",
      "Number of Fhet gene names:\n",
      "20439\n",
      "F_parvapinis\n",
      "F_parvapinis.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(126200, 8)\n",
      "Number of contigs:\n",
      "(352342, 2)\n",
      "Number of unique gene names:\n",
      "34375\n",
      "Number of Fhet gene names:\n",
      "20647\n",
      "F_grandis\n",
      "F_grandis.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(234074, 8)\n",
      "Number of contigs:\n",
      "(809057, 2)\n",
      "Number of unique gene names:\n",
      "36224\n",
      "Number of Fhet gene names:\n",
      "24275\n",
      "F_sciadicus\n",
      "F_sciadicus.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(98332, 8)\n",
      "Number of contigs:\n",
      "(241279, 2)\n",
      "Number of unique gene names:\n",
      "29980\n",
      "Number of Fhet gene names:\n",
      "20382\n",
      "F_olivaceus\n",
      "F_olivaceous.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(134207, 8)\n",
      "Number of contigs:\n",
      "(350264, 2)\n",
      "Number of unique gene names:\n",
      "37459\n",
      "Number of Fhet gene names:\n",
      "21600\n",
      "F_nottii\n",
      "F_notti.trinity_out.Trinity.fasta.dammit.gff3\n",
      "Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):\n",
      "(69247, 8)\n",
      "Number of contigs:\n",
      "(159768, 2)\n",
      "Number of unique gene names:\n",
      "28788\n",
      "Number of Fhet gene names:\n",
      "18280\n"
     ]
    }
   ],
   "source": [
    "annotations_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"\n",
    "conversion_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"\n",
    "Fhet_counts_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/Fhet_counts/\"\n",
    "annotations = os.listdir(annotations_dir)\n",
    "conversions = os.listdir(conversion_dir)\n",
    "for annotation in annotations:\n",
    "    if annotation != \".DS_Store\":\n",
    "        species = annotation.split(\".\")[0]\n",
    "        if species == \"F_olivaceous\":\n",
    "            species = \"F_olivaceus\"\n",
    "        if species == \"F_notti\":\n",
    "            species = \"F_nottii\"\n",
    "        print(species)\n",
    "        print(annotation)\n",
    "        name = annotations_dir + annotation\n",
    "        annotations = GFF3Parser(filename=name).read()\n",
    "        annotations = annotations.dropna(subset=['Name'])\n",
    "        annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)\n",
    "        pickonename = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','Note','database','Dbxref','start','end','length']]\n",
    "        pickonename = pickonename.dropna(axis=0,how=\"all\")\n",
    "        print('Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):')\n",
    "        print(pickonename.shape)\n",
    "        contigs_w_expression = pd.read_csv(out_dir + species + '_trinity_contig_gene_table_filtered.txt',sep=\"\\t\",names = ['contig','gene'])\n",
    "        contigs = species + \"_trinity_contig_gene_table.txt\"\n",
    "        contigs_table = pd.read_csv(out_dir + contigs,sep=\"\\t\",names = ['contig','gene'])\n",
    "        print('Number of contigs:')\n",
    "        print(contigs_table.shape)\n",
    "        conversion_file = [i for i in conversions if i.startswith(species)][0]\n",
    "        conversion = pd.read_csv(conversion_dir + conversion_file)\n",
    "        conversion['contig'], conversion['info'] = conversion['original'].str.split(' ', 1).str\n",
    "        conversion = conversion[['contig','renamed']]\n",
    "        contigs_w_expression_conversion = pd.merge(conversion, contigs_w_expression, on=\"contig\")\n",
    "        contigs_w_expression_conversion.columns = ['contig','seqid','gene']\n",
    "        annotations_expression = pd.merge(contigs_w_expression_conversion,pickonename,on='seqid')\n",
    "        print(\"Number of unique gene names:\")\n",
    "        print(len(annotations_expression.Name.unique()))\n",
    "        fhet = annotations[annotations['Name'].str.startswith(\"gi\")]\n",
    "        fhet_filtered = fhet.query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "        print(\"Number of Fhet gene names:\")\n",
    "        print(len(fhet_filtered.Name.unique()))\n",
    "        fhet_out = Fhet_counts_dir + species + \"_genenames.csv\"\n",
    "        fhet_filtered.to_csv(fhet_out)\n",
    "        fhet_annotations_epxression = pd.merge(annotations_expression,fhet_filtered,on='seqid',how='inner')\n",
    "        names = fhet_annotations_epxression.drop_duplicates(subset='seqid')\n",
    "        # filetered out contigs with no expression\n",
    "        names = names.dropna(subset=['Name_x'])\n",
    "        names_out = \"/Users/johnsolk/Documents/UCDavis/Whitehead/annotation_files_test/\" + species + \"_genenames.csv\"\n",
    "        names.to_csv(names_out)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *Fundulus heteroclitus* gene models only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/johnsolk/anaconda3/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    }
   ],
   "source": [
    "RefSeq_NCBI_annotations = GFF3Parser(filename=\"/Users/johnsolk/Documents/UCDavis/Whitehead/tmp_Fhet_files/ref_Fundulus_heteroclitus-3.0.2_top_level.gff3\").read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>Gap</th>\n",
       "      <th>ID</th>\n",
       "      <th>Is_circular</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>anticodon</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>splices</th>\n",
       "      <th>standard_name</th>\n",
       "      <th>start</th>\n",
       "      <th>start_range</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transl_except</th>\n",
       "      <th>transl_table</th>\n",
       "      <th>type</th>\n",
       "      <th>weighted_identity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>taxon:8078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>id0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>region</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GeneID:105920463</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cdx1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GeneID:105920463,Genbank:XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rna0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GeneID:105920463,Genbank:XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>id1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rna0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GeneID:105920463,Genbank:XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>id2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rna0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8334</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>XM_012856390.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Dbxref  Gap     ID Is_circular  \\\n",
       "0                               taxon:8078  NaN    id0         NaN   \n",
       "1                         GeneID:105920463  NaN  gene0         NaN   \n",
       "2  GeneID:105920463,Genbank:XM_012856390.2  NaN   rna0         NaN   \n",
       "3  GeneID:105920463,Genbank:XM_012856390.2  NaN    id1         NaN   \n",
       "4  GeneID:105920463,Genbank:XM_012856390.2  NaN    id2         NaN   \n",
       "\n",
       "             Name Note Parent Target anticodon chromosome        ...         \\\n",
       "0         Unknown  NaN    NaN    NaN       NaN    Unknown        ...          \n",
       "1            cdx1  NaN    NaN    NaN       NaN        NaN        ...          \n",
       "2  XM_012856390.2  NaN  gene0    NaN       NaN        NaN        ...          \n",
       "3             NaN  NaN   rna0    NaN       NaN        NaN        ...          \n",
       "4             NaN  NaN   rna0    NaN       NaN        NaN        ...          \n",
       "\n",
       "  splices standard_name  start start_range  strand   transcript_id  \\\n",
       "0     NaN           NaN      0         NaN       +             NaN   \n",
       "1     NaN           NaN   5029         NaN       -             NaN   \n",
       "2     NaN           NaN   5029         NaN       -  XM_012856390.2   \n",
       "3     NaN           NaN  10044         NaN       -  XM_012856390.2   \n",
       "4     NaN           NaN   8334         NaN       -  XM_012856390.2   \n",
       "\n",
       "  transl_except transl_table    type weighted_identity  \n",
       "0           NaN          NaN  region               NaN  \n",
       "1           NaN          NaN    gene               NaN  \n",
       "2           NaN          NaN    mRNA               NaN  \n",
       "3           NaN          NaN    exon               NaN  \n",
       "4           NaN          NaN    exon               NaN  \n",
       "\n",
       "[5 rows x 61 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RefSeq_NCBI_annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Convert protein ID, e.g. \"XP_\" to gene ID\n",
    "# how many gene models?\n",
    "RefSeq_NCBI_genenames = RefSeq_NCBI_annotations.dropna(subset=['seqid','Dbxref','transcript_id']).drop_duplicates(subset='Dbxref')[['seqid','Dbxref','transcript_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(41170, 3)\n"
     ]
    }
   ],
   "source": [
    "print(RefSeq_NCBI_genenames.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              seqid                                   Dbxref   transcript_id\n",
      "2    NW_012224401.1  GeneID:105920463,Genbank:XM_012856390.2  XM_012856390.2\n",
      "10   NW_012224401.1  GeneID:105915213,Genbank:XM_021321288.1  XM_021321288.1\n",
      "54   NW_012224401.1  GeneID:105915271,Genbank:XM_012849362.2  XM_012849362.2\n",
      "98   NW_012224401.1  GeneID:105915271,Genbank:XM_012849437.2  XM_012849437.2\n",
      "143  NW_012224401.1  GeneID:105915419,Genbank:XM_021323826.1  XM_021323826.1\n"
     ]
    }
   ],
   "source": [
    "print(RefSeq_NCBI_genenames.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "RefSeq_NCBI_genenames['split_1'], RefSeq_NCBI_genenames['split_2'], RefSeq_NCBI_genenames['split_3'] = RefSeq_NCBI_genenames['Dbxref'].str.split(':',0).str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              seqid                                   Dbxref   transcript_id  \\\n",
      "2    NW_012224401.1  GeneID:105920463,Genbank:XM_012856390.2  XM_012856390.2   \n",
      "10   NW_012224401.1  GeneID:105915213,Genbank:XM_021321288.1  XM_021321288.1   \n",
      "54   NW_012224401.1  GeneID:105915271,Genbank:XM_012849362.2  XM_012849362.2   \n",
      "98   NW_012224401.1  GeneID:105915271,Genbank:XM_012849437.2  XM_012849437.2   \n",
      "143  NW_012224401.1  GeneID:105915419,Genbank:XM_021323826.1  XM_021323826.1   \n",
      "\n",
      "    split_1            split_2         split_3  \n",
      "2    GeneID  105920463,Genbank  XM_012856390.2  \n",
      "10   GeneID  105915213,Genbank  XM_021321288.1  \n",
      "54   GeneID  105915271,Genbank  XM_012849362.2  \n",
      "98   GeneID  105915271,Genbank  XM_012849437.2  \n",
      "143  GeneID  105915419,Genbank  XM_021323826.1  \n"
     ]
    }
   ],
   "source": [
    "print(RefSeq_NCBI_genenames.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              seqid                                   Dbxref   transcript_id  \\\n",
      "2    NW_012224401.1  GeneID:105920463,Genbank:XM_012856390.2  XM_012856390.2   \n",
      "10   NW_012224401.1  GeneID:105915213,Genbank:XM_021321288.1  XM_021321288.1   \n",
      "54   NW_012224401.1  GeneID:105915271,Genbank:XM_012849362.2  XM_012849362.2   \n",
      "98   NW_012224401.1  GeneID:105915271,Genbank:XM_012849437.2  XM_012849437.2   \n",
      "143  NW_012224401.1  GeneID:105915419,Genbank:XM_021323826.1  XM_021323826.1   \n",
      "\n",
      "    split_1            split_2         split_3  \n",
      "2    GeneID  105920463,Genbank  XM_012856390.2  \n",
      "10   GeneID  105915213,Genbank  XM_021321288.1  \n",
      "54   GeneID  105915271,Genbank  XM_012849362.2  \n",
      "98   GeneID  105915271,Genbank  XM_012849437.2  \n",
      "143  GeneID  105915419,Genbank  XM_021323826.1  \n"
     ]
    }
   ],
   "source": [
    "RefSeq_NCBI_genenames_sub = RefSeq_NCBI_genenames[(RefSeq_NCBI_genenames[\"split_1\"] == 'GeneID')]\n",
    "print(RefSeq_NCBI_genenames_sub.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "RefSeq_NCBI_genenames_sub['GeneID'], RefSeq_NCBI_genenames_sub['Genbank'] = RefSeq_NCBI_genenames_sub['split_2'].str.split(',', 0).str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              seqid                                   Dbxref   transcript_id  \\\n",
      "2    NW_012224401.1  GeneID:105920463,Genbank:XM_012856390.2  XM_012856390.2   \n",
      "10   NW_012224401.1  GeneID:105915213,Genbank:XM_021321288.1  XM_021321288.1   \n",
      "54   NW_012224401.1  GeneID:105915271,Genbank:XM_012849362.2  XM_012849362.2   \n",
      "98   NW_012224401.1  GeneID:105915271,Genbank:XM_012849437.2  XM_012849437.2   \n",
      "143  NW_012224401.1  GeneID:105915419,Genbank:XM_021323826.1  XM_021323826.1   \n",
      "\n",
      "    split_1            split_2         split_3     GeneID  Genbank  \n",
      "2    GeneID  105920463,Genbank  XM_012856390.2  105920463  Genbank  \n",
      "10   GeneID  105915213,Genbank  XM_021321288.1  105915213  Genbank  \n",
      "54   GeneID  105915271,Genbank  XM_012849362.2  105915271  Genbank  \n",
      "98   GeneID  105915271,Genbank  XM_012849437.2  105915271  Genbank  \n",
      "143  GeneID  105915419,Genbank  XM_021323826.1  105915419  Genbank  \n"
     ]
    }
   ],
   "source": [
    "print(RefSeq_NCBI_genenames_sub.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          GeneID           seqid   transcript_id\n",
      "2      105920463  NW_012224401.1  XM_012856390.2\n",
      "7500   105932240  NW_012224402.1  XM_021322060.1\n",
      "13527  105915837  NW_012224403.1  XM_021309415.1\n",
      "17167  105916749  NW_012224404.1  XM_021310176.1\n",
      "21573  110366941  NW_012224405.1  XM_021310901.1\n",
      "(3260, 3)\n"
     ]
    }
   ],
   "source": [
    "geneID_unique_scaffold = RefSeq_NCBI_genenames_sub.dropna(subset=['GeneID','seqid','transcript_id']).drop_duplicates(subset='seqid')[['GeneID','seqid','transcript_id']]\n",
    "print(geneID_unique_scaffold.head())\n",
    "print(geneID_unique_scaffold.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        GeneID           seqid   transcript_id\n",
      "2    105920463  NW_012224401.1  XM_012856390.2\n",
      "10   105915213  NW_012224401.1  XM_021321288.1\n",
      "54   105915271  NW_012224401.1  XM_012849362.2\n",
      "143  105915419  NW_012224401.1  XM_021323826.1\n",
      "228  110366472  NW_012224401.1  XM_021307801.1\n",
      "(25600, 3)\n"
     ]
    }
   ],
   "source": [
    "geneID = RefSeq_NCBI_genenames_sub.dropna(subset=['GeneID','seqid','transcript_id']).drop_duplicates(subset='GeneID')[['GeneID','seqid','transcript_id']]\n",
    "print(geneID.head())\n",
    "print(geneID.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        GeneID           seqid   transcript_id\n",
      "2    105920463  NW_012224401.1  XM_012856390.2\n",
      "10   105915213  NW_012224401.1  XM_021321288.1\n",
      "54   105915271  NW_012224401.1  XM_012849362.2\n",
      "98   105915271  NW_012224401.1  XM_012849437.2\n",
      "143  105915419  NW_012224401.1  XM_021323826.1\n",
      "(41170, 3)\n"
     ]
    }
   ],
   "source": [
    "transcript_ID = RefSeq_NCBI_genenames_sub.dropna(subset=['GeneID','seqid','transcript_id']).drop_duplicates(subset='transcript_id')[['GeneID','seqid','transcript_id']]\n",
    "print(transcript_ID.head())\n",
    "print(transcript_ID.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         protein_id           seqid  \\\n",
      "6    XP_012711844.1  NW_012224401.1   \n",
      "32   XP_021176963.1  NW_012224401.1   \n",
      "77   XP_012704816.1  NW_012224401.1   \n",
      "121  XP_012704891.1  NW_012224401.1   \n",
      "165  XP_021179501.1  NW_012224401.1   \n",
      "\n",
      "                                             product    gene  \n",
      "6                             homeobox protein CDX-1    cdx1  \n",
      "32      platelet-derived growth factor receptor beta  pdgfrb  \n",
      "77   macrophage colony-stimulating factor 1 receptor   csf1r  \n",
      "121  macrophage colony-stimulating factor 1 receptor   csf1r  \n",
      "165       HMG domain-containing protein 3 isoform X1  hmgxb3  \n",
      "(38234, 4)\n"
     ]
    }
   ],
   "source": [
    "protein_ID = RefSeq_NCBI_annotations.dropna(subset=['protein_id','seqid','product','gene']).drop_duplicates(subset='protein_id')[['protein_id','seqid','product','gene']]\n",
    "print(protein_ID.head())\n",
    "print(protein_ID.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3042359, 6)\n",
      "       protein_id           seqid                 product  gene     GeneID  \\\n",
      "0  XP_012711844.1  NW_012224401.1  homeobox protein CDX-1  cdx1  105920463   \n",
      "1  XP_012711844.1  NW_012224401.1  homeobox protein CDX-1  cdx1  105915213   \n",
      "2  XP_012711844.1  NW_012224401.1  homeobox protein CDX-1  cdx1  105915271   \n",
      "3  XP_012711844.1  NW_012224401.1  homeobox protein CDX-1  cdx1  105915271   \n",
      "4  XP_012711844.1  NW_012224401.1  homeobox protein CDX-1  cdx1  105915419   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_012856390.2  \n",
      "1  XM_021321288.1  \n",
      "2  XM_012849362.2  \n",
      "3  XM_012849437.2  \n",
      "4  XM_021323826.1  \n"
     ]
    }
   ],
   "source": [
    "#merge protein ID and transcript ID\n",
    "protein_transcript = pd.merge(protein_ID,transcript_ID,how=\"inner\",on=\"seqid\").dropna()\n",
    "print(protein_transcript.shape)\n",
    "print(protein_transcript.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique scaffold ID ('NW_') : 10180\n",
      "Number of unique transcripts ('XM_') : 41170\n",
      "Number of unique protein ID ('XP_') : 38234\n",
      "Number of unique gene ID (parsed from 'Dbxref') : 25600\n"
     ]
    }
   ],
   "source": [
    "print(\"Number of unique scaffold ID ('NW_') :\",len(pd.unique(RefSeq_NCBI_annotations['seqid'].dropna())))\n",
    "print(\"Number of unique transcripts ('XM_') :\",len(pd.unique(RefSeq_NCBI_annotations['transcript_id'].dropna())))\n",
    "print(\"Number of unique protein ID ('XP_') :\",len(pd.unique(RefSeq_NCBI_annotations['protein_id'].dropna())))\n",
    "print(\"Number of unique gene ID (parsed from 'Dbxref') :\",len(geneID))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge NCBI annotations with contigs annotated to Fhet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['F_catanatus_genenames.csv', 'F_chrysotus_genenames.csv', 'F_notatus_genenames.csv', 'F_olivaceus_genenames.csv', 'L_goodei_genenames.csv', 'F_nottii_genenames.csv', '.DS_Store', 'F_grandis_genenames.csv', 'F_diaphanus_genenames.csv', 'A_xenica_genenames.csv', 'F_parvapinis_genenames.csv', 'F_rathbuni_genenames.csv', 'F_similis_genenames.csv', 'F_heteroclitusMDPL_genenames.csv', 'L_parva_genenames.csv', 'F_zebrinus_genenames.csv', 'F_sciadicus_genenames.csv', 'F_heteroclitusMDPP_genenames.csv']\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (78538, 5)\n",
      "Unique Fhet genes: (22435, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100057  gi|831528406|ref|XP_012720105.1| PREDICTED: E3...   \n",
      "1  Transcript_100057  gi|831528406|ref|XP_012720105.1| PREDICTED: E3...   \n",
      "2  Transcript_100057  gi|831528406|ref|XP_012720105.1| PREDICTED: E3...   \n",
      "3  Transcript_100057  gi|831528406|ref|XP_012720105.1| PREDICTED: E3...   \n",
      "4  Transcript_100057  gi|831528406|ref|XP_012720105.1| PREDICTED: E3...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     57  287     230  831528406  XP_012720105.1   \n",
      "1     57  287     230  831528406  XP_012720105.1   \n",
      "2     57  287     230  831528406  XP_012720105.1   \n",
      "3     57  287     230  831528406  XP_012720105.1   \n",
      "4     57  287     230  831528406  XP_012720105.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: E3 SUMO-protein ligase NSE2 [Fundu...  NW_012234283.1   \n",
      "1   PREDICTED: E3 SUMO-protein ligase NSE2 [Fundu...  NW_012234283.1   \n",
      "2   PREDICTED: E3 SUMO-protein ligase NSE2 [Fundu...  NW_012234283.1   \n",
      "3   PREDICTED: E3 SUMO-protein ligase NSE2 [Fundu...  NW_012234283.1   \n",
      "4   PREDICTED: E3 SUMO-protein ligase NSE2 [Fundu...  NW_012234283.1   \n",
      "\n",
      "                       product    gene     GeneID   transcript_id  \n",
      "0  E3 SUMO-protein ligase NSE2  nsmce2  105927720  XM_012864623.2  \n",
      "1  E3 SUMO-protein ligase NSE2  nsmce2  110368654  XR_002428266.1  \n",
      "2  E3 SUMO-protein ligase NSE2  nsmce2  105927718  XM_012864621.2  \n",
      "3  E3 SUMO-protein ligase NSE2  nsmce2  110368655  XR_002428267.1  \n",
      "4  E3 SUMO-protein ligase NSE2  nsmce2  105927708  XM_012864610.2  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN56902_c1_g1_i1  Transcript_160      0  113     113  831490423   \n",
      "1  TRINITY_DN56910_c0_g1_i1  Transcript_165      1  140     139  831535516   \n",
      "2   TRINITY_DN7043_c0_g1_i1  Transcript_177      0  109     109  831510576   \n",
      "3  TRINITY_DN64187_c0_g1_i1  Transcript_240      0  225     225  831501107   \n",
      "4  TRINITY_DN64187_c0_g2_i1  Transcript_241      0  187     187  831501107   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012706902.1   PREDICTED: cell death activator CIDE-B [Fundu...   \n",
      "1  XP_012722638.1   PREDICTED: neuroligin-3-like isoform X2 [Fund...   \n",
      "2  XP_012713947.1   PREDICTED: UDP-glucuronosyltransferase 2A1-li...   \n",
      "3  XP_012710668.1   PREDICTED: F-box only protein 39 [Fundulus he...   \n",
      "4  XP_012710668.1   PREDICTED: F-box only protein 39 [Fundulus he...   \n",
      "\n",
      "  NCBI.trasncript                     NCBI.product          gene  NCBI.gene  \n",
      "0  NW_012224604.1      cell death activator CIDE-B         cideb  105916808  \n",
      "1  NW_012234309.1     neuroligin-3-like isoform X2  LOC105929411  105929337  \n",
      "2  NW_012225612.1  UDP-glucuronosyltransferase 2A1  LOC105922597  105922594  \n",
      "3  NW_012224870.1            F-box only protein 39        fbxo39  105919806  \n",
      "4  NW_012224870.1            F-box only protein 39        fbxo39  105919806  \n",
      "(44400, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (80057, 5)\n",
      "Unique Fhet genes: (22455, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100003  gi|831565769|ref|XP_012733144.1| PREDICTED: 60...   \n",
      "1  Transcript_100003  gi|831565769|ref|XP_012733144.1| PREDICTED: 60...   \n",
      "2  Transcript_100003  gi|831565769|ref|XP_012733144.1| PREDICTED: 60...   \n",
      "3  Transcript_100003  gi|831565769|ref|XP_012733144.1| PREDICTED: 60...   \n",
      "4  Transcript_100003  gi|831565769|ref|XP_012733144.1| PREDICTED: 60...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     76   99      23  831565769  XP_012733144.1   \n",
      "1     76   99      23  831565769  XP_012733144.1   \n",
      "2     76   99      23  831565769  XP_012733144.1   \n",
      "3     76   99      23  831565769  XP_012733144.1   \n",
      "4     76   99      23  831565769  XP_012733144.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: 60S ribosomal protein L18a [Fundul...  NW_012234461.1   \n",
      "1   PREDICTED: 60S ribosomal protein L18a [Fundul...  NW_012234461.1   \n",
      "2   PREDICTED: 60S ribosomal protein L18a [Fundul...  NW_012234461.1   \n",
      "3   PREDICTED: 60S ribosomal protein L18a [Fundul...  NW_012234461.1   \n",
      "4   PREDICTED: 60S ribosomal protein L18a [Fundul...  NW_012234461.1   \n",
      "\n",
      "                      product    gene     GeneID   transcript_id  \n",
      "0  60S ribosomal protein L18a  rpl18a  105936707  XM_012877667.2  \n",
      "1  60S ribosomal protein L18a  rpl18a  105936713  XM_012877674.2  \n",
      "2  60S ribosomal protein L18a  rpl18a  105936738  XM_021324122.1  \n",
      "3  60S ribosomal protein L18a  rpl18a  105936732  XM_012877694.2  \n",
      "4  60S ribosomal protein L18a  rpl18a  105936731  XM_021324124.1  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73275_c0_g1_i1    Transcript_1    117  803     686  831517552   \n",
      "1  TRINITY_DN81303_c0_g1_i1   Transcript_97      0   73      73  831503614   \n",
      "2   TRINITY_DN7034_c1_g1_i1  Transcript_176     10   77      67  831515782   \n",
      "3   TRINITY_DN7043_c0_g1_i1  Transcript_177     12   79      67  831515782   \n",
      "4   TRINITY_DN7000_c0_g1_i1  Transcript_180     19   67      48  831514899   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012716268.1   PREDICTED: sacsin-like, partial [Fundulus het...   \n",
      "1  XP_012711549.1   PREDICTED: salivary acidic proline-rich phosp...   \n",
      "2  XP_012715679.1   PREDICTED: tripartite motif-containing protei...   \n",
      "3  XP_012715679.1   PREDICTED: tripartite motif-containing protei...   \n",
      "4  XP_012715391.1   PREDICTED: tripartite motif-containing protei...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012231991.1                                             sacsin   \n",
      "1  NW_012224961.1  salivary acidic proline-rich phosphoprotein 1/...   \n",
      "2  NW_012227997.1        tripartite motif-containing protein 16-like   \n",
      "3  NW_012227997.1        tripartite motif-containing protein 16-like   \n",
      "4  NW_012227105.1        tripartite motif-containing protein 16-like   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105925064  105925064  \n",
      "1  LOC105920491  105920494  \n",
      "2  LOC105924386  105924386  \n",
      "3  LOC105924386  105924386  \n",
      "4  LOC105924037  105924037  \n",
      "(45181, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (82318, 5)\n",
      "Unique Fhet genes: (22530, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100003  gi|831547417|ref|XP_012726940.1| PREDICTED: el...   \n",
      "1  Transcript_100003  gi|831547417|ref|XP_012726940.1| PREDICTED: el...   \n",
      "2  Transcript_100003  gi|831547417|ref|XP_012726940.1| PREDICTED: el...   \n",
      "3  Transcript_100003  gi|831547417|ref|XP_012726940.1| PREDICTED: el...   \n",
      "4  Transcript_100003  gi|831547417|ref|XP_012726940.1| PREDICTED: el...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     87  420     333  831547417  XP_012726940.1   \n",
      "1     87  420     333  831547417  XP_012726940.1   \n",
      "2     87  420     333  831547417  XP_012726940.1   \n",
      "3     87  420     333  831547417  XP_012726940.1   \n",
      "4     87  420     333  831547417  XP_012726940.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: electron transfer flavoprotein sub...  NW_012234358.1   \n",
      "1   PREDICTED: electron transfer flavoprotein sub...  NW_012234358.1   \n",
      "2   PREDICTED: electron transfer flavoprotein sub...  NW_012234358.1   \n",
      "3   PREDICTED: electron transfer flavoprotein sub...  NW_012234358.1   \n",
      "4   PREDICTED: electron transfer flavoprotein sub...  NW_012234358.1   \n",
      "\n",
      "                                             product  gene     GeneID  \\\n",
      "0  electron transfer flavoprotein subunit alpha%2...  etfa  105932326   \n",
      "1  electron transfer flavoprotein subunit alpha%2...  etfa  105932351   \n",
      "2  electron transfer flavoprotein subunit alpha%2...  etfa  110369071   \n",
      "3  electron transfer flavoprotein subunit alpha%2...  etfa  105932344   \n",
      "4  electron transfer flavoprotein subunit alpha%2...  etfa  105932345   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_012871419.2  \n",
      "1  XM_021320958.1  \n",
      "2  XR_002428667.1  \n",
      "3  XM_012871461.2  \n",
      "4  XM_012871464.2  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     contig          seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73275_c0_g2_i1   Transcript_2     30  133     103  831573055   \n",
      "1  TRINITY_DN73287_c0_g1_i1   Transcript_3     30  180     150  831573058   \n",
      "2  TRINITY_DN73227_c0_g1_i1   Transcript_4     30  138     108  831573058   \n",
      "3  TRINITY_DN73260_c0_g1_i1   Transcript_5     30  148     118  831573058   \n",
      "4  TRINITY_DN73242_c0_g2_i1  Transcript_15      0   68      68  831544836   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012735565.1   PREDICTED: bcl-2-modifying factor isoform X1 ...   \n",
      "1  XP_012735566.1   PREDICTED: bcl-2-modifying factor isoform X2 ...   \n",
      "2  XP_012735566.1   PREDICTED: bcl-2-modifying factor isoform X2 ...   \n",
      "3  XP_012735566.1   PREDICTED: bcl-2-modifying factor isoform X2 ...   \n",
      "4  XP_012726018.1   PREDICTED: probable G-protein coupled recepto...   \n",
      "\n",
      "  NCBI.trasncript                            NCBI.product          gene  \\\n",
      "0  NW_012234514.1       bcl-2-modifying factor isoform X1           bmf   \n",
      "1  NW_012234514.1       bcl-2-modifying factor isoform X3           bmf   \n",
      "2  NW_012234514.1       bcl-2-modifying factor isoform X3           bmf   \n",
      "3  NW_012234514.1       bcl-2-modifying factor isoform X3           bmf   \n",
      "4  NW_012234347.1  probable G-protein coupled receptor 61  LOC105931746   \n",
      "\n",
      "   NCBI.gene  \n",
      "0  105938397  \n",
      "1  105938397  \n",
      "2  105938397  \n",
      "3  105938397  \n",
      "4  105931768  \n",
      "(46201, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (68944, 5)\n",
      "Unique Fhet genes: (21600, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100060  gi|831567173|ref|XP_012733621.1| PREDICTED: cl...   \n",
      "1  Transcript_100060  gi|831567173|ref|XP_012733621.1| PREDICTED: cl...   \n",
      "2  Transcript_100060  gi|831567173|ref|XP_012733621.1| PREDICTED: cl...   \n",
      "3  Transcript_100060  gi|831567173|ref|XP_012733621.1| PREDICTED: cl...   \n",
      "4  Transcript_100060  gi|831567173|ref|XP_012733621.1| PREDICTED: cl...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0    201  537     336  831567173  XP_012733621.1   \n",
      "1    201  537     336  831567173  XP_012733621.1   \n",
      "2    201  537     336  831567173  XP_012733621.1   \n",
      "3    201  537     336  831567173  XP_012733621.1   \n",
      "4    201  537     336  831567173  XP_012733621.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: class E basic helix-loop-helix pro...  NW_012234467.1   \n",
      "1   PREDICTED: class E basic helix-loop-helix pro...  NW_012234467.1   \n",
      "2   PREDICTED: class E basic helix-loop-helix pro...  NW_012234467.1   \n",
      "3   PREDICTED: class E basic helix-loop-helix pro...  NW_012234467.1   \n",
      "4   PREDICTED: class E basic helix-loop-helix pro...  NW_012234467.1   \n",
      "\n",
      "                                     product          gene     GeneID  \\\n",
      "0  class E basic helix-loop-helix protein 40  LOC105937041  105937011   \n",
      "1  class E basic helix-loop-helix protein 40  LOC105937041  105937011   \n",
      "2  class E basic helix-loop-helix protein 40  LOC105937041  105937044   \n",
      "3  class E basic helix-loop-helix protein 40  LOC105937041  105937044   \n",
      "4  class E basic helix-loop-helix protein 40  LOC105937041  105937015   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_012878115.2  \n",
      "1  XM_012878117.2  \n",
      "2  XM_021324358.1  \n",
      "3  XM_012878170.2  \n",
      "4  XM_021324366.1  \n",
      "                     contig          seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73227_c0_g1_i1   Transcript_4     95  576     481  831532028   \n",
      "1  TRINITY_DN73257_c0_g1_i1  Transcript_12     49  660     611  831552547   \n",
      "2  TRINITY_DN73200_c0_g1_i1  Transcript_13     60  453     393  831552547   \n",
      "3  TRINITY_DN73242_c0_g1_i1  Transcript_14     49  273     224  831552547   \n",
      "4  TRINITY_DN73242_c0_g2_i1  Transcript_15    102  177      75  831552547   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012721397.1   PREDICTED: acidic mammalian chitinase-like [F...   \n",
      "1  XP_012728737.1   PREDICTED: vacuolar fusion protein MON1 homol...   \n",
      "2  XP_012728737.1   PREDICTED: vacuolar fusion protein MON1 homol...   \n",
      "3  XP_012728737.1   PREDICTED: vacuolar fusion protein MON1 homol...   \n",
      "4  XP_012728737.1   PREDICTED: vacuolar fusion protein MON1 homol...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234296.1                         acidic mammalian chitinase   \n",
      "1  NW_012234382.1  vacuolar fusion protein MON1 homolog B isoform X1   \n",
      "2  NW_012234382.1  vacuolar fusion protein MON1 homolog B isoform X1   \n",
      "3  NW_012234382.1  vacuolar fusion protein MON1 homolog B isoform X1   \n",
      "4  NW_012234382.1  vacuolar fusion protein MON1 homolog B isoform X1   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105928602  105928559  \n",
      "1  LOC105933658  105933650  \n",
      "2  LOC105933658  105933650  \n",
      "3  LOC105933658  105933650  \n",
      "4  LOC105933658  105933650  \n",
      "(38563, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (75067, 5)\n",
      "Unique Fhet genes: (22631, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100003  gi|831567809|ref|XP_012733833.1| PREDICTED: my...   \n",
      "1  Transcript_100003  gi|831567809|ref|XP_012733833.1| PREDICTED: my...   \n",
      "2  Transcript_100003  gi|831567809|ref|XP_012733833.1| PREDICTED: my...   \n",
      "3  Transcript_100003  gi|831567809|ref|XP_012733833.1| PREDICTED: my...   \n",
      "4  Transcript_100003  gi|831567809|ref|XP_012733833.1| PREDICTED: my...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     95  334     239  831567809  XP_012733833.1   \n",
      "1     95  334     239  831567809  XP_012733833.1   \n",
      "2     95  334     239  831567809  XP_012733833.1   \n",
      "3     95  334     239  831567809  XP_012733833.1   \n",
      "4     95  334     239  831567809  XP_012733833.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: myozenin-1-like [Fundulus heterocl...  NW_012234473.1   \n",
      "1   PREDICTED: myozenin-1-like [Fundulus heterocl...  NW_012234473.1   \n",
      "2   PREDICTED: myozenin-1-like [Fundulus heterocl...  NW_012234473.1   \n",
      "3   PREDICTED: myozenin-1-like [Fundulus heterocl...  NW_012234473.1   \n",
      "4   PREDICTED: myozenin-1-like [Fundulus heterocl...  NW_012234473.1   \n",
      "\n",
      "      product   gene     GeneID   transcript_id  \n",
      "0  myozenin-1  myoz1  105937189  XM_021324480.1  \n",
      "1  myozenin-1  myoz1  105937192  XM_012878380.2  \n",
      "2  myozenin-1  myoz1  105937187  XM_012878376.2  \n",
      "3  myozenin-1  myoz1  105937221  XM_012878418.2  \n",
      "4  myozenin-1  myoz1  105937212  XM_012878409.2  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73205_c0_g1_i1   Transcript_25      0   90      90  831524180   \n",
      "1  TRINITY_DN73217_c0_g1_i1   Transcript_27      0   28      28  831580603   \n",
      "2  TRINITY_DN81250_c0_g1_i1  Transcript_402     25  315     290  831552443   \n",
      "3  TRINITY_DN81282_c0_g1_i1  Transcript_403     25  321     296  831552443   \n",
      "4  TRINITY_DN87415_c0_g1_i1  Transcript_404      0  208     208  831552443   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012718601.1   PREDICTED: leucine-rich repeat-containing pro...   \n",
      "1  XP_012738233.1   PREDICTED: abl interactor 2-like isoform X4 [...   \n",
      "2  XP_012728699.1   PREDICTED: toll-like receptor 7 [Fundulus het...   \n",
      "3  XP_012728699.1   PREDICTED: toll-like receptor 7 [Fundulus het...   \n",
      "4  XP_012728699.1   PREDICTED: toll-like receptor 7 [Fundulus het...   \n",
      "\n",
      "  NCBI.trasncript                                    NCBI.product  \\\n",
      "0  NW_012234271.1  leucine-rich repeat-containing protein 4B-like   \n",
      "1  NW_012234577.1                abl interactor 2-like isoform X7   \n",
      "2  NW_012234381.1                            toll-like receptor 7   \n",
      "3  NW_012234381.1                            toll-like receptor 7   \n",
      "4  NW_012234381.1                            toll-like receptor 7   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105926720  105926618  \n",
      "1  LOC105940289  105940270  \n",
      "2  LOC105933632  105933587  \n",
      "3  LOC105933632  105933587  \n",
      "4  LOC105933632  105933587  \n",
      "(42091, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (41837, 5)\n",
      "Unique Fhet genes: (18280, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100003  gi|831483713|ref|XP_012735550.1| PREDICTED: FR...   \n",
      "1  Transcript_100003  gi|831483713|ref|XP_012735550.1| PREDICTED: FR...   \n",
      "2  Transcript_100003  gi|831483713|ref|XP_012735550.1| PREDICTED: FR...   \n",
      "3  Transcript_100003  gi|831483713|ref|XP_012735550.1| PREDICTED: FR...   \n",
      "4  Transcript_100003  gi|831483713|ref|XP_012735550.1| PREDICTED: FR...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0      0  424     424  831483713  XP_012735550.1   \n",
      "1      0  424     424  831483713  XP_012735550.1   \n",
      "2      0  424     424  831483713  XP_012735550.1   \n",
      "3      0  424     424  831483713  XP_012735550.1   \n",
      "4      0  424     424  831483713  XP_012735550.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: FRAS1-related extracellular matrix...  NW_012224496.1   \n",
      "1   PREDICTED: FRAS1-related extracellular matrix...  NW_012224496.1   \n",
      "2   PREDICTED: FRAS1-related extracellular matrix...  NW_012224496.1   \n",
      "3   PREDICTED: FRAS1-related extracellular matrix...  NW_012224496.1   \n",
      "4   PREDICTED: FRAS1-related extracellular matrix...  NW_012224496.1   \n",
      "\n",
      "                                             product   gene     GeneID  \\\n",
      "0  FRAS1-related extracellular matrix protein 1 i...  frem1  105938378   \n",
      "1  FRAS1-related extracellular matrix protein 1 i...  frem1  105938378   \n",
      "2  FRAS1-related extracellular matrix protein 1 i...  frem1  105938324   \n",
      "3  FRAS1-related extracellular matrix protein 1 i...  frem1  105938388   \n",
      "4  FRAS1-related extracellular matrix protein 1 i...  frem1  105938401   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_021307220.1  \n",
      "1  XM_012880096.2  \n",
      "2  XM_012880025.1  \n",
      "3  XM_012880107.2  \n",
      "4  XM_012880136.2  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     contig          seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73240_c0_g1_i1  Transcript_16     37  355     318  831482748   \n",
      "1  TRINITY_DN32561_c0_g1_i1  Transcript_34    327  557     230  831543115   \n",
      "2  TRINITY_DN32542_c0_g1_i1  Transcript_43     54  376     322  831523103   \n",
      "3  TRINITY_DN32516_c0_g1_i1  Transcript_46     20  400     380  831580562   \n",
      "4  TRINITY_DN32505_c0_g1_i1  Transcript_47    131  750     619  831580562   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012732171.1   PREDICTED: deoxyribose-phosphate aldolase [Fu...   \n",
      "1  XP_012725411.1   PREDICTED: ephrin-A5 isoform X1 [Fundulus het...   \n",
      "2  XP_012718222.1   PREDICTED: leucine-rich repeat-containing pro...   \n",
      "3  XP_012738220.1       PREDICTED: plastin-2 [Fundulus heteroclitus]   \n",
      "4  XP_012738220.1       PREDICTED: plastin-2 [Fundulus heteroclitus]   \n",
      "\n",
      "  NCBI.trasncript                               NCBI.product    gene  \\\n",
      "0  NW_012224481.1             deoxyribose-phosphate aldolase    dera   \n",
      "1  NW_012234342.1                                  ephrin-A5   efna5   \n",
      "2  NW_012234268.1  leucine-rich repeat-containing protein 59  lrrc59   \n",
      "3  NW_012234577.1                                  plastin-2    lcp1   \n",
      "4  NW_012234577.1                                  plastin-2    lcp1   \n",
      "\n",
      "   NCBI.gene  \n",
      "0  105935912  \n",
      "1  105931302  \n",
      "2  105926418  \n",
      "3  105940270  \n",
      "4  105940270  \n",
      "(23974, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (94672, 5)\n",
      "Unique Fhet genes: (24275, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100075  gi|831498985|ref|XP_012709928.1| PREDICTED: tr...   \n",
      "1  Transcript_100075  gi|831498985|ref|XP_012709928.1| PREDICTED: tr...   \n",
      "2  Transcript_100075  gi|831498985|ref|XP_012709928.1| PREDICTED: tr...   \n",
      "3  Transcript_100075  gi|831498985|ref|XP_012709928.1| PREDICTED: tr...   \n",
      "4  Transcript_100075  gi|831498985|ref|XP_012709928.1| PREDICTED: tr...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     23  176     153  831498985  XP_012709928.1   \n",
      "1     23  176     153  831498985  XP_012709928.1   \n",
      "2     23  176     153  831498985  XP_012709928.1   \n",
      "3     23  176     153  831498985  XP_012709928.1   \n",
      "4     23  176     153  831498985  XP_012709928.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: triadin-like isoform X2 [Fundulus ...  NW_012224807.1   \n",
      "1   PREDICTED: triadin-like isoform X2 [Fundulus ...  NW_012224807.1   \n",
      "2   PREDICTED: triadin-like isoform X2 [Fundulus ...  NW_012224807.1   \n",
      "3   PREDICTED: triadin-like isoform X2 [Fundulus ...  NW_012224807.1   \n",
      "4   PREDICTED: triadin-like isoform X2 [Fundulus ...  NW_012224807.1   \n",
      "\n",
      "                   product          gene     GeneID   transcript_id  \n",
      "0  triadin-like isoform X2  LOC105919208  105919208  XM_021311673.1  \n",
      "1  triadin-like isoform X2  LOC105919208  105919208  XM_012854473.2  \n",
      "2  triadin-like isoform X2  LOC105919208  105919208  XM_012854476.2  \n",
      "3  triadin-like isoform X2  LOC105919208  105919208  XM_012854474.2  \n",
      "4  triadin-like isoform X2  LOC105919208  105919208  XM_012854477.2  \n",
      "                     contig          seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32550_c0_g1_i1  Transcript_49     18  160     142  831525402   \n",
      "1  TRINITY_DN32535_c0_g1_i1  Transcript_51     21  467     446  831525402   \n",
      "2  TRINITY_DN32523_c0_g1_i1  Transcript_52     62  328     266  831525402   \n",
      "3  TRINITY_DN32599_c0_g1_i1  Transcript_54     62  299     237  831525402   \n",
      "4  TRINITY_DN32524_c0_g1_i1  Transcript_57      0  263     263  831525402   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012719037.1   PREDICTED: serine/threonine-protein kinase RI...   \n",
      "1  XP_012719037.1   PREDICTED: serine/threonine-protein kinase RI...   \n",
      "2  XP_012719037.1   PREDICTED: serine/threonine-protein kinase RI...   \n",
      "3  XP_012719037.1   PREDICTED: serine/threonine-protein kinase RI...   \n",
      "4  XP_012719037.1   PREDICTED: serine/threonine-protein kinase RI...   \n",
      "\n",
      "  NCBI.trasncript                          NCBI.product   gene  NCBI.gene  \n",
      "0  NW_012234274.1  serine/threonine-protein kinase RIO2  riok2  105926901  \n",
      "1  NW_012234274.1  serine/threonine-protein kinase RIO2  riok2  105926901  \n",
      "2  NW_012234274.1  serine/threonine-protein kinase RIO2  riok2  105926901  \n",
      "3  NW_012234274.1  serine/threonine-protein kinase RIO2  riok2  105926901  \n",
      "4  NW_012234274.1  serine/threonine-protein kinase RIO2  riok2  105926901  \n",
      "(37227, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (63254, 5)\n",
      "Unique Fhet genes: (21624, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100002  gi|831530264|ref|XP_012720768.1| PREDICTED: ST...   \n",
      "1  Transcript_100002  gi|831530264|ref|XP_012720768.1| PREDICTED: ST...   \n",
      "2  Transcript_100002  gi|831530264|ref|XP_012720768.1| PREDICTED: ST...   \n",
      "3  Transcript_100002  gi|831530264|ref|XP_012720768.1| PREDICTED: ST...   \n",
      "4  Transcript_100002  gi|831530264|ref|XP_012720768.1| PREDICTED: ST...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     11  151     140  831530264  XP_012720768.1   \n",
      "1     11  151     140  831530264  XP_012720768.1   \n",
      "2     11  151     140  831530264  XP_012720768.1   \n",
      "3     11  151     140  831530264  XP_012720768.1   \n",
      "4     11  151     140  831530264  XP_012720768.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: STAM-binding protein [Fundulus het...  NW_012234291.1   \n",
      "1   PREDICTED: STAM-binding protein [Fundulus het...  NW_012234291.1   \n",
      "2   PREDICTED: STAM-binding protein [Fundulus het...  NW_012234291.1   \n",
      "3   PREDICTED: STAM-binding protein [Fundulus het...  NW_012234291.1   \n",
      "4   PREDICTED: STAM-binding protein [Fundulus het...  NW_012234291.1   \n",
      "\n",
      "                product    gene     GeneID   transcript_id  \n",
      "0  STAM-binding protein  stambp  105928165  XR_001166324.2  \n",
      "1  STAM-binding protein  stambp  105928165  XR_002428305.1  \n",
      "2  STAM-binding protein  stambp  105928165  XM_021317915.1  \n",
      "3  STAM-binding protein  stambp  105928165  XM_012865301.2  \n",
      "4  STAM-binding protein  stambp  105928165  XM_012865302.2  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32549_c0_g1_i1   Transcript_82     50  228     178  831555246   \n",
      "1  TRINITY_DN81321_c1_g1_i1  Transcript_107     13  193     180  831517887   \n",
      "2  TRINITY_DN81374_c0_g1_i1  Transcript_108      0  172     172  831517887   \n",
      "3   TRINITY_DN7003_c0_g1_i1  Transcript_179      0  156     156  831544190   \n",
      "4  TRINITY_DN64187_c0_g2_i1  Transcript_241     35  316     281  831570358   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012729643.1   PREDICTED: tubulin polymerization-promoting p...   \n",
      "1  XP_012716379.1   PREDICTED: alpha-2-HS-glycoprotein-like [Fund...   \n",
      "2  XP_012716379.1   PREDICTED: alpha-2-HS-glycoprotein-like [Fund...   \n",
      "3  XP_012725788.1   PREDICTED: uncharacterized protein LOC1059315...   \n",
      "4  XP_012734689.1   PREDICTED: RNA/RNP complex-1-interacting phos...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234398.1  tubulin polymerization-promoting protein famil...   \n",
      "1  NW_012234254.1                            alpha-2-HS-glycoprotein   \n",
      "2  NW_012234254.1                            alpha-2-HS-glycoprotein   \n",
      "3  NW_012234344.1    uncharacterized protein LOC105931591 isoform X1   \n",
      "4  NW_012234493.1  RNA/RNP complex-1-interacting phosphatase-like...   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105934278  105934258  \n",
      "1  LOC105925199  105925170  \n",
      "2  LOC105925199  105925170  \n",
      "3  LOC105931591  105931550  \n",
      "4  LOC105937751  105937761  \n",
      "(35049, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (73458, 5)\n",
      "Unique Fhet genes: (21806, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100005  gi|831543230|ref|XP_012725447.1| PREDICTED: F-...   \n",
      "1  Transcript_100005  gi|831543230|ref|XP_012725447.1| PREDICTED: F-...   \n",
      "2  Transcript_100005  gi|831543230|ref|XP_012725447.1| PREDICTED: F-...   \n",
      "3  Transcript_100005  gi|831543230|ref|XP_012725447.1| PREDICTED: F-...   \n",
      "4  Transcript_100005  gi|831543230|ref|XP_012725447.1| PREDICTED: F-...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0      0  650     650  831543230  XP_012725447.1   \n",
      "1      0  650     650  831543230  XP_012725447.1   \n",
      "2      0  650     650  831543230  XP_012725447.1   \n",
      "3      0  650     650  831543230  XP_012725447.1   \n",
      "4      0  650     650  831543230  XP_012725447.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: F-box/LRR-repeat protein 17 [Fundu...  NW_012234342.1   \n",
      "1   PREDICTED: F-box/LRR-repeat protein 17 [Fundu...  NW_012234342.1   \n",
      "2   PREDICTED: F-box/LRR-repeat protein 17 [Fundu...  NW_012234342.1   \n",
      "3   PREDICTED: F-box/LRR-repeat protein 17 [Fundu...  NW_012234342.1   \n",
      "4   PREDICTED: F-box/LRR-repeat protein 17 [Fundu...  NW_012234342.1   \n",
      "\n",
      "                       product    gene     GeneID   transcript_id  \n",
      "0  F-box/LRR-repeat protein 17  fbxl17  105931302  XM_012869928.2  \n",
      "1  F-box/LRR-repeat protein 17  fbxl17  105931348  XM_021320235.1  \n",
      "2  F-box/LRR-repeat protein 17  fbxl17  105931348  XM_012869982.2  \n",
      "3  F-box/LRR-repeat protein 17  fbxl17  105931303  XM_021320228.1  \n",
      "4  F-box/LRR-repeat protein 17  fbxl17  105931344  XM_012869974.2  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73209_c0_g1_i1   Transcript_26      0   66      66  831534962   \n",
      "1  TRINITY_DN73217_c0_g1_i1   Transcript_27      0   51      51  831534962   \n",
      "2  TRINITY_DN32550_c0_g1_i1   Transcript_49      0   21      21  831522803   \n",
      "3  TRINITY_DN81321_c1_g1_i1  Transcript_107      1   91      90  831493469   \n",
      "4  TRINITY_DN81390_c0_g1_i1  Transcript_113     18  125     107  831473992   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012722439.1   PREDICTED: basement membrane-specific heparan...   \n",
      "1  XP_012722439.1   PREDICTED: basement membrane-specific heparan...   \n",
      "2  XP_012718119.1   PREDICTED: chromobox protein homolog 3-like i...   \n",
      "3  XP_012707969.1   PREDICTED: cAMP-specific 3',5'-cyclic phospho...   \n",
      "4  XP_012717323.1   PREDICTED: glutamate receptor 3 isoform X3 [F...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234307.1  basement membrane-specific heparan sulfate pro...   \n",
      "1  NW_012234307.1  basement membrane-specific heparan sulfate pro...   \n",
      "2  NW_012234267.1             chromobox protein homolog 3 isoform X1   \n",
      "3  NW_012224662.1  cAMP-specific 3'%2C5'-cyclic phosphodiesterase...   \n",
      "4  NW_012224401.1                    glutamate receptor 3 isoform X3   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105929277  105929256  \n",
      "1  LOC105929277  105929256  \n",
      "2  LOC105926367  105926280  \n",
      "3  LOC105917643  105917640  \n",
      "4  LOC105925698  105920463  \n",
      "(41755, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (62852, 5)\n",
      "Unique Fhet genes: (20647, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100015  gi|831480331|ref|XP_012723558.1| PREDICTED: cy...   \n",
      "1  Transcript_100015  gi|831480331|ref|XP_012723558.1| PREDICTED: cy...   \n",
      "2  Transcript_100015  gi|831480331|ref|XP_012723558.1| PREDICTED: cy...   \n",
      "3  Transcript_100015  gi|831480331|ref|XP_012723558.1| PREDICTED: cy...   \n",
      "4  Transcript_100015  gi|831480331|ref|XP_012723558.1| PREDICTED: cy...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     19  478     459  831480331  XP_012723558.1   \n",
      "1     19  478     459  831480331  XP_012723558.1   \n",
      "2     19  478     459  831480331  XP_012723558.1   \n",
      "3     19  478     459  831480331  XP_012723558.1   \n",
      "4     19  478     459  831480331  XP_012723558.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: cytochrome b-c1 complex subunit 2,...  NW_012224459.1   \n",
      "1   PREDICTED: cytochrome b-c1 complex subunit 2,...  NW_012224459.1   \n",
      "2   PREDICTED: cytochrome b-c1 complex subunit 2,...  NW_012224459.1   \n",
      "3   PREDICTED: cytochrome b-c1 complex subunit 2,...  NW_012224459.1   \n",
      "4   PREDICTED: cytochrome b-c1 complex subunit 2,...  NW_012224459.1   \n",
      "\n",
      "                                             product          gene     GeneID  \\\n",
      "0  cytochrome b-c1 complex subunit 2%2C mitochond...  LOC105930091  105929961   \n",
      "1  cytochrome b-c1 complex subunit 2%2C mitochond...  LOC105930091  105929948   \n",
      "2  cytochrome b-c1 complex subunit 2%2C mitochond...  LOC105930091  105929948   \n",
      "3  cytochrome b-c1 complex subunit 2%2C mitochond...  LOC105930091  105929948   \n",
      "4  cytochrome b-c1 complex subunit 2%2C mitochond...  LOC105930091  105929948   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_021319363.1  \n",
      "1  XM_021319344.1  \n",
      "2  XM_021319339.1  \n",
      "3  XM_021319355.1  \n",
      "4  XM_021319349.1  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32517_c0_g2_i1   Transcript_40      5   78      73  831484161   \n",
      "1  TRINITY_DN56987_c0_g1_i1  Transcript_166      0   75      75  831536829   \n",
      "2   TRINITY_DN7001_c0_g1_i1  Transcript_187      0   66      66  831491733   \n",
      "3  TRINITY_DN48699_c0_g1_i1  Transcript_217      0   76      76  831523926   \n",
      "4  TRINITY_DN48605_c0_g1_i1  Transcript_235     37   77      40  831529475   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012737200.1   PREDICTED: calmodulin-regulated spectrin-asso...   \n",
      "1  XP_012723102.1   PREDICTED: endophilin-A3 [Fundulus heteroclitus]   \n",
      "2  XP_012707359.1   PREDICTED: GTP-binding protein REM 2 [Fundulu...   \n",
      "3  XP_012718509.1   PREDICTED: heparan sulfate glucosamine 3-O-su...   \n",
      "4  XP_012720487.1   PREDICTED: serine/arginine repetitive matrix ...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012224503.1  calmodulin-regulated spectrin-associated prote...   \n",
      "1  NW_012234311.1                           endophilin-A3 isoform X1   \n",
      "2  NW_012224629.1                          GTP-binding protein REM 2   \n",
      "3  NW_012234271.1  heparan sulfate glucosamine 3-O-sulfotransfera...   \n",
      "4  NW_012234288.1        serine/arginine repetitive matrix protein 3   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105939518  105939452  \n",
      "1  LOC105929749  105929642  \n",
      "2  LOC105917180  105917174  \n",
      "3  LOC105926660  105926618  \n",
      "4         srrm3  105927980  \n",
      "(35861, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (81920, 5)\n",
      "Unique Fhet genes: (22718, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100003  gi|831474071|ref|XP_012719905.1| PREDICTED: PD...   \n",
      "1  Transcript_100003  gi|831474071|ref|XP_012719905.1| PREDICTED: PD...   \n",
      "2  Transcript_100003  gi|831474071|ref|XP_012719905.1| PREDICTED: PD...   \n",
      "3  Transcript_100003  gi|831474071|ref|XP_012719905.1| PREDICTED: PD...   \n",
      "4  Transcript_100003  gi|831474071|ref|XP_012719905.1| PREDICTED: PD...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0      0  153     153  831474071  XP_012719905.1   \n",
      "1      0  153     153  831474071  XP_012719905.1   \n",
      "2      0  153     153  831474071  XP_012719905.1   \n",
      "3      0  153     153  831474071  XP_012719905.1   \n",
      "4      0  153     153  831474071  XP_012719905.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: PDZ and LIM domain protein 7 isofo...  NW_012224401.1   \n",
      "1   PREDICTED: PDZ and LIM domain protein 7 isofo...  NW_012224401.1   \n",
      "2   PREDICTED: PDZ and LIM domain protein 7 isofo...  NW_012224401.1   \n",
      "3   PREDICTED: PDZ and LIM domain protein 7 isofo...  NW_012224401.1   \n",
      "4   PREDICTED: PDZ and LIM domain protein 7 isofo...  NW_012224401.1   \n",
      "\n",
      "                                   product    gene     GeneID   transcript_id  \n",
      "0  PDZ and LIM domain protein 7 isoform X2  pdlim7  105920463  XM_012856390.2  \n",
      "1  PDZ and LIM domain protein 7 isoform X2  pdlim7  105915213  XM_021321288.1  \n",
      "2  PDZ and LIM domain protein 7 isoform X2  pdlim7  105915271  XM_012849362.2  \n",
      "3  PDZ and LIM domain protein 7 isoform X2  pdlim7  105915271  XM_012849437.2  \n",
      "4  PDZ and LIM domain protein 7 isoform X2  pdlim7  105915419  XM_021323826.1  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32564_c0_g1_i1   Transcript_44      0  120     120  831500865   \n",
      "1  TRINITY_DN48652_c0_g1_i1  Transcript_203     21   68      47  831488241   \n",
      "2  TRINITY_DN48646_c0_g1_i1  Transcript_207     51  388     337  831484967   \n",
      "3  TRINITY_DN23110_c0_g1_i1  Transcript_282      0   93      93  831542339   \n",
      "4  TRINITY_DN23122_c0_g1_i1  Transcript_317     35   87      52  831550128   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012710584.1   PREDICTED: UPF0606 protein KIAA1549L homolog ...   \n",
      "1  XP_012706118.1   PREDICTED: transmembrane protein 178B-like [F...   \n",
      "2  XP_012704904.1   PREDICTED: uncharacterized protein LOC1059153...   \n",
      "3  XP_012725041.1   PREDICTED: corticotropin-releasing factor-bin...   \n",
      "4  XP_012727885.1   PREDICTED: synapse differentiation-inducing g...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012224869.1       UPF0606 protein KIAA1549L homolog isoform X1   \n",
      "1  NW_012224574.1                    transmembrane protein 178B-like   \n",
      "2  NW_012224519.1               uncharacterized protein LOC105915370   \n",
      "3  NW_012234337.1     corticotropin-releasing factor-binding protein   \n",
      "4  NW_012234371.1  synapse differentiation-inducing gene protein ...   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0     kiaa1549l  105919702  \n",
      "1  LOC105916264  105916268  \n",
      "2  LOC105915370  105915372  \n",
      "3         crhbp  110368956  \n",
      "4  LOC105933059  105933072  \n",
      "(44176, 12)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (74555, 5)\n",
      "Unique Fhet genes: (22338, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100000  gi|831539324|ref|XP_012723974.1| PREDICTED: ad...   \n",
      "1  Transcript_100000  gi|831539324|ref|XP_012723974.1| PREDICTED: ad...   \n",
      "2  Transcript_100000  gi|831539324|ref|XP_012723974.1| PREDICTED: ad...   \n",
      "3  Transcript_100000  gi|831539324|ref|XP_012723974.1| PREDICTED: ad...   \n",
      "4  Transcript_100000  gi|831539324|ref|XP_012723974.1| PREDICTED: ad...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     73  132      59  831539324  XP_012723974.1   \n",
      "1     73  132      59  831539324  XP_012723974.1   \n",
      "2     73  132      59  831539324  XP_012723974.1   \n",
      "3     73  132      59  831539324  XP_012723974.1   \n",
      "4     73  132      59  831539324  XP_012723974.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: adenylate cyclase type 5 isoform X...  NW_012234322.1   \n",
      "1   PREDICTED: adenylate cyclase type 5 isoform X...  NW_012234322.1   \n",
      "2   PREDICTED: adenylate cyclase type 5 isoform X...  NW_012234322.1   \n",
      "3   PREDICTED: adenylate cyclase type 5 isoform X...  NW_012234322.1   \n",
      "4   PREDICTED: adenylate cyclase type 5 isoform X...  NW_012234322.1   \n",
      "\n",
      "                               product   gene     GeneID   transcript_id  \n",
      "0  adenylate cyclase type 5 isoform X2  adcy5  105930348  XM_021319478.1  \n",
      "1  adenylate cyclase type 5 isoform X2  adcy5  105930315  XM_021319473.1  \n",
      "2  adenylate cyclase type 5 isoform X2  adcy5  105930314  XM_012868460.2  \n",
      "3  adenylate cyclase type 5 isoform X2  adcy5  105930335  XM_012868512.2  \n",
      "4  adenylate cyclase type 5 isoform X2  adcy5  105930335  XM_012868510.2  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN56913_c0_g1_i1  Transcript_116     61   83      22  831573625   \n",
      "1   TRINITY_DN7039_c0_g1_i1  Transcript_178     33   69      36  831578823   \n",
      "2   TRINITY_DN7003_c0_g1_i1  Transcript_179     33   67      34  831578823   \n",
      "3   TRINITY_DN7063_c0_g2_i1  Transcript_184      0   88      88  831519976   \n",
      "4  TRINITY_DN48670_c0_g1_i1  Transcript_227     61   91      30  831513323   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012735765.1   PREDICTED: pleckstrin homology domain-contain...   \n",
      "1  XP_012737611.1   PREDICTED: zinc finger and BTB domain-contain...   \n",
      "2  XP_012737611.1   PREDICTED: zinc finger and BTB domain-contain...   \n",
      "3  XP_012717112.1   PREDICTED: teashirt homolog 3 [Fundulus heter...   \n",
      "4  XP_012714877.1   PREDICTED: dynein assembly factor with WDR re...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234517.1  pleckstrin homology domain-containing family A...   \n",
      "1  NW_012234559.1  zinc finger and BTB domain-containing protein ...   \n",
      "2  NW_012234559.1  zinc finger and BTB domain-containing protein ...   \n",
      "3  NW_012234259.1                                 teashirt homolog 3   \n",
      "4  NW_012226213.1  dynein assembly factor with WDR repeat domains...   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105938538  105938477  \n",
      "1  LOC105939840  105939850  \n",
      "2  LOC105939840  105939850  \n",
      "3         tshz3  105925649  \n",
      "4          daw1  105923370  \n",
      "(40219, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (78820, 5)\n",
      "Unique Fhet genes: (22849, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100000  gi|831558983|ref|XP_012730895.1| PREDICTED: or...   \n",
      "1  Transcript_100000  gi|831558983|ref|XP_012730895.1| PREDICTED: or...   \n",
      "2  Transcript_100000  gi|831558983|ref|XP_012730895.1| PREDICTED: or...   \n",
      "3  Transcript_100000  gi|831558983|ref|XP_012730895.1| PREDICTED: or...   \n",
      "4  Transcript_100000  gi|831558983|ref|XP_012730895.1| PREDICTED: or...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     51  517     466  831558983  XP_012730895.1   \n",
      "1     51  517     466  831558983  XP_012730895.1   \n",
      "2     51  517     466  831558983  XP_012730895.1   \n",
      "3     51  517     466  831558983  XP_012730895.1   \n",
      "4     51  517     466  831558983  XP_012730895.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: origin recognition complex subunit...  NW_012234418.1   \n",
      "1   PREDICTED: origin recognition complex subunit...  NW_012234418.1   \n",
      "2   PREDICTED: origin recognition complex subunit...  NW_012234418.1   \n",
      "3   PREDICTED: origin recognition complex subunit...  NW_012234418.1   \n",
      "4   PREDICTED: origin recognition complex subunit...  NW_012234418.1   \n",
      "\n",
      "                                     product          gene     GeneID  \\\n",
      "0  origin recognition complex subunit 4-like  LOC105935160  105935146   \n",
      "1  origin recognition complex subunit 4-like  LOC105935160  105935083   \n",
      "2  origin recognition complex subunit 4-like  LOC105935160  105935132   \n",
      "3  origin recognition complex subunit 4-like  LOC105935160  105935132   \n",
      "4  origin recognition complex subunit 4-like  LOC105935160  105935157   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_021323008.1  \n",
      "1  XM_012875310.2  \n",
      "2  XM_021322934.1  \n",
      "3  XM_012875401.2  \n",
      "4  XM_012875436.2  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73227_c0_g1_i1    Transcript_4      0  178     178  831554879   \n",
      "1  TRINITY_DN73216_c0_g1_i1   Transcript_29     36   75      39  831570279   \n",
      "2  TRINITY_DN32566_c0_g1_i1   Transcript_30      0  130     130  831570279   \n",
      "3  TRINITY_DN81374_c0_g1_i1  Transcript_108      0   68      68  831551401   \n",
      "4  TRINITY_DN81374_c1_g1_i1  Transcript_109      0   72      72  831551401   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012729519.1   PREDICTED: G protein-activated inward rectifi...   \n",
      "1  XP_012734659.1   PREDICTED: snRNA-activating protein complex s...   \n",
      "2  XP_012734659.1   PREDICTED: snRNA-activating protein complex s...   \n",
      "3  XP_012728336.1   PREDICTED: POU domain, class 3, transcription...   \n",
      "4  XP_012728336.1   PREDICTED: POU domain, class 3, transcription...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234395.1  G protein-activated inward rectifier potassium...   \n",
      "1  NW_012234492.1    snRNA-activating protein complex subunit 1-like   \n",
      "2  NW_012234492.1    snRNA-activating protein complex subunit 1-like   \n",
      "3  NW_012234377.1    POU domain%2C class 3%2C transcription factor 2   \n",
      "4  NW_012234377.1    POU domain%2C class 3%2C transcription factor 2   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0         kcnj3  105934158  \n",
      "1  LOC105937725  105937719  \n",
      "2  LOC105937725  105937719  \n",
      "3        pou3f2  105933429  \n",
      "4        pou3f2  105933429  \n",
      "(33424, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (73426, 5)\n",
      "Unique Fhet genes: (22601, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100000  gi|831537298|ref|XP_012723266.1| PREDICTED: un...   \n",
      "1  Transcript_100000  gi|831537298|ref|XP_012723266.1| PREDICTED: un...   \n",
      "2  Transcript_100000  gi|831537298|ref|XP_012723266.1| PREDICTED: un...   \n",
      "3  Transcript_100000  gi|831537298|ref|XP_012723266.1| PREDICTED: un...   \n",
      "4  Transcript_100000  gi|831537298|ref|XP_012723266.1| PREDICTED: un...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0      0  130     130  831537298  XP_012723266.1   \n",
      "1      0  130     130  831537298  XP_012723266.1   \n",
      "2      0  130     130  831537298  XP_012723266.1   \n",
      "3      0  130     130  831537298  XP_012723266.1   \n",
      "4      0  130     130  831537298  XP_012723266.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: unconventional myosin-VI-like isof...  NW_012234314.1   \n",
      "1   PREDICTED: unconventional myosin-VI-like isof...  NW_012234314.1   \n",
      "2   PREDICTED: unconventional myosin-VI-like isof...  NW_012234314.1   \n",
      "3   PREDICTED: unconventional myosin-VI-like isof...  NW_012234314.1   \n",
      "4   PREDICTED: unconventional myosin-VI-like isof...  NW_012234314.1   \n",
      "\n",
      "                                    product          gene     GeneID  \\\n",
      "0  unconventional myosin-VI-like isoform X4  LOC105929881  105929877   \n",
      "1  unconventional myosin-VI-like isoform X4  LOC105929881  105929877   \n",
      "2  unconventional myosin-VI-like isoform X4  LOC105929881  105929877   \n",
      "3  unconventional myosin-VI-like isoform X4  LOC105929881  105929875   \n",
      "4  unconventional myosin-VI-like isoform X4  LOC105929881  105929876   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_021319148.1  \n",
      "1  XM_021319149.1  \n",
      "2  XM_021319147.1  \n",
      "3  XM_012867799.2  \n",
      "4  XM_012867801.2  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32550_c0_g1_i1   Transcript_49      1  206     205  831545408   \n",
      "1  TRINITY_DN32550_c0_g2_i1   Transcript_50      0  172     172  831538787   \n",
      "2  TRINITY_DN32577_c0_g1_i1   Transcript_63      0   55      55  831503985   \n",
      "3  TRINITY_DN67518_c1_g1_i1  Transcript_357      0  138     138  831528929   \n",
      "4  TRINITY_DN87446_c0_g1_i1  Transcript_407     47  211     164  831520549   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012726222.1   PREDICTED: photoreceptor-specific nuclear rec...   \n",
      "1  XP_012723788.1   PREDICTED: protocadherin-8 [Fundulus heterocl...   \n",
      "2  XP_012711686.1   PREDICTED: solute carrier family 13 member 5-...   \n",
      "3  XP_012720290.1   PREDICTED: choline transporter-like protein 5...   \n",
      "4  XP_012717313.1   PREDICTED: UDP-N-acetylglucosamine transferas...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234349.1       photoreceptor-specific nuclear receptor-like   \n",
      "1  NW_012234321.1                                    protocadherin-8   \n",
      "2  NW_012224983.1       solute carrier family 13 member 5 isoform X1   \n",
      "3  NW_012234285.1      choline transporter-like protein 5 isoform X1   \n",
      "4  NW_012234260.1  UDP-N-acetylglucosamine transferase subunit AL...   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105931886  105931902  \n",
      "1         pcdh8  105930227  \n",
      "2  LOC105920619  105920616  \n",
      "3       slc44a5  110368665  \n",
      "4  LOC105925823  105925704  \n",
      "(41184, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (57035, 5)\n",
      "Unique Fhet genes: (20439, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100002  gi|831552584|ref|XP_012728750.1| PREDICTED: ep...   \n",
      "1  Transcript_100002  gi|831552584|ref|XP_012728750.1| PREDICTED: ep...   \n",
      "2  Transcript_100002  gi|831552584|ref|XP_012728750.1| PREDICTED: ep...   \n",
      "3  Transcript_100002  gi|831552584|ref|XP_012728750.1| PREDICTED: ep...   \n",
      "4  Transcript_100002  gi|831552584|ref|XP_012728750.1| PREDICTED: ep...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     37  983     946  831552584  XP_012728750.1   \n",
      "1     37  983     946  831552584  XP_012728750.1   \n",
      "2     37  983     946  831552584  XP_012728750.1   \n",
      "3     37  983     946  831552584  XP_012728750.1   \n",
      "4     37  983     946  831552584  XP_012728750.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: ephrin type-A receptor 2-like [Fun...  NW_012234382.1   \n",
      "1   PREDICTED: ephrin type-A receptor 2-like [Fun...  NW_012234382.1   \n",
      "2   PREDICTED: ephrin type-A receptor 2-like [Fun...  NW_012234382.1   \n",
      "3   PREDICTED: ephrin type-A receptor 2-like [Fun...  NW_012234382.1   \n",
      "4   PREDICTED: ephrin type-A receptor 2-like [Fun...  NW_012234382.1   \n",
      "\n",
      "                         product          gene     GeneID   transcript_id  \n",
      "0  ephrin type-A receptor 2-like  LOC105933666  105933650  XM_021321885.1  \n",
      "1  ephrin type-A receptor 2-like  LOC105933666  105933677  XM_021321925.1  \n",
      "2  ephrin type-A receptor 2-like  LOC105933666  105933677  XM_021321926.1  \n",
      "3  ephrin type-A receptor 2-like  LOC105933666  105933651  XM_021321927.1  \n",
      "4  ephrin type-A receptor 2-like  LOC105933666  105933681  XM_021321902.1  \n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN32573_c0_g1_i1   Transcript_71      0  116     116  831571436   \n",
      "1  TRINITY_DN32562_c1_g1_i1   Transcript_74      0   68      68  831492447   \n",
      "2  TRINITY_DN32582_c0_g1_i1   Transcript_85     16   44      28  831529266   \n",
      "3  TRINITY_DN56926_c1_g1_i1  Transcript_140      0   61      61  831497728   \n",
      "4  TRINITY_DN56984_c0_g2_i1  Transcript_152      0   23      23  831564224   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012735038.1   PREDICTED: protocadherin-16-like [Fundulus he...   \n",
      "1  XP_012707613.1   PREDICTED: ankyrin repeat domain-containing p...   \n",
      "2  XP_012720412.1   PREDICTED: calmodulin-lysine N-methyltransfer...   \n",
      "3  XP_012709481.1   PREDICTED: nuclear receptor ROR-alpha-like is...   \n",
      "4  XP_012732631.1   PREDICTED: taste receptor type 1 member 1 [Fu...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012234501.1                              protocadherin-16-like   \n",
      "1  NW_012224642.1  ankyrin repeat domain-containing protein 34B-like   \n",
      "2  NW_012234287.1              calmodulin-lysine N-methyltransferase   \n",
      "3  NW_012224770.1         nuclear receptor ROR-alpha-like isoform X1   \n",
      "4  NW_012234452.1                     taste receptor type 1 member 2   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0  LOC105937994  105937982  \n",
      "1  LOC105917370  105917366  \n",
      "2        camkmt  105927939  \n",
      "3  LOC105918865  110367056  \n",
      "4        tas1r2  105936411  \n",
      "(32741, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (54836, 5)\n",
      "Unique Fhet genes: (20382, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100001  gi|831492530|ref|XP_012707644.1| PREDICTED: ex...   \n",
      "1  Transcript_100001  gi|831492530|ref|XP_012707644.1| PREDICTED: ex...   \n",
      "2  Transcript_100001  gi|831492530|ref|XP_012707644.1| PREDICTED: ex...   \n",
      "3  Transcript_100001  gi|831492530|ref|XP_012707644.1| PREDICTED: ex...   \n",
      "4  Transcript_100001  gi|831492530|ref|XP_012707644.1| PREDICTED: ex...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0     63   95      32  831492530  XP_012707644.1   \n",
      "1     63   95      32  831492530  XP_012707644.1   \n",
      "2     63   95      32  831492530  XP_012707644.1   \n",
      "3     63   95      32  831492530  XP_012707644.1   \n",
      "4     63   95      32  831492530  XP_012707644.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: exostosin-1c-like [Fundulus hetero...  NW_012224643.1   \n",
      "1   PREDICTED: exostosin-1c-like [Fundulus hetero...  NW_012224643.1   \n",
      "2   PREDICTED: exostosin-1c-like [Fundulus hetero...  NW_012224643.1   \n",
      "3   PREDICTED: exostosin-1c-like [Fundulus hetero...  NW_012224643.1   \n",
      "4   PREDICTED: exostosin-1c-like [Fundulus hetero...  NW_012224643.1   \n",
      "\n",
      "             product          gene     GeneID   transcript_id  \n",
      "0  exostosin-1c-like  LOC105917394  110366876  XR_002427042.1  \n",
      "1  exostosin-1c-like  LOC105917394  105917378  XM_021310386.1  \n",
      "2  exostosin-1c-like  LOC105917394  105917379  XM_021310387.1  \n",
      "3  exostosin-1c-like  LOC105917394  110366870  XR_002427040.1  \n",
      "4  exostosin-1c-like  LOC105917394  105917391  XM_012852186.2  \n",
      "                     contig          seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN73231_c1_g1_i1  Transcript_10      0  122     122  831499494   \n",
      "1  TRINITY_DN32508_c0_g1_i1  Transcript_41      0   73      73  831575894   \n",
      "2  TRINITY_DN32524_c1_g1_i1  Transcript_58      1   54      53  831483578   \n",
      "3  TRINITY_DN32538_c0_g1_i1  Transcript_70     81  190     109  831549865   \n",
      "4  TRINITY_DN32585_c0_g1_i1  Transcript_75    155  240      85  831475751   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012710103.1   PREDICTED: ERI1 exoribonuclease 2 [Fundulus h...   \n",
      "1  XP_012736574.1   PREDICTED: E3 ubiquitin-protein ligase MARCH9...   \n",
      "2  XP_012735121.1   PREDICTED: corticoliberin [Fundulus heterocli...   \n",
      "3  XP_012727793.1   PREDICTED: parvalbumin, thymic CPV3-like [Fun...   \n",
      "4  XP_012708360.1   PREDICTED: leucine-rich repeat transmembrane ...   \n",
      "\n",
      "  NCBI.trasncript                                       NCBI.product  \\\n",
      "0  NW_012224821.1                             ERI1 exoribonuclease 2   \n",
      "1  NW_012234529.1            E3 ubiquitin-protein ligase MARCH9-like   \n",
      "2  NW_012224494.1                                     corticoliberin   \n",
      "3  NW_012234370.1                         parvalbumin%2C thymic CPV3   \n",
      "4  NW_012224404.1  leucine-rich repeat transmembrane protein FLRT...   \n",
      "\n",
      "           gene  NCBI.gene  \n",
      "0          eri2  105919335  \n",
      "1  LOC105939084  105939110  \n",
      "2           crh  105937907  \n",
      "3  LOC105932983  105933022  \n",
      "4  LOC105917946  105916749  \n",
      "(30706, 12)\n",
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (77688, 5)\n",
      "Unique Fhet genes: (22994, 5)\n",
      "             seqid_x                                               Name  \\\n",
      "0  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
      "1  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
      "2  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
      "3  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
      "4  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
      "\n",
      "   start  end  length         id      protein_id  \\\n",
      "0      0  122     122  831554709  XP_012729460.1   \n",
      "1      0  122     122  831554709  XP_012729460.1   \n",
      "2      0  122     122  831554709  XP_012729460.1   \n",
      "3      0  122     122  831554709  XP_012729460.1   \n",
      "4      0  122     122  831554709  XP_012729460.1   \n",
      "\n",
      "                                               Notes         seqid_y  \\\n",
      "0   PREDICTED: multidrug resistance-associated pr...  NW_012234394.1   \n",
      "1   PREDICTED: multidrug resistance-associated pr...  NW_012234394.1   \n",
      "2   PREDICTED: multidrug resistance-associated pr...  NW_012234394.1   \n",
      "3   PREDICTED: multidrug resistance-associated pr...  NW_012234394.1   \n",
      "4   PREDICTED: multidrug resistance-associated pr...  NW_012234394.1   \n",
      "\n",
      "                                     product    gene     GeneID  \\\n",
      "0  multidrug resistance-associated protein 9  abcc12  105934121   \n",
      "1  multidrug resistance-associated protein 9  abcc12  105934136   \n",
      "2  multidrug resistance-associated protein 9  abcc12  105934136   \n",
      "3  multidrug resistance-associated protein 9  abcc12  105934136   \n",
      "4  multidrug resistance-associated protein 9  abcc12  105934144   \n",
      "\n",
      "    transcript_id  \n",
      "0  XM_021322283.1  \n",
      "1  XM_012873992.2  \n",
      "2  XM_012873993.2  \n",
      "3  XM_012873994.2  \n",
      "4  XM_012874005.2  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN81334_c0_g1_i1   Transcript_95     28   90      62  831497731   \n",
      "1  TRINITY_DN56987_c0_g1_i1  Transcript_166      0   72      72  831491849   \n",
      "2  TRINITY_DN48616_c0_g1_i1  Transcript_233     62   86      24  831484829   \n",
      "3  TRINITY_DN33983_c0_g1_i1  Transcript_586      5   40      35  831576519   \n",
      "4  TRINITY_DN33983_c0_g2_i1  Transcript_587      0   23      23  831576519   \n",
      "\n",
      "     NCBI.protein                                              Notes  \\\n",
      "0  XP_012709482.1   PREDICTED: nuclear receptor ROR-alpha-like is...   \n",
      "1  XP_012707401.1   PREDICTED: protein delta homolog 1 [Fundulus ...   \n",
      "2  XP_012704828.1   PREDICTED: homeobox protein goosecoid-2 [Fund...   \n",
      "3  XP_012736794.1   PREDICTED: uncharacterized protein LOC1059392...   \n",
      "4  XP_012736794.1   PREDICTED: uncharacterized protein LOC1059392...   \n",
      "\n",
      "  NCBI.trasncript                                NCBI.product          gene  \\\n",
      "0  NW_012224770.1  nuclear receptor ROR-alpha-like isoform X2  LOC105918865   \n",
      "1  NW_012224631.1                     protein delta homolog 1          dlk1   \n",
      "2  NW_012224516.1                homeobox protein goosecoid-2          gsc2   \n",
      "3  NW_012234534.1        uncharacterized protein LOC105939251  LOC105939251   \n",
      "4  NW_012234534.1        uncharacterized protein LOC105939251  LOC105939251   \n",
      "\n",
      "   NCBI.gene  \n",
      "0  110367056  \n",
      "1  105917211  \n",
      "2  105915293  \n",
      "3  105939231  \n",
      "4  105939231  \n",
      "(23987, 12)\n"
     ]
    }
   ],
   "source": [
    "# print number of Fhet \"genes\" in each:\n",
    "Fhet_counts_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/Fhet_counts/\"\n",
    "Fhet_counts_ncbi = \"/Users/johnsolk/Documents/UCDavis/Whitehead/Fhet_counts_ncbi/\"\n",
    "files = os.listdir(Fhet_counts_dir)\n",
    "print(files)\n",
    "for table_file in files:\n",
    "    if table_file.endswith(\".csv\"):\n",
    "        table = pd.read_csv(Fhet_counts_dir + table_file)\n",
    "        print(\"Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig):\",table.groupby('seqid').count().shape)\n",
    "        print(\"Unique Fhet genes:\",table.groupby('Name').count().shape)\n",
    "        table['split_1'], table['split_2'], table['split_3'],table['split_4'],table['split_5'] = table['Name'].str.split('|',0).str\n",
    "        table.columns = ['index','seqid', 'Name', 'start', 'end', 'length', 'split_1', 'id','split_3', 'protein_id', 'Notes']\n",
    "        table = table[['seqid', 'Name', 'start', 'end', 'length', 'id','protein_id', 'Notes']]\n",
    "        table = pd.merge(table,protein_transcript,how=\"inner\",on=\"protein_id\")\n",
    "        print(table.head())\n",
    "        table = table.drop_duplicates(subset='seqid_x')[['seqid_x', 'Name', 'start', 'end', 'length', 'id','protein_id', 'Notes','seqid_y','product','gene','GeneID']]\n",
    "        # there are multiple NCBI GeneID and tarnscript_id per seqid (Trinity transcript)\n",
    "        # Yet, it's the same protein...\n",
    "        # Meaning, the Fhet reference is fragemented\n",
    "        table = table[['seqid_x', 'start', 'end', 'length', 'id','protein_id', 'Notes','seqid_y','product','gene','GeneID']]\n",
    "        table.columns = ['seqid', 'start', 'end', 'length', 'NCBI.id','NCBI.protein', 'Notes','NCBI.trasncript','NCBI.product','gene','NCBI.gene']\n",
    "        conversion_file = [i for i in conversions if i.startswith(species)][0]\n",
    "        conversion = pd.read_csv(conversion_dir + conversion_file)\n",
    "        conversion['contig'], conversion['info'] = conversion['original'].str.split(' ', 1).str\n",
    "        conversion = conversion[['contig','renamed']]\n",
    "        conversion.columns = ['contig','seqid']\n",
    "        table = pd.merge(conversion, table, on=\"seqid\")\n",
    "        print(table.head())\n",
    "        print(table.shape)\n",
    "        file_out = Fhet_counts_ncbi + table_file + \"_NCBI.csv\"\n",
    "        table.to_csv(file_out)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig): (78538, 5)\n",
      "Unique Fhet genes: (22435, 5)\n",
      "                     contig           seqid  start  end  length    NCBI.id  \\\n",
      "0  TRINITY_DN56902_c1_g1_i1  Transcript_160      0  113     113  831490423   \n",
      "1  TRINITY_DN56910_c0_g1_i1  Transcript_165      1  140     139  831535516   \n",
      "2   TRINITY_DN7043_c0_g1_i1  Transcript_177      0  109     109  831510576   \n",
      "3  TRINITY_DN64187_c0_g1_i1  Transcript_240      0  225     225  831501107   \n",
      "4  TRINITY_DN64187_c0_g2_i1  Transcript_241      0  187     187  831501107   \n",
      "\n",
      "       protein_id                                              Notes  \n",
      "0  XP_012706902.1   PREDICTED: cell death activator CIDE-B [Fundu...  \n",
      "1  XP_012722638.1   PREDICTED: neuroligin-3-like isoform X2 [Fund...  \n",
      "2  XP_012713947.1   PREDICTED: UDP-glucuronosyltransferase 2A1-li...  \n",
      "3  XP_012710668.1   PREDICTED: F-box only protein 39 [Fundulus he...  \n",
      "4  XP_012710668.1   PREDICTED: F-box only protein 39 [Fundulus he...  \n"
     ]
    }
   ],
   "source": [
    "table_file = \"F_catanatus_genenames.csv\"\n",
    "species = \"F_catanatus\"\n",
    "table = pd.read_csv(Fhet_counts_dir + table_file)\n",
    "print(\"Contigs (one Fhet gene selected, best E-value < 1E-05 match per contig):\",table.groupby('seqid').count().shape)\n",
    "print(\"Unique Fhet genes:\",table.groupby('Name').count().shape)\n",
    "table['split_1'], table['split_2'], table['split_3'],table['split_4'],table['split_5'] = table['Name'].str.split('|',0).str\n",
    "table.columns = ['index','seqid', 'Name', 'start', 'end', 'length', 'split_1', 'id','split_3', 'protein_id', 'Notes']\n",
    "table = table[['seqid', 'Name', 'start', 'end', 'length', 'id','protein_id', 'Notes']]\n",
    "table = pd.merge(table,protein_transcript,how=\"inner\",on=\"protein_id\")\n",
    "table = table.drop_duplicates(subset='seqid_x')[['seqid_x', 'Name', 'start', 'end', 'length', 'id','protein_id', 'Notes']]\n",
    "table = table[['seqid_x', 'start', 'end', 'length', 'id','protein_id', 'Notes']]\n",
    "table.columns = ['seqid', 'start', 'end', 'length', 'NCBI.id','protein_id', 'Notes']\n",
    "conversion_file = [i for i in conversions if i.startswith(species)][0]\n",
    "conversion = pd.read_csv(conversion_dir + conversion_file)\n",
    "conversion['contig'], conversion['info'] = conversion['original'].str.split(' ', 1).str\n",
    "conversion = conversion[['contig','renamed']]\n",
    "conversion.columns = ['contig','seqid']\n",
    "table = pd.merge(conversion, table, on=\"seqid\")\n",
    "print(table.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
