{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "# requires dammit env\n",
    "# source activate dammit\n",
    "from dammit.fileio.gff3 import GFF3Parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['F_heteroclitusMDPP_counts.csv', 'F_parvapinis_counts.csv', 'L_parva_counts.csv', 'F_olivaceous_counts.csv', 'F_notatus_counts.csv', 'F_heteroclitusMDPL_counts.csv', 'F_grandis_counts.csv', 'F_zebrinus_counts.csv', 'A_xenica_counts.csv', 'F_catanatus_counts.csv', 'F_sciadicus_counts.csv', 'F_rathbuni_counts.csv', 'F_chrysotus_counts.csv', 'F_diaphanus_counts.csv', 'L_goodei_counts.csv', 'F_nottii_counts.csv', 'F_similis_counts.csv']\n"
     ]
    }
   ],
   "source": [
    "# salmon counts per sample\n",
    "# merged with tximport\n",
    "# at Trinity \"gene\" level\n",
    "counts_matrices = \"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_July2019/\"\n",
    "counts_files = os.listdir(counts_matrices)\n",
    "# contig-gene conversion names\n",
    "gene_out_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_name_July2019_filtnew/\"\n",
    "print(counts_files)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test one species, F. heteroclitus MDPP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(369971, 10)\n"
     ]
    }
   ],
   "source": [
    "gene_out = gene_out_dir + \"F_heteroclitusMDPP\" + \"_gene_counts_annotations_filt.csv\"\n",
    "# new table salmon v0.14.0\n",
    "table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_July2019/\"+\"F_heteroclitusMDPP_counts.csv\")\n",
    "# old table salmon v0.12.0\n",
    "#table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_Sept2018/\"+\"F_heteroclitusMDPP_counts.csv\")\n",
    "print('Number of Trinity \"genes\" (this is how we summarized expression):')\n",
    "print(table.shape)\n",
    "table = table.rename(columns={'Unnamed: 0': 'Gene'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN0_c10_g1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0   TRINITY_DN0_c0_g1                       2118.367   \n",
       "1   TRINITY_DN0_c0_g2                       4033.616   \n",
       "2   TRINITY_DN0_c0_g3                        519.020   \n",
       "3   TRINITY_DN0_c1_g1                        921.000   \n",
       "4  TRINITY_DN0_c10_g1                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "4                          0.000                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "4                          0.000                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "4                          0.000                                0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "4                                0.000                                0.000  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Contigs filtered:\n",
      "(147578, 10)\n"
     ]
    }
   ],
   "source": [
    "# if countsvalue is >5 in any column, then keep\n",
    "table_filt = table[(table.iloc[:,1:] > 5).any(1)]\n",
    "print('Contigs filtered:')\n",
    "print(table_filt.shape)\n",
    "# gets rid of a lot of genes with no expression - bad assembly?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# annotation names from gff file per dammit contig\n",
    "name = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.fixed.gff3\"\n",
    "# conversion between contig and Trinity gene\n",
    "conversion_contig = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_23June2018/\"+\"F_heteroclitusMDPP\"+\"_contig_gene.csv\"\n",
    "# conversion between Trinity contig and dammit contig\n",
    "conversion_dammit = \"/Users/johnsolk/Documents/UCDavis/Whitehead/kfish_namemaps/kfish_namemaps/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.namemap.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "conversions_dammit = pd.read_csv(conversion_dammit)\n",
    "conversions_contig = pd.read_csv(conversion_contig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN300070_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300070_c0_g1</td>\n",
       "      <td>Transcript_543249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN300063_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300063_c0_g1</td>\n",
       "      <td>Transcript_543274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN300041_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300041_c0_g1</td>\n",
       "      <td>Transcript_543217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN300082_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300082_c0_g1</td>\n",
       "      <td>Transcript_543250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN300017_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300017_c0_g1</td>\n",
       "      <td>Transcript_543304</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene              seqid\n",
       "0  TRINITY_DN300070_c0_g1_i1  TRINITY_DN300070_c0_g1  Transcript_543249\n",
       "1  TRINITY_DN300063_c0_g1_i1  TRINITY_DN300063_c0_g1  Transcript_543274\n",
       "2  TRINITY_DN300041_c0_g1_i1  TRINITY_DN300041_c0_g1  Transcript_543217\n",
       "3  TRINITY_DN300082_c0_g1_i1  TRINITY_DN300082_c0_g1  Transcript_543250\n",
       "4  TRINITY_DN300017_c0_g1_i1  TRINITY_DN300017_c0_g1  Transcript_543304"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_dammit['Name'], conversions_dammit['info'] = conversions_dammit['original'].str.split(' ', 1).str\n",
    "conversions_dammit = conversions_dammit[['Name','renamed']]\n",
    "conversions_dammit.columns = ['Name','seqid']\n",
    "coversions_contig = conversions_contig[['Name','Gene']]\n",
    "conversions_contig = pd.merge(conversions_contig,conversions_dammit,on=\"Name\",how='right')\n",
    "conversions_contig = conversions_contig.drop('Unnamed: 0', 1)\n",
    "conversions_contig.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(552205, 3)"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_contig.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge filtered counts with Trinity contig and gene ID\n",
    "merged_table = pd.merge(conversions_contig,table_filt,on=\"Gene\",how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147876, 12)"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN300070_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300070_c0_g1</td>\n",
       "      <td>Transcript_543249</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN300063_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300063_c0_g1</td>\n",
       "      <td>Transcript_543274</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN300085_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300085_c0_g1</td>\n",
       "      <td>Transcript_543285</td>\n",
       "      <td>14.87</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN300054_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300054_c0_g1</td>\n",
       "      <td>Transcript_543295</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN300093_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300093_c0_g1</td>\n",
       "      <td>Transcript_543235</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene              seqid  \\\n",
       "0  TRINITY_DN300070_c0_g1_i1  TRINITY_DN300070_c0_g1  Transcript_543249   \n",
       "1  TRINITY_DN300063_c0_g1_i1  TRINITY_DN300063_c0_g1  Transcript_543274   \n",
       "2  TRINITY_DN300085_c0_g1_i1  TRINITY_DN300085_c0_g1  Transcript_543285   \n",
       "3  TRINITY_DN300054_c0_g1_i1  TRINITY_DN300054_c0_g1  Transcript_543295   \n",
       "4  TRINITY_DN300093_c0_g1_i1  TRINITY_DN300093_c0_g1  Transcript_543235   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "0                           0.00                            0.0   \n",
       "1                           0.00                            0.0   \n",
       "2                          14.87                            9.0   \n",
       "3                           2.00                            3.0   \n",
       "4                           0.00                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "0                            0.0                           11.0   \n",
       "1                            2.0                            8.0   \n",
       "2                           11.0                            0.0   \n",
       "3                            6.0                            3.0   \n",
       "4                            0.0                            6.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            0.0                            2.0   \n",
       "4                            0.0                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_1.quant  F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "0                                  3.0                                  0.0   \n",
       "1                                 11.0                                  2.0   \n",
       "2                                  5.0                                  0.0   \n",
       "3                                  1.0                                  2.0   \n",
       "4                                  2.0                                  0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                                  0.0  \n",
       "1                                  1.0  \n",
       "2                                  0.0  \n",
       "3                                  2.0  \n",
       "4                                  2.0  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/johnsolk/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/dammit-1.0rc2-py3.6.egg/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n",
      "/Users/johnsolk/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/dammit-1.0rc2-py3.6.egg/dammit/fileio/base.py:79: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  return pd.concat(self, ignore_index=True)\n"
     ]
    }
   ],
   "source": [
    "annotations = GFF3Parser(filename=name).read()\n",
    "annotations = annotations.dropna(subset=['Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>database</th>\n",
       "      <th>end</th>\n",
       "      <th>env_coords</th>\n",
       "      <th>phase</th>\n",
       "      <th>score</th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>trunc</th>\n",
       "      <th>type</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"Rfam:RF00381\"</td>\n",
       "      <td>homology:497023</td>\n",
       "      <td>Antizyme_FSE</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Antizyme_FSE 1 59 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.700000e-19</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>Infernal</td>\n",
       "      <td>336</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "      <td>RNA_sequence_secondary_structure</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:503492</td>\n",
       "      <td>ENSONIP00000006020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSONIP00000006020 0 192 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrthoDB</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.400000e-158</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:683059</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE 0 186 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sprot</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.600000e-128</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659499</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 59 179 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.800000e-84</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>107</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1444722</td>\n",
       "      <td>Funhe2EKm033591t1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Funhe2EKm033591t1 26 148 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kfish2rae5g.pub.aa</td>\n",
       "      <td>152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.100000e-77</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Dbxref                ID                  Name Note Parent  \\\n",
       "0  \"Rfam:RF00381\"   homology:497023          Antizyme_FSE    -    NaN   \n",
       "1             NaN   homology:503492    ENSONIP00000006020  NaN    NaN   \n",
       "2             NaN   homology:683059  sp|Q9YI98|OAZ1_DANRE  NaN    NaN   \n",
       "3             NaN  homology:1659499    ENSFHEP00000001247  NaN    NaN   \n",
       "4             NaN  homology:1444722     Funhe2EKm033591t1  NaN    NaN   \n",
       "\n",
       "                         Target accuracy bitscore  \\\n",
       "0           Antizyme_FSE 1 59 +      NaN     80.8   \n",
       "1    ENSONIP00000006020 0 192 +      NaN      NaN   \n",
       "2  sp|Q9YI98|OAZ1_DANRE 0 186 +      NaN      NaN   \n",
       "3   ENSFHEP00000001247 59 179 +      NaN      NaN   \n",
       "4    Funhe2EKm033591t1 26 148 +      NaN      NaN   \n",
       "\n",
       "                                            database  end env_coords  phase  \\\n",
       "0                                                NaN  395        NaN    NaN   \n",
       "1                                            OrthoDB  683        NaN    NaN   \n",
       "2                                              sprot  683        NaN    NaN   \n",
       "3  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  227        NaN    NaN   \n",
       "4                                 kfish2rae5g.pub.aa  152        NaN    NaN   \n",
       "\n",
       "           score         seqid        source  start strand trunc  \\\n",
       "0   2.700000e-19  Transcript_0      Infernal    336      -    no   \n",
       "1  1.400000e-158  Transcript_0          LAST    109      -   NaN   \n",
       "2  3.600000e-128  Transcript_0          LAST    109      -   NaN   \n",
       "3   4.800000e-84  Transcript_0  shmlast.LAST    107      +   NaN   \n",
       "4   1.100000e-77  Transcript_0  shmlast.LAST     30      +   NaN   \n",
       "\n",
       "                               type  length  \n",
       "0  RNA_sequence_secondary_structure      59  \n",
       "1       translated_nucleotide_match     574  \n",
       "2       translated_nucleotide_match     574  \n",
       "3  conditional_reciprocal_best_LAST     120  \n",
       "4  conditional_reciprocal_best_LAST     122  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keeps track of how long the annotation is\n",
    "annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSONIP00000006020</td>\n",
       "      <td>109</td>\n",
       "      <td>683</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110685</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>Funhe2EKm026362t2</td>\n",
       "      <td>709</td>\n",
       "      <td>871</td>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11229</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSXMAP00000019040</td>\n",
       "      <td>284</td>\n",
       "      <td>1373</td>\n",
       "      <td>1089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1425</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>ENSXETP00000058488</td>\n",
       "      <td>88</td>\n",
       "      <td>217</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSDARP00000070224</td>\n",
       "      <td>156</td>\n",
       "      <td>3633</td>\n",
       "      <td>3477</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   seqid                Name  start   end  length\n",
       "1           Transcript_0  ENSONIP00000006020    109   683     574\n",
       "110685     Transcript_10   Funhe2EKm026362t2    709   871     162\n",
       "11229     Transcript_100  ENSXMAP00000019040    284  1373    1089\n",
       "1425     Transcript_1000  ENSXETP00000058488     88   217     129\n",
       "82      Transcript_10000  ENSDARP00000070224    156  3633    3477"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# any annotation\n",
    "pickone = annotations.sort_values(by=['seqid','score'],ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "pickone.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(177922, 5)"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickone.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2661027</th>\n",
       "      <td>Transcript_99999</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>46</td>\n",
       "      <td>125</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2661013</th>\n",
       "      <td>Transcript_99998</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>69</td>\n",
       "      <td>423</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2661067</th>\n",
       "      <td>Transcript_9999</td>\n",
       "      <td>ENSFHEP00000021559</td>\n",
       "      <td>52</td>\n",
       "      <td>1154</td>\n",
       "      <td>1102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2660934</th>\n",
       "      <td>Transcript_99987</td>\n",
       "      <td>ENSFHEP00000001582</td>\n",
       "      <td>552</td>\n",
       "      <td>1633</td>\n",
       "      <td>1081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2660922</th>\n",
       "      <td>Transcript_99986</td>\n",
       "      <td>ENSFHEP00000001582</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    seqid                Name  start   end  length\n",
       "2661027  Transcript_99999  ENSFHEP00000010891     46   125      79\n",
       "2661013  Transcript_99998  ENSFHEP00000010891     69   423     354\n",
       "2661067   Transcript_9999  ENSFHEP00000021559     52  1154    1102\n",
       "2660934  Transcript_99987  ENSFHEP00000001582    552  1633    1081\n",
       "2660922  Transcript_99986  ENSFHEP00000001582      0    57      57"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# only Ensembl, many per contig\n",
    "fhet = annotations.loc[annotations['database'] == 'Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.pep.all.fa']\n",
    "# pick on Fhet per contig\n",
    "# sort by length\n",
    "fhet = fhet.sort_values(by=['seqid','length'],ascending=False).drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "fhet.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76704, 5)"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name_x</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN300070_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300070_c0_g1</td>\n",
       "      <td>Transcript_543249</td>\n",
       "      <td>ENSFHEP00000022444</td>\n",
       "      <td>10</td>\n",
       "      <td>149</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN300029_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300029_c0_g1</td>\n",
       "      <td>Transcript_543216</td>\n",
       "      <td>ENSFHEP00000004619</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN300074_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300074_c0_g1</td>\n",
       "      <td>Transcript_543288</td>\n",
       "      <td>ENSFHEP00000006207</td>\n",
       "      <td>7</td>\n",
       "      <td>85</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN300046_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300046_c0_g1</td>\n",
       "      <td>Transcript_543220</td>\n",
       "      <td>ENSFHEP00000026745</td>\n",
       "      <td>0</td>\n",
       "      <td>132</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN300084_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300084_c0_g1</td>\n",
       "      <td>Transcript_543301</td>\n",
       "      <td>ENSFHEP00000013759.1 pep primary_assembly:Fund...</td>\n",
       "      <td>0</td>\n",
       "      <td>82</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Name_x                    Gene              seqid  \\\n",
       "0  TRINITY_DN300070_c0_g1_i1  TRINITY_DN300070_c0_g1  Transcript_543249   \n",
       "1  TRINITY_DN300029_c0_g1_i1  TRINITY_DN300029_c0_g1  Transcript_543216   \n",
       "2  TRINITY_DN300074_c0_g1_i1  TRINITY_DN300074_c0_g1  Transcript_543288   \n",
       "3  TRINITY_DN300046_c0_g1_i1  TRINITY_DN300046_c0_g1  Transcript_543220   \n",
       "4  TRINITY_DN300084_c0_g1_i1  TRINITY_DN300084_c0_g1  Transcript_543301   \n",
       "\n",
       "                                              Name_y  start  end  length  \n",
       "0                                 ENSFHEP00000022444     10  149     139  \n",
       "1                                 ENSFHEP00000004619      0   67      67  \n",
       "2                                 ENSFHEP00000006207      7   85      78  \n",
       "3                                 ENSFHEP00000026745      0  132     132  \n",
       "4  ENSFHEP00000013759.1 pep primary_assembly:Fund...      0   82      82  "
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged = pd.merge(conversions_contig,fhet,on=\"seqid\",how='right')\n",
    "fhet_merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147876, 12)"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_merged = pd.merge(fhet_merged,merged_table,on=\"Gene\",how=\"outer\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(220149, 18)"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_merged.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.23.4'"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_merged.to_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/test.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
