{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. After tximport in R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Script: full_model.Rmd\n",
    "* Input: \"~/Documents/UCDavis/Whitehead/RNAseq_15killifish/salmon_denovo_by_species/\"\n",
    "* Output: \"~/Documents/UCDavis/Whitehead/counts_matrices/*_counts.csv\" for each species "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Merge annotations for each species, connecting Trinity contigs/genes to annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using dammit's `GFF3Parser` function \n",
    "1. Digests the gff3 file for each species (output from dammit, downloaded from the farm cluster)\n",
    "2. Sorts each contig by E-value score\n",
    "3. Assigns the lowest E-value score for each contig \n",
    "4. Separately, for each contig, saves gene names from the NCBI annotated F. heteroclitus genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "# requires dammit env\n",
    "# source activate dammit\n",
    "from dammit.fileio.gff3 import GFF3Parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['F_heteroclitusMDPP_counts.csv', 'F_parvapinis_counts.csv', 'L_parva_counts.csv', 'F_olivaceous_counts.csv', 'F_notatus_counts.csv', 'F_heteroclitusMDPL_counts.csv', 'F_grandis_counts.csv', 'F_zebrinus_counts.csv', 'A_xenica_counts.csv', 'F_catanatus_counts.csv', 'F_sciadicus_counts.csv', 'F_rathbuni_counts.csv', 'F_chrysotus_counts.csv', 'F_diaphanus_counts.csv', 'L_goodei_counts.csv', 'F_nottii_counts.csv', 'F_similis_counts.csv']\n"
     ]
    }
   ],
   "source": [
    "counts_matrices = \"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_July2019/\"\n",
    "counts_files = os.listdir(counts_matrices)\n",
    "gene_out_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_name_July2019_filtnew/\"\n",
    "print(counts_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "========\n",
      "F_heteroclitusMDPP\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(369971, 10)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(147578, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22922\n",
      "Unique Fhet gene names (one name per contig):\n",
      "2665\n",
      "========\n",
      "F_parvapinis\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(219865, 9)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(65341, 9)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20432\n",
      "Unique Fhet gene names (one name per contig):\n",
      "1564\n",
      "========\n",
      "L_parva\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(204418, 10)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(65308, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22157\n",
      "Unique Fhet gene names (one name per contig):\n",
      "1018\n",
      "========\n",
      "F_olivaceous\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(192082, 9)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(59830, 9)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21482\n",
      "Unique Fhet gene names (one name per contig):\n",
      "828\n",
      "========\n",
      "F_notatus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(211914, 10)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(73487, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22290\n",
      "Unique Fhet gene names (one name per contig):\n",
      "1030\n",
      "========\n",
      "F_heteroclitusMDPL\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(287734, 10)\n",
      "Contigs filtered for expression (each row must have 5 counts):\n",
      "(109315, 10)\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-121-e14349e7eef0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     23\u001b[0m         \u001b[0mconversion_contig\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_collapsed_July2019/\"\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mspecies\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\"_contig_gene.csv\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m         \u001b[0mconversion_dammit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mspecies\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\".trinity_out.Trinity.fasta.dammit.namemap.csv\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 25\u001b[0;31m         \u001b[0mannotations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mGFF3Parser\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     26\u001b[0m         \u001b[0mannotations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mannotations\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     27\u001b[0m         \u001b[0;31m# keeps track of how long the annotation is\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/dammit-1.0rc2-py3.6.egg/dammit/fileio/base.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     77\u001b[0m         '''\n\u001b[1;32m     78\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 79\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mignore_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     80\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mEmptyFile\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     81\u001b[0m             \u001b[0;31m# no objects, return an empty dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36mconcat\u001b[0;34m(objs, axis, join, join_axes, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[0m\n\u001b[1;32m    223\u001b[0m                        \u001b[0mkeys\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlevels\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlevels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnames\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnames\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    224\u001b[0m                        \u001b[0mverify_integrity\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverify_integrity\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 225\u001b[0;31m                        copy=copy, sort=sort)\n\u001b[0m\u001b[1;32m    226\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    227\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, objs, axis, join, join_axes, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[0m\n\u001b[1;32m    254\u001b[0m             \u001b[0mobjs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mobjs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mk\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mkeys\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    255\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 256\u001b[0;31m             \u001b[0mobjs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobjs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    257\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    258\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobjs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/dammit_dev/lib/python3.6/site-packages/dammit-1.0rc2-py3.6.egg/dammit/fileio/gff3.py\u001b[0m in \u001b[0;36m__iter__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     76\u001b[0m             \u001b[0mgroup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minplace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m             df = pd.merge(group, pd.DataFrame(list(group.attributes)),\n\u001b[0;32m---> 78\u001b[0;31m                           left_index=True, right_index=True)\n\u001b[0m\u001b[1;32m     79\u001b[0m             \u001b[0;32mdel\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'attributes'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     80\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# 08/01/2019\n",
    "# keep only F. het\n",
    "# then pick one per contig\n",
    "# if a contig does not have an F. het annotation, \n",
    "# keep track of this? but then drop it\n",
    "for counts_file in counts_files:\n",
    "    if counts_file != \".DS_Store\":\n",
    "        species = counts_file.split(\"_\")[0]+\"_\"+counts_file.split(\"_\")[1]\n",
    "        print(\"========\")\n",
    "        print(species)\n",
    "        print(\"========\")\n",
    "        gene_out = gene_out_dir + species + \"_gene_counts_annotations_filt.csv\"\n",
    "        table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_July2019/\"+counts_file)\n",
    "        print('Number of Trinity \"genes\" (this is how we summarized expression):')\n",
    "        print(table.shape)\n",
    "        table = table.rename(columns={'Unnamed: 0': 'Gene'})\n",
    "        # make sure expression counts >5\n",
    "        # if countsvalue is >5 in any column, then keep\n",
    "        table_filt = table[(table.iloc[:,1:] > 5).any(1)]\n",
    "        print('Contigs filtered for expression (each row must have 5 counts):')\n",
    "        print(table_filt.shape)\n",
    "        name = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"+species+\".trinity_out.Trinity.fasta.dammit.fixed.gff3\"\n",
    "        conversion_contig = \"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_collapsed_July2019/\"+species+\"_contig_gene.csv\"\n",
    "        conversion_dammit = \"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"+species+\".trinity_out.Trinity.fasta.dammit.namemap.csv\"\n",
    "        annotations = GFF3Parser(filename=name).read()\n",
    "        annotations = annotations.dropna(subset=['Name'])\n",
    "        # keeps track of how long the annotation is\n",
    "        annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)\n",
    "        conversions_dammit = pd.read_csv(conversion_dammit)\n",
    "        conversions_contig = pd.read_csv(conversion_contig)\n",
    "        conversions_dammit['Name'], conversions_dammit['info'] = conversions_dammit['original'].str.split(' ', 1).str\n",
    "        conversions_dammit = conversions_dammit[['Name','renamed']]\n",
    "        conversions_dammit.columns = ['Name','seqid']\n",
    "        coversions_contig = conversions_contig[['Name','Gene']]\n",
    "        # merge filtered counts with Trinity contig and gene ID\n",
    "        merged_table = pd.merge(table_filt,conversions_contig,on=\"Gene\",how='left')\n",
    "        # merge with dammit transcript ID\n",
    "        merged_table = pd.merge(merged_table,conversions_dammit,on=\"Name\",how='left')\n",
    "        # merge with annotation names\n",
    "        fhet = annotations[annotations['Name'].str.startswith(\"ENSFHEP\")]\n",
    "        # pick best e-value match\n",
    "        fhet_filtered = fhet.sort_values(by=['seqid','length'],ascending=False).drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "        print(\"Unique Fhet gene names (one name per contig):\")\n",
    "        print(len(fhet_filtered.Name.unique()))\n",
    "        fhet_merged_table = pd.merge(merged_table,fhet_filtered,on='seqid',how='inner')\n",
    "        # gets rid of contigs without any F. het annotation\n",
    "        # we probably lose ~2000 because of low counts\n",
    "        # more \"Genes\" than annotations because the contigs are probably fragmented across a gene\n",
    "        print(\"Unique Fhet gene names (one name per contig):\")\n",
    "        print(len(fhet_merged_table.Name_y.unique()))\n",
    "        \n",
    "        \n",
    "        #fhet_merged_table = fhet_merged_table.drop('Unnamed: 0', 1)\n",
    "        #fhet_merged_table = fhet_merged_table.rename(columns = {'Name_x':'TrinityContig','Name_y':'Fhet_GeneName'}) \n",
    "        #fhet_merged_table['split1'], fhet_merged_table['split2'],fhet_merged_table['split3'],fhet_merged_table['NCBIproteinID'],fhet_merged_table['NCBIproteinName'] = fhet_merged_table['Fhet_GeneName'].str.split('|', 5).str\n",
    "        #fhet_merged_table = fhet_merged_table.drop('split1',1)\n",
    "        #fhet_merged_table = fhet_merged_table.drop('split2',1)\n",
    "        #fhet_merged_table = fhet_merged_table.drop('split3',1)\n",
    "        #print(\"Unique Fhet annotations, contigs with expression\")\n",
    "        #print(len(fhet_merged_table.Fhet_GeneName.unique()))\n",
    "        #print('Unique NCBI protein ID')\n",
    "        #print(len(fhet_merged_table.NCBIproteinID.unique()))\n",
    "        #print('Unique Trinity \"genes\"')\n",
    "        #print(len(fhet_merged_table.Gene.unique()))\n",
    "        #print('Unique Trinity \"transcripts\" (contigs) ')\n",
    "        #print(len(fhet_merged_table.TrinityContig.unique()))\n",
    "        #fhet = fhet_merged_table.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "        #print('Collaposed genes')\n",
    "        #print(fhet.shape)\n",
    "        #fhet_collapse = fhet.groupby(['NCBIproteinID'])[list(fhet.columns)[1:len(list(fhet.columns))-1]].agg('sum')\n",
    "        #print('Collaposed proteins')\n",
    "        #print(fhet_collapse.shape)\n",
    "        #fhet_collapse.head()\n",
    "        #fhet_collapse.to_csv(gene_out)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test one species, F. heteroclitus MDPP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(369971, 10)\n"
     ]
    }
   ],
   "source": [
    "gene_out = gene_out_dir + \"F_heteroclitusMDPP\" + \"_gene_counts_annotations_filt.csv\"\n",
    "table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_July2019/\"+\"F_heteroclitusMDPP_counts.csv\")\n",
    "#table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices_Sept2018/\"+\"F_heteroclitusMDPP_counts.csv\")\n",
    "print('Number of Trinity \"genes\" (this is how we summarized expression):')\n",
    "print(table.shape)\n",
    "table = table.rename(columns={'Unnamed: 0': 'Gene'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN0_c10_g1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0   TRINITY_DN0_c0_g1                       2118.367   \n",
       "1   TRINITY_DN0_c0_g2                       4033.616   \n",
       "2   TRINITY_DN0_c0_g3                        519.020   \n",
       "3   TRINITY_DN0_c1_g1                        921.000   \n",
       "4  TRINITY_DN0_c10_g1                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "4                          0.000                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "4                          0.000                          0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "4                          0.000                                0.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "4                                0.000                                0.000  "
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Contigs filtered:\n",
      "(147578, 10)\n"
     ]
    }
   ],
   "source": [
    "# if countsvalue is >5 in any column, then keep\n",
    "table_filt = table[(table.iloc[:,1:] > 5).any(1)]\n",
    "print('Contigs filtered:')\n",
    "print(table_filt.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "name = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.fixed.gff3\"\n",
    "conversion_contig = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_23June2018/\"+\"F_heteroclitusMDPP\"+\"_contig_gene.csv\"\n",
    "conversion_dammit = \"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.namemap.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations = GFF3Parser(filename=name).read()\n",
    "annotations = annotations.dropna(subset=['Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>database</th>\n",
       "      <th>end</th>\n",
       "      <th>env_coords</th>\n",
       "      <th>phase</th>\n",
       "      <th>score</th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>trunc</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"Rfam:RF00381\"</td>\n",
       "      <td>homology:497023</td>\n",
       "      <td>Antizyme_FSE</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Antizyme_FSE 1 59 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.700000e-19</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>Infernal</td>\n",
       "      <td>336</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "      <td>RNA_sequence_secondary_structure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:503492</td>\n",
       "      <td>ENSONIP00000006020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSONIP00000006020 0 192 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrthoDB</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.400000e-158</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:683059</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE 0 186 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sprot</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.600000e-128</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659499</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 59 179 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.800000e-84</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>107</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1444722</td>\n",
       "      <td>Funhe2EKm033591t1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Funhe2EKm033591t1 26 148 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kfish2rae5g.pub.aa</td>\n",
       "      <td>152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.100000e-77</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Dbxref                ID                  Name Note Parent  \\\n",
       "0  \"Rfam:RF00381\"   homology:497023          Antizyme_FSE    -    NaN   \n",
       "1             NaN   homology:503492    ENSONIP00000006020  NaN    NaN   \n",
       "2             NaN   homology:683059  sp|Q9YI98|OAZ1_DANRE  NaN    NaN   \n",
       "3             NaN  homology:1659499    ENSFHEP00000001247  NaN    NaN   \n",
       "4             NaN  homology:1444722     Funhe2EKm033591t1  NaN    NaN   \n",
       "\n",
       "                         Target accuracy bitscore  \\\n",
       "0           Antizyme_FSE 1 59 +      NaN     80.8   \n",
       "1    ENSONIP00000006020 0 192 +      NaN      NaN   \n",
       "2  sp|Q9YI98|OAZ1_DANRE 0 186 +      NaN      NaN   \n",
       "3   ENSFHEP00000001247 59 179 +      NaN      NaN   \n",
       "4    Funhe2EKm033591t1 26 148 +      NaN      NaN   \n",
       "\n",
       "                                            database  end env_coords  phase  \\\n",
       "0                                                NaN  395        NaN    NaN   \n",
       "1                                            OrthoDB  683        NaN    NaN   \n",
       "2                                              sprot  683        NaN    NaN   \n",
       "3  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  227        NaN    NaN   \n",
       "4                                 kfish2rae5g.pub.aa  152        NaN    NaN   \n",
       "\n",
       "           score         seqid        source  start strand trunc  \\\n",
       "0   2.700000e-19  Transcript_0      Infernal    336      -    no   \n",
       "1  1.400000e-158  Transcript_0          LAST    109      -   NaN   \n",
       "2  3.600000e-128  Transcript_0          LAST    109      -   NaN   \n",
       "3   4.800000e-84  Transcript_0  shmlast.LAST    107      +   NaN   \n",
       "4   1.100000e-77  Transcript_0  shmlast.LAST     30      +   NaN   \n",
       "\n",
       "                               type  \n",
       "0  RNA_sequence_secondary_structure  \n",
       "1       translated_nucleotide_match  \n",
       "2       translated_nucleotide_match  \n",
       "3  conditional_reciprocal_best_LAST  \n",
       "4  conditional_reciprocal_best_LAST  "
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keeps track of how long the annotation is\n",
    "annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pickonename = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','Note','database','Dbxref','start','end','length']]\n",
    "#pickonename = pickonename.dropna(axis=0,how=\"all\")\n",
    "#print('Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):')\n",
    "#print(pickonename.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "conversions_dammit = pd.read_csv(conversion_dammit)\n",
    "conversions_contig = pd.read_csv(conversion_contig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [],
   "source": [
    "conversions_dammit['Name'], conversions_dammit['info'] = conversions_dammit['original'].str.split(' ', 1).str\n",
    "conversions_dammit = conversions_dammit[['Name','renamed']]\n",
    "conversions_dammit.columns = ['Name','seqid']\n",
    "coversions_contig = conversions_contig[['Name','Gene']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN341363_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341363_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN341306_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341306_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN341347_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341347_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN341308_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341308_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN341353_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341353_c0_g1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene\n",
       "0  TRINITY_DN341363_c0_g1_i1  TRINITY_DN341363_c0_g1\n",
       "1  TRINITY_DN341306_c0_g1_i1  TRINITY_DN341306_c0_g1\n",
       "2  TRINITY_DN341347_c0_g1_i1  TRINITY_DN341347_c0_g1\n",
       "3  TRINITY_DN341308_c0_g1_i1  TRINITY_DN341308_c0_g1\n",
       "4  TRINITY_DN341353_c0_g1_i1  TRINITY_DN341353_c0_g1"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coversions_contig.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(668484, 2)"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coversions_contig.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>seqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN341363_c0_g1_i1</td>\n",
       "      <td>Transcript_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN341306_c0_g1_i1</td>\n",
       "      <td>Transcript_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN341347_c0_g1_i1</td>\n",
       "      <td>Transcript_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN341308_c0_g1_i1</td>\n",
       "      <td>Transcript_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN341353_c0_g1_i1</td>\n",
       "      <td>Transcript_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name         seqid\n",
       "0  TRINITY_DN341363_c0_g1_i1  Transcript_0\n",
       "1  TRINITY_DN341306_c0_g1_i1  Transcript_1\n",
       "2  TRINITY_DN341347_c0_g1_i1  Transcript_2\n",
       "3  TRINITY_DN341308_c0_g1_i1  Transcript_3\n",
       "4  TRINITY_DN341353_c0_g1_i1  Transcript_4"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_dammit.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(668487, 2)"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_dammit.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "conversions_contig = pd.merge(conversions_contig,conversions_dammit,on=\"Name\",how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN341363_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341363_c0_g1</td>\n",
       "      <td>Transcript_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN341306_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341306_c0_g1</td>\n",
       "      <td>Transcript_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN341347_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341347_c0_g1</td>\n",
       "      <td>Transcript_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN341308_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341308_c0_g1</td>\n",
       "      <td>Transcript_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN341353_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341353_c0_g1</td>\n",
       "      <td>Transcript_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene         seqid\n",
       "0  TRINITY_DN341363_c0_g1_i1  TRINITY_DN341363_c0_g1  Transcript_0\n",
       "1  TRINITY_DN341306_c0_g1_i1  TRINITY_DN341306_c0_g1  Transcript_1\n",
       "2  TRINITY_DN341347_c0_g1_i1  TRINITY_DN341347_c0_g1  Transcript_2\n",
       "3  TRINITY_DN341308_c0_g1_i1  TRINITY_DN341308_c0_g1  Transcript_3\n",
       "4  TRINITY_DN341353_c0_g1_i1  TRINITY_DN341353_c0_g1  Transcript_4"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_contig = conversions_contig.drop('Unnamed: 0', 1)\n",
    "conversions_contig.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(668484, 3)"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conversions_contig.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge filtered counts with Trinity contig and gene ID\n",
    "merged_table = pd.merge(conversions_contig,table_filt,on=\"Gene\",how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN300070_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300070_c0_g1</td>\n",
       "      <td>Transcript_657</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN300063_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300063_c0_g1</td>\n",
       "      <td>Transcript_658</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN300085_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300085_c0_g1</td>\n",
       "      <td>Transcript_673</td>\n",
       "      <td>14.87</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN300054_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300054_c0_g1</td>\n",
       "      <td>Transcript_682</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN300093_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300093_c0_g1</td>\n",
       "      <td>Transcript_687</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene           seqid  \\\n",
       "0  TRINITY_DN300070_c0_g1_i1  TRINITY_DN300070_c0_g1  Transcript_657   \n",
       "1  TRINITY_DN300063_c0_g1_i1  TRINITY_DN300063_c0_g1  Transcript_658   \n",
       "2  TRINITY_DN300085_c0_g1_i1  TRINITY_DN300085_c0_g1  Transcript_673   \n",
       "3  TRINITY_DN300054_c0_g1_i1  TRINITY_DN300054_c0_g1  Transcript_682   \n",
       "4  TRINITY_DN300093_c0_g1_i1  TRINITY_DN300093_c0_g1  Transcript_687   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "0                           0.00                            0.0   \n",
       "1                           0.00                            0.0   \n",
       "2                          14.87                            9.0   \n",
       "3                           2.00                            3.0   \n",
       "4                           0.00                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "0                            0.0                           11.0   \n",
       "1                            2.0                            8.0   \n",
       "2                           11.0                            0.0   \n",
       "3                            6.0                            3.0   \n",
       "4                            0.0                            6.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            0.0                            2.0   \n",
       "4                            0.0                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_1.quant  F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "0                                  3.0                                  0.0   \n",
       "1                                 11.0                                  2.0   \n",
       "2                                  5.0                                  0.0   \n",
       "3                                  1.0                                  2.0   \n",
       "4                                  2.0                                  0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                                  0.0  \n",
       "1                                  1.0  \n",
       "2                                  0.0  \n",
       "3                                  2.0  \n",
       "4                                  2.0  "
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(150470, 12)"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### merge with annotation names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2006410, 20)"
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>database</th>\n",
       "      <th>end</th>\n",
       "      <th>env_coords</th>\n",
       "      <th>phase</th>\n",
       "      <th>score</th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>trunc</th>\n",
       "      <th>type</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"Rfam:RF00381\"</td>\n",
       "      <td>homology:497023</td>\n",
       "      <td>Antizyme_FSE</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Antizyme_FSE 1 59 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.700000e-19</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>Infernal</td>\n",
       "      <td>336</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "      <td>RNA_sequence_secondary_structure</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:503492</td>\n",
       "      <td>ENSONIP00000006020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSONIP00000006020 0 192 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OrthoDB</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.400000e-158</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:683059</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q9YI98|OAZ1_DANRE 0 186 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sprot</td>\n",
       "      <td>683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.600000e-128</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>LAST</td>\n",
       "      <td>109</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>translated_nucleotide_match</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659499</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 59 179 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.800000e-84</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>107</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1444722</td>\n",
       "      <td>Funhe2EKm033591t1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Funhe2EKm033591t1 26 148 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kfish2rae5g.pub.aa</td>\n",
       "      <td>152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.100000e-77</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>30</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Dbxref                ID                  Name Note Parent  \\\n",
       "0  \"Rfam:RF00381\"   homology:497023          Antizyme_FSE    -    NaN   \n",
       "1             NaN   homology:503492    ENSONIP00000006020  NaN    NaN   \n",
       "2             NaN   homology:683059  sp|Q9YI98|OAZ1_DANRE  NaN    NaN   \n",
       "3             NaN  homology:1659499    ENSFHEP00000001247  NaN    NaN   \n",
       "4             NaN  homology:1444722     Funhe2EKm033591t1  NaN    NaN   \n",
       "\n",
       "                         Target accuracy bitscore  \\\n",
       "0           Antizyme_FSE 1 59 +      NaN     80.8   \n",
       "1    ENSONIP00000006020 0 192 +      NaN      NaN   \n",
       "2  sp|Q9YI98|OAZ1_DANRE 0 186 +      NaN      NaN   \n",
       "3   ENSFHEP00000001247 59 179 +      NaN      NaN   \n",
       "4    Funhe2EKm033591t1 26 148 +      NaN      NaN   \n",
       "\n",
       "                                            database  end env_coords  phase  \\\n",
       "0                                                NaN  395        NaN    NaN   \n",
       "1                                            OrthoDB  683        NaN    NaN   \n",
       "2                                              sprot  683        NaN    NaN   \n",
       "3  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  227        NaN    NaN   \n",
       "4                                 kfish2rae5g.pub.aa  152        NaN    NaN   \n",
       "\n",
       "           score         seqid        source  start strand trunc  \\\n",
       "0   2.700000e-19  Transcript_0      Infernal    336      -    no   \n",
       "1  1.400000e-158  Transcript_0          LAST    109      -   NaN   \n",
       "2  3.600000e-128  Transcript_0          LAST    109      -   NaN   \n",
       "3   4.800000e-84  Transcript_0  shmlast.LAST    107      +   NaN   \n",
       "4   1.100000e-77  Transcript_0  shmlast.LAST     30      +   NaN   \n",
       "\n",
       "                               type  length  \n",
       "0  RNA_sequence_secondary_structure      59  \n",
       "1       translated_nucleotide_match     574  \n",
       "2       translated_nucleotide_match     574  \n",
       "3  conditional_reciprocal_best_LAST     120  \n",
       "4  conditional_reciprocal_best_LAST     122  "
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {},
   "outputs": [],
   "source": [
    "# any annotation\n",
    "pickone = annotations.sort_values(by=['seqid','score'],ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSONIP00000006020</td>\n",
       "      <td>109</td>\n",
       "      <td>683</td>\n",
       "      <td>574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110685</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>Funhe2EKm026362t2</td>\n",
       "      <td>709</td>\n",
       "      <td>871</td>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11229</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSXMAP00000019040</td>\n",
       "      <td>284</td>\n",
       "      <td>1373</td>\n",
       "      <td>1089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1425</th>\n",
       "      <td>Transcript_1000</td>\n",
       "      <td>ENSXETP00000058488</td>\n",
       "      <td>88</td>\n",
       "      <td>217</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSDARP00000070224</td>\n",
       "      <td>156</td>\n",
       "      <td>3633</td>\n",
       "      <td>3477</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   seqid                Name  start   end  length\n",
       "1           Transcript_0  ENSONIP00000006020    109   683     574\n",
       "110685     Transcript_10   Funhe2EKm026362t2    709   871     162\n",
       "11229     Transcript_100  ENSXMAP00000019040    284  1373    1089\n",
       "1425     Transcript_1000  ENSXETP00000058488     88   217     129\n",
       "82      Transcript_10000  ENSDARP00000070224    156  3633    3477"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickone.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(177922, 5)"
      ]
     },
     "execution_count": 291,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pickone.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(152572, 20)"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# only Ensembl\n",
    "# many per contig\n",
    "fhet = annotations.loc[annotations['database'] == 'Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.pep.all.fa']\n",
    "#fhet = annotations[annotations['Name'].str.startswith(\"ENSFHEP\")]\n",
    "fhet.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>database</th>\n",
       "      <th>end</th>\n",
       "      <th>env_coords</th>\n",
       "      <th>phase</th>\n",
       "      <th>score</th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>trunc</th>\n",
       "      <th>type</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659499</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 59 179 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.800000e-84</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>107</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659500</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 0 59 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.200000e-37</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>36</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510136</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 0 354 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.600000e-253</td>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>26</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510135</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 75 354 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.200000e-196</td>\n",
       "      <td>Transcript_100002</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>125</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510134</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 0 79 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>129</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.700000e-49</td>\n",
       "      <td>Transcript_100002</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>50</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Dbxref                ID                Name Note Parent  \\\n",
       "3     NaN  homology:1659499  ENSFHEP00000001247  NaN    NaN   \n",
       "5     NaN  homology:1659500  ENSFHEP00000001247  NaN    NaN   \n",
       "18    NaN  homology:1510136  ENSFHEP00000010891  NaN    NaN   \n",
       "34    NaN  homology:1510135  ENSFHEP00000010891  NaN    NaN   \n",
       "42    NaN  homology:1510134  ENSFHEP00000010891  NaN    NaN   \n",
       "\n",
       "                         Target accuracy bitscore  \\\n",
       "3   ENSFHEP00000001247 59 179 +      NaN      NaN   \n",
       "5     ENSFHEP00000001247 0 59 +      NaN      NaN   \n",
       "18   ENSFHEP00000010891 0 354 +      NaN      NaN   \n",
       "34  ENSFHEP00000010891 75 354 +      NaN      NaN   \n",
       "42    ENSFHEP00000010891 0 79 +      NaN      NaN   \n",
       "\n",
       "                                             database  end env_coords  phase  \\\n",
       "3   Fundulus_heteroclitus.Fundulus_heteroclitus-3....  227        NaN    NaN   \n",
       "5   Fundulus_heteroclitus.Fundulus_heteroclitus-3....   95        NaN    NaN   \n",
       "18  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  380        NaN    NaN   \n",
       "34  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  404        NaN    NaN   \n",
       "42  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  129        NaN    NaN   \n",
       "\n",
       "            score              seqid        source  start strand trunc  \\\n",
       "3    4.800000e-84       Transcript_0  shmlast.LAST    107      +   NaN   \n",
       "5    8.200000e-37       Transcript_0  shmlast.LAST     36      +   NaN   \n",
       "18  4.600000e-253  Transcript_100001  shmlast.LAST     26      +   NaN   \n",
       "34  4.200000e-196  Transcript_100002  shmlast.LAST    125      +   NaN   \n",
       "42   7.700000e-49  Transcript_100002  shmlast.LAST     50      +   NaN   \n",
       "\n",
       "                                type  length  \n",
       "3   conditional_reciprocal_best_LAST     120  \n",
       "5   conditional_reciprocal_best_LAST      59  \n",
       "18  conditional_reciprocal_best_LAST     354  \n",
       "34  conditional_reciprocal_best_LAST     279  \n",
       "42  conditional_reciprocal_best_LAST      79  "
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# still multiple Ensembl per seqid contig\n",
    "fhet.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pick on Fhet per contig\n",
    "# sort by length\n",
    "ens = fhet.sort_values(by=['seqid','length'],ascending=True).drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76704, 5)"
      ]
     },
     "execution_count": 303,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now only one annotation/seqid\n",
    "ens.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>36</td>\n",
       "      <td>95</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110687</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11231</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    seqid                Name  start   end  length\n",
       "5            Transcript_0  ENSFHEP00000001247     36    95      59\n",
       "110687      Transcript_10  ENSFHEP00000022696    740   879     139\n",
       "11231      Transcript_100  ENSFHEP00000028412     94   457     363\n",
       "100      Transcript_10000  ENSFHEP00000000648     52  1227    1175\n",
       "18      Transcript_100001  ENSFHEP00000010891     26   380     354"
      ]
     },
     "execution_count": 302,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet Ensembl gene names (one name per contig):\n",
      "23074\n"
     ]
    }
   ],
   "source": [
    "fhet_filtered = fhet.sort_values(by=['seqid','score'],ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "print(\"Unique Fhet Ensembl gene names (one name per contig):\")\n",
    "print(len(fhet_filtered.Name.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>107</td>\n",
       "      <td>227</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110687</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11231</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    seqid                Name  start   end  length\n",
       "3            Transcript_0  ENSFHEP00000001247    107   227     120\n",
       "110687      Transcript_10  ENSFHEP00000022696    740   879     139\n",
       "11231      Transcript_100  ENSFHEP00000028412     94   457     363\n",
       "100      Transcript_10000  ENSFHEP00000000648     52  1227    1175\n",
       "18      Transcript_100001  ENSFHEP00000010891     26   380     354"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76702, 5)"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_filtered.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>seqid</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN300070_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300070_c0_g1</td>\n",
       "      <td>Transcript_657</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN300063_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300063_c0_g1</td>\n",
       "      <td>Transcript_658</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN300085_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300085_c0_g1</td>\n",
       "      <td>Transcript_673</td>\n",
       "      <td>14.87</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN300054_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300054_c0_g1</td>\n",
       "      <td>Transcript_682</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN300093_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN300093_c0_g1</td>\n",
       "      <td>Transcript_687</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Name                    Gene           seqid  \\\n",
       "0  TRINITY_DN300070_c0_g1_i1  TRINITY_DN300070_c0_g1  Transcript_657   \n",
       "1  TRINITY_DN300063_c0_g1_i1  TRINITY_DN300063_c0_g1  Transcript_658   \n",
       "2  TRINITY_DN300085_c0_g1_i1  TRINITY_DN300085_c0_g1  Transcript_673   \n",
       "3  TRINITY_DN300054_c0_g1_i1  TRINITY_DN300054_c0_g1  Transcript_682   \n",
       "4  TRINITY_DN300093_c0_g1_i1  TRINITY_DN300093_c0_g1  Transcript_687   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "0                           0.00                            0.0   \n",
       "1                           0.00                            0.0   \n",
       "2                          14.87                            9.0   \n",
       "3                           2.00                            3.0   \n",
       "4                           0.00                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "0                            0.0                           11.0   \n",
       "1                            2.0                            8.0   \n",
       "2                           11.0                            0.0   \n",
       "3                            6.0                            3.0   \n",
       "4                            0.0                            6.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            0.0                            2.0   \n",
       "4                            0.0                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_1.quant  F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "0                                  3.0                                  0.0   \n",
       "1                                 11.0                                  2.0   \n",
       "2                                  5.0                                  0.0   \n",
       "3                                  1.0                                  2.0   \n",
       "4                                  2.0                                  0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                                  0.0  \n",
       "1                                  1.0  \n",
       "2                                  0.0  \n",
       "3                                  2.0  \n",
       "4                                  2.0  "
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(150470, 12)"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>36</td>\n",
       "      <td>95</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110687</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11231</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    seqid                Name  start   end  length\n",
       "5            Transcript_0  ENSFHEP00000001247     36    95      59\n",
       "110687      Transcript_10  ENSFHEP00000022696    740   879     139\n",
       "11231      Transcript_100  ENSFHEP00000028412     94   457     363\n",
       "100      Transcript_10000  ENSFHEP00000000648     52  1227    1175\n",
       "18      Transcript_100001  ENSFHEP00000010891     26   380     354"
      ]
     },
     "execution_count": 329,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 304,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 305,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>36</td>\n",
       "      <td>95</td>\n",
       "      <td>59</td>\n",
       "      <td>TRINITY_DN341363_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341363_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "      <td>TRINITY_DN341343_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341343_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "      <td>TRINITY_DN311101_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN311101_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "      <td>TRINITY_DN308331_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN308331_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "      <td>TRINITY_DN115512_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115512_c0_g1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               seqid              Name_x  start   end  length  \\\n",
       "0       Transcript_0  ENSFHEP00000001247     36    95      59   \n",
       "1      Transcript_10  ENSFHEP00000022696    740   879     139   \n",
       "2     Transcript_100  ENSFHEP00000028412     94   457     363   \n",
       "3   Transcript_10000  ENSFHEP00000000648     52  1227    1175   \n",
       "4  Transcript_100001  ENSFHEP00000010891     26   380     354   \n",
       "\n",
       "                      Name_y                    Gene  \n",
       "0  TRINITY_DN341363_c0_g1_i1  TRINITY_DN341363_c0_g1  \n",
       "1  TRINITY_DN341343_c0_g1_i1  TRINITY_DN341343_c0_g1  \n",
       "2  TRINITY_DN311101_c0_g1_i1  TRINITY_DN311101_c0_g1  \n",
       "3  TRINITY_DN308331_c0_g1_i1  TRINITY_DN308331_c0_g1  \n",
       "4  TRINITY_DN115512_c0_g1_i1  TRINITY_DN115512_c0_g1  "
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table = pd.merge(ens,conversions_contig,on='seqid',how='left')\n",
    "fhet_merged_table.head()\n",
    "# gets rid of contigs without any F. het annotation Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76704, 7)"
      ]
     },
     "execution_count": 307,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet gene names (one name per contig):\n",
      "23882\n"
     ]
    }
   ],
   "source": [
    "# Ensembl\n",
    "print(\"Unique Fhet gene names (one name per contig):\")\n",
    "print(len(fhet_merged_table.Name_x.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "      <td>TRINITY_DN115512_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115512_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100019</td>\n",
       "      <td>ENSFHEP00000022304.1 pep primary_assembly:Fund...</td>\n",
       "      <td>264</td>\n",
       "      <td>628</td>\n",
       "      <td>364</td>\n",
       "      <td>TRINITY_DN115592_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115592_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_100021</td>\n",
       "      <td>ENSFHEP00000022304.1 pep primary_assembly:Fund...</td>\n",
       "      <td>241</td>\n",
       "      <td>425</td>\n",
       "      <td>184</td>\n",
       "      <td>TRINITY_DN115535_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115535_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_100047</td>\n",
       "      <td>ENSFHEP00000000967</td>\n",
       "      <td>62</td>\n",
       "      <td>374</td>\n",
       "      <td>312</td>\n",
       "      <td>TRINITY_DN106787_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN106787_c0_g1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_100062</td>\n",
       "      <td>ENSFHEP00000024506.1 pep primary_assembly:Fund...</td>\n",
       "      <td>117</td>\n",
       "      <td>285</td>\n",
       "      <td>168</td>\n",
       "      <td>TRINITY_DN106759_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN106759_c0_g1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               seqid                                             Name_x  \\\n",
       "0  Transcript_100001                                 ENSFHEP00000010891   \n",
       "1  Transcript_100019  ENSFHEP00000022304.1 pep primary_assembly:Fund...   \n",
       "2  Transcript_100021  ENSFHEP00000022304.1 pep primary_assembly:Fund...   \n",
       "3  Transcript_100047                                 ENSFHEP00000000967   \n",
       "4  Transcript_100062  ENSFHEP00000024506.1 pep primary_assembly:Fund...   \n",
       "\n",
       "   start  end  length                     Name_y                    Gene  \\\n",
       "0     26  380     354  TRINITY_DN115512_c0_g1_i1  TRINITY_DN115512_c0_g1   \n",
       "1    264  628     364  TRINITY_DN115592_c0_g1_i1  TRINITY_DN115592_c0_g1   \n",
       "2    241  425     184  TRINITY_DN115535_c0_g1_i1  TRINITY_DN115535_c0_g1   \n",
       "3     62  374     312  TRINITY_DN106787_c0_g1_i1  TRINITY_DN106787_c0_g1   \n",
       "4    117  285     168  TRINITY_DN106759_c0_g1_i1  TRINITY_DN106759_c0_g1   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            6.0                           10.0   \n",
       "4                           26.0                           34.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "0                            1.0                            8.0   \n",
       "1                            0.0                           11.0   \n",
       "2                            0.0                           11.0   \n",
       "3                            6.0                            3.0   \n",
       "4                            9.0                           31.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            1.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            5.0                            9.0   \n",
       "4                            7.0                           23.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_1.quant  F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "0                                  2.0                                  0.0   \n",
       "1                                  0.0                                  0.0   \n",
       "2                                  2.0                                  0.0   \n",
       "3                                  2.0                                  9.0   \n",
       "4                                 13.0                                 27.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                                  0.0  \n",
       "1                                  0.0  \n",
       "2                                  0.0  \n",
       "3                                 14.0  \n",
       "4                                 19.0  "
      ]
     },
     "execution_count": 316,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filtered based on low expression\n",
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [],
   "source": [
    "fhet_merged_table = pd.merge(fhet_merged_table,table_filt,on='Gene',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3662, 16)"
      ]
     },
     "execution_count": 313,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>36</td>\n",
       "      <td>95</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110687</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11231</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    seqid                Name  start   end  length\n",
       "5            Transcript_0  ENSFHEP00000001247     36    95      59\n",
       "110687      Transcript_10  ENSFHEP00000022696    740   879     139\n",
       "11231      Transcript_100  ENSFHEP00000028412     94   457     363\n",
       "100      Transcript_10000  ENSFHEP00000000648     52  1227    1175\n",
       "18      Transcript_100001  ENSFHEP00000010891     26   380     354"
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Target</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>database</th>\n",
       "      <th>end</th>\n",
       "      <th>env_coords</th>\n",
       "      <th>phase</th>\n",
       "      <th>score</th>\n",
       "      <th>seqid</th>\n",
       "      <th>source</th>\n",
       "      <th>start</th>\n",
       "      <th>strand</th>\n",
       "      <th>trunc</th>\n",
       "      <th>type</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659499</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 59 179 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.800000e-84</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>107</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1659500</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000001247 0 59 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.200000e-37</td>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>36</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510136</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 0 354 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.600000e-253</td>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>26</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510135</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 75 354 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.200000e-196</td>\n",
       "      <td>Transcript_100002</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>125</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NaN</td>\n",
       "      <td>homology:1510134</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSFHEP00000010891 0 79 +</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fundulus_heteroclitus.Fundulus_heteroclitus-3....</td>\n",
       "      <td>129</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.700000e-49</td>\n",
       "      <td>Transcript_100002</td>\n",
       "      <td>shmlast.LAST</td>\n",
       "      <td>50</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>conditional_reciprocal_best_LAST</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Dbxref                ID                Name Note Parent  \\\n",
       "3     NaN  homology:1659499  ENSFHEP00000001247  NaN    NaN   \n",
       "5     NaN  homology:1659500  ENSFHEP00000001247  NaN    NaN   \n",
       "18    NaN  homology:1510136  ENSFHEP00000010891  NaN    NaN   \n",
       "34    NaN  homology:1510135  ENSFHEP00000010891  NaN    NaN   \n",
       "42    NaN  homology:1510134  ENSFHEP00000010891  NaN    NaN   \n",
       "\n",
       "                         Target accuracy bitscore  \\\n",
       "3   ENSFHEP00000001247 59 179 +      NaN      NaN   \n",
       "5     ENSFHEP00000001247 0 59 +      NaN      NaN   \n",
       "18   ENSFHEP00000010891 0 354 +      NaN      NaN   \n",
       "34  ENSFHEP00000010891 75 354 +      NaN      NaN   \n",
       "42    ENSFHEP00000010891 0 79 +      NaN      NaN   \n",
       "\n",
       "                                             database  end env_coords  phase  \\\n",
       "3   Fundulus_heteroclitus.Fundulus_heteroclitus-3....  227        NaN    NaN   \n",
       "5   Fundulus_heteroclitus.Fundulus_heteroclitus-3....   95        NaN    NaN   \n",
       "18  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  380        NaN    NaN   \n",
       "34  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  404        NaN    NaN   \n",
       "42  Fundulus_heteroclitus.Fundulus_heteroclitus-3....  129        NaN    NaN   \n",
       "\n",
       "            score              seqid        source  start strand trunc  \\\n",
       "3    4.800000e-84       Transcript_0  shmlast.LAST    107      +   NaN   \n",
       "5    8.200000e-37       Transcript_0  shmlast.LAST     36      +   NaN   \n",
       "18  4.600000e-253  Transcript_100001  shmlast.LAST     26      +   NaN   \n",
       "34  4.200000e-196  Transcript_100002  shmlast.LAST    125      +   NaN   \n",
       "42   7.700000e-49  Transcript_100002  shmlast.LAST     50      +   NaN   \n",
       "\n",
       "                                type  length  \n",
       "3   conditional_reciprocal_best_LAST     120  \n",
       "5   conditional_reciprocal_best_LAST      59  \n",
       "18  conditional_reciprocal_best_LAST     354  \n",
       "34  conditional_reciprocal_best_LAST     279  \n",
       "42  conditional_reciprocal_best_LAST      79  "
      ]
     },
     "execution_count": 337,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76704, 5)"
      ]
     },
     "execution_count": 335,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [],
   "source": [
    "ens_contigs = pd.merge(ens,conversions_contig,on='seqid',how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_0</td>\n",
       "      <td>ENSFHEP00000001247</td>\n",
       "      <td>36</td>\n",
       "      <td>95</td>\n",
       "      <td>59</td>\n",
       "      <td>TRINITY_DN341363_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341363_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_10</td>\n",
       "      <td>ENSFHEP00000022696</td>\n",
       "      <td>740</td>\n",
       "      <td>879</td>\n",
       "      <td>139</td>\n",
       "      <td>TRINITY_DN341343_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN341343_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_100</td>\n",
       "      <td>ENSFHEP00000028412</td>\n",
       "      <td>94</td>\n",
       "      <td>457</td>\n",
       "      <td>363</td>\n",
       "      <td>TRINITY_DN311101_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN311101_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_10000</td>\n",
       "      <td>ENSFHEP00000000648</td>\n",
       "      <td>52</td>\n",
       "      <td>1227</td>\n",
       "      <td>1175</td>\n",
       "      <td>TRINITY_DN308331_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN308331_c0_g1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "      <td>TRINITY_DN115512_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115512_c0_g1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               seqid              Name_x  start   end  length  \\\n",
       "0       Transcript_0  ENSFHEP00000001247     36    95      59   \n",
       "1      Transcript_10  ENSFHEP00000022696    740   879     139   \n",
       "2     Transcript_100  ENSFHEP00000028412     94   457     363   \n",
       "3   Transcript_10000  ENSFHEP00000000648     52  1227    1175   \n",
       "4  Transcript_100001  ENSFHEP00000010891     26   380     354   \n",
       "\n",
       "                      Name_y                    Gene  \n",
       "0  TRINITY_DN341363_c0_g1_i1  TRINITY_DN341363_c0_g1  \n",
       "1  TRINITY_DN341343_c0_g1_i1  TRINITY_DN341343_c0_g1  \n",
       "2  TRINITY_DN311101_c0_g1_i1  TRINITY_DN311101_c0_g1  \n",
       "3  TRINITY_DN308331_c0_g1_i1  TRINITY_DN308331_c0_g1  \n",
       "4  TRINITY_DN115512_c0_g1_i1  TRINITY_DN115512_c0_g1  "
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens_contigs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(76704, 7)"
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens_contigs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(147578, 10)"
      ]
     },
     "execution_count": 341,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>2118.367</td>\n",
       "      <td>3823.061</td>\n",
       "      <td>2043.860</td>\n",
       "      <td>1875.973</td>\n",
       "      <td>1741.001</td>\n",
       "      <td>3107.106</td>\n",
       "      <td>1181.237</td>\n",
       "      <td>1615.228</td>\n",
       "      <td>1885.474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN0_c0_g2</td>\n",
       "      <td>4033.616</td>\n",
       "      <td>4634.489</td>\n",
       "      <td>3109.738</td>\n",
       "      <td>3562.385</td>\n",
       "      <td>1954.666</td>\n",
       "      <td>4909.949</td>\n",
       "      <td>1501.673</td>\n",
       "      <td>2716.748</td>\n",
       "      <td>3228.895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN0_c0_g3</td>\n",
       "      <td>519.020</td>\n",
       "      <td>264.704</td>\n",
       "      <td>168.547</td>\n",
       "      <td>425.641</td>\n",
       "      <td>233.355</td>\n",
       "      <td>149.159</td>\n",
       "      <td>120.091</td>\n",
       "      <td>557.423</td>\n",
       "      <td>377.390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN0_c1_g1</td>\n",
       "      <td>921.000</td>\n",
       "      <td>1300.001</td>\n",
       "      <td>853.000</td>\n",
       "      <td>1026.000</td>\n",
       "      <td>1329.000</td>\n",
       "      <td>1190.011</td>\n",
       "      <td>391.000</td>\n",
       "      <td>1101.000</td>\n",
       "      <td>936.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TRINITY_DN0_c3_g1</td>\n",
       "      <td>150.000</td>\n",
       "      <td>134.000</td>\n",
       "      <td>87.000</td>\n",
       "      <td>186.000</td>\n",
       "      <td>94.000</td>\n",
       "      <td>141.000</td>\n",
       "      <td>108.000</td>\n",
       "      <td>102.000</td>\n",
       "      <td>126.975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN0_c0_g1                       2118.367   \n",
       "1  TRINITY_DN0_c0_g2                       4033.616   \n",
       "2  TRINITY_DN0_c0_g3                        519.020   \n",
       "3  TRINITY_DN0_c1_g1                        921.000   \n",
       "5  TRINITY_DN0_c3_g1                        150.000   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                       3823.061                       2043.860   \n",
       "1                       4634.489                       3109.738   \n",
       "2                        264.704                        168.547   \n",
       "3                       1300.001                        853.000   \n",
       "5                        134.000                         87.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                       1875.973                       1741.001   \n",
       "1                       3562.385                       1954.666   \n",
       "2                        425.641                        233.355   \n",
       "3                       1026.000                       1329.000   \n",
       "5                        186.000                         94.000   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                       3107.106                             1181.237   \n",
       "1                       4909.949                             1501.673   \n",
       "2                        149.159                              120.091   \n",
       "3                       1190.011                              391.000   \n",
       "5                        141.000                              108.000   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                             1615.228                             1885.474  \n",
       "1                             2716.748                             3228.895  \n",
       "2                              557.423                              377.390  \n",
       "3                             1101.000                              936.000  \n",
       "5                              102.000                              126.975  "
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [],
   "source": [
    "ens_merge = pd.merge(ens_contigs,table_filt,on='Gene',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3662, 16)"
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript_100001</td>\n",
       "      <td>ENSFHEP00000010891</td>\n",
       "      <td>26</td>\n",
       "      <td>380</td>\n",
       "      <td>354</td>\n",
       "      <td>TRINITY_DN115512_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115512_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript_100019</td>\n",
       "      <td>ENSFHEP00000022304.1 pep primary_assembly:Fund...</td>\n",
       "      <td>264</td>\n",
       "      <td>628</td>\n",
       "      <td>364</td>\n",
       "      <td>TRINITY_DN115592_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115592_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript_100021</td>\n",
       "      <td>ENSFHEP00000022304.1 pep primary_assembly:Fund...</td>\n",
       "      <td>241</td>\n",
       "      <td>425</td>\n",
       "      <td>184</td>\n",
       "      <td>TRINITY_DN115535_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN115535_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript_100047</td>\n",
       "      <td>ENSFHEP00000000967</td>\n",
       "      <td>62</td>\n",
       "      <td>374</td>\n",
       "      <td>312</td>\n",
       "      <td>TRINITY_DN106787_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN106787_c0_g1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript_100062</td>\n",
       "      <td>ENSFHEP00000024506.1 pep primary_assembly:Fund...</td>\n",
       "      <td>117</td>\n",
       "      <td>285</td>\n",
       "      <td>168</td>\n",
       "      <td>TRINITY_DN106759_c0_g1_i1</td>\n",
       "      <td>TRINITY_DN106759_c0_g1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               seqid                                             Name_x  \\\n",
       "0  Transcript_100001                                 ENSFHEP00000010891   \n",
       "1  Transcript_100019  ENSFHEP00000022304.1 pep primary_assembly:Fund...   \n",
       "2  Transcript_100021  ENSFHEP00000022304.1 pep primary_assembly:Fund...   \n",
       "3  Transcript_100047                                 ENSFHEP00000000967   \n",
       "4  Transcript_100062  ENSFHEP00000024506.1 pep primary_assembly:Fund...   \n",
       "\n",
       "   start  end  length                     Name_y                    Gene  \\\n",
       "0     26  380     354  TRINITY_DN115512_c0_g1_i1  TRINITY_DN115512_c0_g1   \n",
       "1    264  628     364  TRINITY_DN115592_c0_g1_i1  TRINITY_DN115592_c0_g1   \n",
       "2    241  425     184  TRINITY_DN115535_c0_g1_i1  TRINITY_DN115535_c0_g1   \n",
       "3     62  374     312  TRINITY_DN106787_c0_g1_i1  TRINITY_DN106787_c0_g1   \n",
       "4    117  285     168  TRINITY_DN106759_c0_g1_i1  TRINITY_DN106759_c0_g1   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            6.0                           10.0   \n",
       "4                           26.0                           34.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "0                            1.0                            8.0   \n",
       "1                            0.0                           11.0   \n",
       "2                            0.0                           11.0   \n",
       "3                            6.0                            3.0   \n",
       "4                            9.0                           31.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            1.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            5.0                            9.0   \n",
       "4                            7.0                           23.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_1.quant  F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "0                                  2.0                                  0.0   \n",
       "1                                  0.0                                  0.0   \n",
       "2                                  2.0                                  0.0   \n",
       "3                                  2.0                                  9.0   \n",
       "4                                 13.0                                 27.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_3.quant  \n",
       "0                                  0.0  \n",
       "1                                  0.0  \n",
       "2                                  0.0  \n",
       "3                                 14.0  \n",
       "4                                 19.0  "
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3662, 16)"
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ens_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2924\n"
     ]
    }
   ],
   "source": [
    "print(len(ens_merge.Name_x.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fhet_merged_table = fhet_merged_table.rename(columns = {'Name_x':'TrinityContig','Name_y':'Fhet_GeneName'})                                           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN100767_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TRINITY_DN100767_c0_g1_i1</td>\n",
       "      <td>Transcript_221533</td>\n",
       "      <td>gi|831525541|ref|XP_012719085.1| PREDICTED: zo...</td>\n",
       "      <td>9</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10249_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN10249_c0_g1_i1</td>\n",
       "      <td>Transcript_618557</td>\n",
       "      <td>gi|831577816|ref|XP_012737259.1| PREDICTED: ez...</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN102633_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN102633_c0_g1_i1</td>\n",
       "      <td>Transcript_201922</td>\n",
       "      <td>gi|831498824|ref|XP_012709872.1| PREDICTED: co...</td>\n",
       "      <td>30</td>\n",
       "      <td>68</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN102877_c0_g1</td>\n",
       "      <td>125.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>TRINITY_DN102877_c0_g1_i1</td>\n",
       "      <td>Transcript_143501</td>\n",
       "      <td>gi|831491323|ref|XP_012707217.1| PREDICTED: LY...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN100767_c0_g1                            0.0   \n",
       "1   TRINITY_DN10175_c0_g1                            1.0   \n",
       "2   TRINITY_DN10249_c0_g1                            0.0   \n",
       "3  TRINITY_DN102633_c0_g1                            0.0   \n",
       "4  TRINITY_DN102877_c0_g1                          125.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            2.0                            0.0   \n",
       "4                           79.0                           78.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            1.0                            0.0   \n",
       "3                            9.0                            2.0   \n",
       "4                           52.0                           55.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  6.0   \n",
       "1                            2.0                                  1.0   \n",
       "2                            0.0                                 17.0   \n",
       "3                            1.0                                  5.0   \n",
       "4                          105.0                                 40.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  1.0   \n",
       "1                                  2.0                                  9.0   \n",
       "2                                  0.0                                  0.0   \n",
       "3                                  0.0                                  0.0   \n",
       "4                                 92.0                                 72.0   \n",
       "\n",
       "               TrinityContig              seqid  \\\n",
       "0  TRINITY_DN100767_c0_g1_i1  Transcript_221533   \n",
       "1   TRINITY_DN10175_c0_g1_i1  Transcript_605742   \n",
       "2   TRINITY_DN10249_c0_g1_i1  Transcript_618557   \n",
       "3  TRINITY_DN102633_c0_g1_i1  Transcript_201922   \n",
       "4  TRINITY_DN102877_c0_g1_i1  Transcript_143501   \n",
       "\n",
       "                                       Fhet_GeneName  start  end  length  \n",
       "0  gi|831525541|ref|XP_012719085.1| PREDICTED: zo...      9   56      47  \n",
       "1  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...      0  220     220  \n",
       "2  gi|831577816|ref|XP_012737259.1| PREDICTED: ez...      0   45      45  \n",
       "3  gi|831498824|ref|XP_012709872.1| PREDICTED: co...     30   68      38  \n",
       "4  gi|831491323|ref|XP_012707217.1| PREDICTED: LY...      0   26      26  "
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fhet_merged_table['split1'], fhet_merged_table['split2'],fhet_merged_table['split3'],fhet_merged_table['NCBIproteinID'],fhet_merged_table['NCBIproteinName'] = fhet_merged_table['Fhet_GeneName'].str.split('|', 5).str\n",
    "fhet_merged_table = fhet_merged_table.drop('split1',1)\n",
    "fhet_merged_table = fhet_merged_table.drop('split2',1)\n",
    "fhet_merged_table = fhet_merged_table.drop('split3',1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th>NCBIproteinName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN100767_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TRINITY_DN100767_c0_g1_i1</td>\n",
       "      <td>Transcript_221533</td>\n",
       "      <td>gi|831525541|ref|XP_012719085.1| PREDICTED: zo...</td>\n",
       "      <td>9</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "      <td>XP_012719085.1</td>\n",
       "      <td>PREDICTED: zona pellucida sperm-binding prote...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10249_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN10249_c0_g1_i1</td>\n",
       "      <td>Transcript_618557</td>\n",
       "      <td>gi|831577816|ref|XP_012737259.1| PREDICTED: ez...</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>XP_012737259.1</td>\n",
       "      <td>PREDICTED: ezrin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN102633_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN102633_c0_g1_i1</td>\n",
       "      <td>Transcript_201922</td>\n",
       "      <td>gi|831498824|ref|XP_012709872.1| PREDICTED: co...</td>\n",
       "      <td>30</td>\n",
       "      <td>68</td>\n",
       "      <td>38</td>\n",
       "      <td>XP_012709872.1</td>\n",
       "      <td>PREDICTED: collagen alpha-1(XXVII) chain B-li...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN102877_c0_g1</td>\n",
       "      <td>125.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>TRINITY_DN102877_c0_g1_i1</td>\n",
       "      <td>Transcript_143501</td>\n",
       "      <td>gi|831491323|ref|XP_012707217.1| PREDICTED: LY...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>XP_012707217.1</td>\n",
       "      <td>PREDICTED: LYR motif-containing protein 4 iso...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN100767_c0_g1                            0.0   \n",
       "1   TRINITY_DN10175_c0_g1                            1.0   \n",
       "2   TRINITY_DN10249_c0_g1                            0.0   \n",
       "3  TRINITY_DN102633_c0_g1                            0.0   \n",
       "4  TRINITY_DN102877_c0_g1                          125.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            2.0                            0.0   \n",
       "4                           79.0                           78.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            1.0                            0.0   \n",
       "3                            9.0                            2.0   \n",
       "4                           52.0                           55.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  6.0   \n",
       "1                            2.0                                  1.0   \n",
       "2                            0.0                                 17.0   \n",
       "3                            1.0                                  5.0   \n",
       "4                          105.0                                 40.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  1.0   \n",
       "1                                  2.0                                  9.0   \n",
       "2                                  0.0                                  0.0   \n",
       "3                                  0.0                                  0.0   \n",
       "4                                 92.0                                 72.0   \n",
       "\n",
       "               TrinityContig              seqid  \\\n",
       "0  TRINITY_DN100767_c0_g1_i1  Transcript_221533   \n",
       "1   TRINITY_DN10175_c0_g1_i1  Transcript_605742   \n",
       "2   TRINITY_DN10249_c0_g1_i1  Transcript_618557   \n",
       "3  TRINITY_DN102633_c0_g1_i1  Transcript_201922   \n",
       "4  TRINITY_DN102877_c0_g1_i1  Transcript_143501   \n",
       "\n",
       "                                       Fhet_GeneName  start  end  length  \\\n",
       "0  gi|831525541|ref|XP_012719085.1| PREDICTED: zo...      9   56      47   \n",
       "1  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...      0  220     220   \n",
       "2  gi|831577816|ref|XP_012737259.1| PREDICTED: ez...      0   45      45   \n",
       "3  gi|831498824|ref|XP_012709872.1| PREDICTED: co...     30   68      38   \n",
       "4  gi|831491323|ref|XP_012707217.1| PREDICTED: LY...      0   26      26   \n",
       "\n",
       "    NCBIproteinID                                    NCBIproteinName  \n",
       "0  XP_012719085.1   PREDICTED: zona pellucida sperm-binding prote...  \n",
       "1  XP_012706852.1      PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "2  XP_012737259.1      PREDICTED: ezrin-like [Fundulus heteroclitus]  \n",
       "3  XP_012709872.1   PREDICTED: collagen alpha-1(XXVII) chain B-li...  \n",
       "4  XP_012707217.1   PREDICTED: LYR motif-containing protein 4 iso...  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(66797, 18)"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th>NCBIproteinName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>TRINITY_DN146980_c0_g1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>TRINITY_DN146980_c0_g1_i1</td>\n",
       "      <td>Transcript_91068</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>497</td>\n",
       "      <td>497</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>TRINITY_DN170990_c2_g1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>TRINITY_DN170990_c2_g1_i1</td>\n",
       "      <td>Transcript_202201</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>293</td>\n",
       "      <td>293</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>TRINITY_DN171266_c0_g2</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN171266_c0_g2_i1</td>\n",
       "      <td>Transcript_175700</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>309</td>\n",
       "      <td>309</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1367</th>\n",
       "      <td>TRINITY_DN179170_c2_g1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>TRINITY_DN179170_c2_g1_i1</td>\n",
       "      <td>Transcript_54728</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>218</td>\n",
       "      <td>218</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "1      TRINITY_DN10175_c0_g1                            1.0   \n",
       "465   TRINITY_DN146980_c0_g1                            2.0   \n",
       "1016  TRINITY_DN170990_c2_g1                            3.0   \n",
       "1022  TRINITY_DN171266_c0_g2                            8.0   \n",
       "1367  TRINITY_DN179170_c2_g1                            4.0   \n",
       "\n",
       "      F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "1                               0.0                            0.0   \n",
       "465                             2.0                            1.0   \n",
       "1016                            1.0                            2.0   \n",
       "1022                            0.0                            0.0   \n",
       "1367                            6.0                            0.0   \n",
       "\n",
       "      F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "1                               0.0                            0.0   \n",
       "465                             3.0                            0.0   \n",
       "1016                            3.0                            3.0   \n",
       "1022                            0.0                            1.0   \n",
       "1367                            2.0                            2.0   \n",
       "\n",
       "      F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "1                               2.0                                  1.0   \n",
       "465                             8.0                                  3.0   \n",
       "1016                            1.0                                  2.0   \n",
       "1022                            2.0                                  2.0   \n",
       "1367                            1.0                                  2.0   \n",
       "\n",
       "      F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "1                                     2.0   \n",
       "465                                   6.0   \n",
       "1016                                  0.0   \n",
       "1022                                  2.5   \n",
       "1367                                  3.0   \n",
       "\n",
       "      F_heteroclitusMDPP_transfer_3.quant              TrinityContig  \\\n",
       "1                                     9.0   TRINITY_DN10175_c0_g1_i1   \n",
       "465                                  27.0  TRINITY_DN146980_c0_g1_i1   \n",
       "1016                                  7.0  TRINITY_DN170990_c2_g1_i1   \n",
       "1022                                  0.0  TRINITY_DN171266_c0_g2_i1   \n",
       "1367                                  3.0  TRINITY_DN179170_c2_g1_i1   \n",
       "\n",
       "                  seqid                                      Fhet_GeneName  \\\n",
       "1     Transcript_605742  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "465    Transcript_91068  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1016  Transcript_202201  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1022  Transcript_175700  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1367   Transcript_54728  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "\n",
       "      start  end  length   NCBIproteinID  \\\n",
       "1         0  220     220  XP_012706852.1   \n",
       "465       0  497     497  XP_012706852.1   \n",
       "1016      0  293     293  XP_012706852.1   \n",
       "1022      0  309     309  XP_012706852.1   \n",
       "1367      0  218     218  XP_012706852.1   \n",
       "\n",
       "                                     NCBIproteinName  \n",
       "1      PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "465    PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1016   PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1022   PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1367   PREDICTED: titin-like [Fundulus heteroclitus]  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "titin = fhet_merged_table.loc[fhet_merged_table['NCBIproteinID'] == 'XP_012706852.1']\n",
    "titin.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort by length descending, drop duplicate genes\n",
    "titin_gene = titin.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "#titin_gene.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>XP_012706852.1</th>\n",
       "      <td>9267.948719</td>\n",
       "      <td>5022.968313</td>\n",
       "      <td>2300.499999</td>\n",
       "      <td>7129.471635</td>\n",
       "      <td>2996.556833</td>\n",
       "      <td>4430.794692</td>\n",
       "      <td>3312.109758</td>\n",
       "      <td>7756.500001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    9267.948719                    5022.968313   \n",
       "\n",
       "                F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    2300.499999                    7129.471635   \n",
       "\n",
       "                F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    2996.556833                    4430.794692   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "NCBIproteinID                                         \n",
       "XP_012706852.1                          3312.109758   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_2.quant  \n",
       "NCBIproteinID                                        \n",
       "XP_012706852.1                          7756.500001  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now, collapse matrix by protein, sum expression for all fragments for each sample\n",
    "# colSum[2:10] for unique proteins 'NCBIproteinID'\n",
    "titin_collapse = titin_gene.groupby(['NCBIproteinID'])[list(titin_gene.columns)[1:9]].agg('sum')\n",
    "#titin_collapse = titin_gene.groupby(['NCBIproteinID']).sum()\n",
    "titin_collapse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(27387, 18)\n",
      "(18626, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>XP_012704699.1</th>\n",
       "      <td>277.294571</td>\n",
       "      <td>329.0</td>\n",
       "      <td>165.227518</td>\n",
       "      <td>250.000000</td>\n",
       "      <td>143.160386</td>\n",
       "      <td>170.000000</td>\n",
       "      <td>110.590974</td>\n",
       "      <td>213.904699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704700.1</th>\n",
       "      <td>373.000000</td>\n",
       "      <td>478.0</td>\n",
       "      <td>242.000001</td>\n",
       "      <td>409.000000</td>\n",
       "      <td>265.202322</td>\n",
       "      <td>334.000000</td>\n",
       "      <td>207.035645</td>\n",
       "      <td>505.690830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704702.1</th>\n",
       "      <td>664.999999</td>\n",
       "      <td>837.0</td>\n",
       "      <td>476.000000</td>\n",
       "      <td>823.000000</td>\n",
       "      <td>520.699682</td>\n",
       "      <td>695.999999</td>\n",
       "      <td>304.000000</td>\n",
       "      <td>526.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704705.1</th>\n",
       "      <td>892.841678</td>\n",
       "      <td>851.0</td>\n",
       "      <td>514.000000</td>\n",
       "      <td>727.999999</td>\n",
       "      <td>353.000000</td>\n",
       "      <td>727.000000</td>\n",
       "      <td>276.000000</td>\n",
       "      <td>731.000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704706.1</th>\n",
       "      <td>7.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     277.294571                          329.0   \n",
       "XP_012704700.1                     373.000000                          478.0   \n",
       "XP_012704702.1                     664.999999                          837.0   \n",
       "XP_012704705.1                     892.841678                          851.0   \n",
       "XP_012704706.1                       7.000000                            5.0   \n",
       "\n",
       "                F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     165.227518                     250.000000   \n",
       "XP_012704700.1                     242.000001                     409.000000   \n",
       "XP_012704702.1                     476.000000                     823.000000   \n",
       "XP_012704705.1                     514.000000                     727.999999   \n",
       "XP_012704706.1                       3.000000                      10.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     143.160386                     170.000000   \n",
       "XP_012704700.1                     265.202322                     334.000000   \n",
       "XP_012704702.1                     520.699682                     695.999999   \n",
       "XP_012704705.1                     353.000000                     727.000000   \n",
       "XP_012704706.1                       0.000000                       1.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "NCBIproteinID                                         \n",
       "XP_012704699.1                           110.590974   \n",
       "XP_012704700.1                           207.035645   \n",
       "XP_012704702.1                           304.000000   \n",
       "XP_012704705.1                           276.000000   \n",
       "XP_012704706.1                             4.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_2.quant  \n",
       "NCBIproteinID                                        \n",
       "XP_012704699.1                           213.904699  \n",
       "XP_012704700.1                           505.690830  \n",
       "XP_012704702.1                           526.000000  \n",
       "XP_012704705.1                           731.000001  \n",
       "XP_012704706.1                             2.000000  "
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now do this on everything\n",
    "fhet = fhet_merged_table.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "print(fhet.shape)\n",
    "fhet_collapse = fhet.groupby(['NCBIproteinID'])[list(fhet.columns)[1:9]].agg('sum')\n",
    "print(fhet_collapse.shape)\n",
    "fhet_collapse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet annotations, contigs with expression\n",
      "20877\n",
      "Unique NCBI protein ID\n",
      "20877\n",
      "Unique Trinity \"genes\"\n",
      "27387\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "66797\n"
     ]
    }
   ],
   "source": [
    "print(\"Unique Fhet annotations, contigs with expression\")\n",
    "print(len(fhet_merged_table.Fhet_GeneName.unique()))\n",
    "print('Unique NCBI protein ID')\n",
    "print(len(fhet_merged_table.NCBIproteinID.unique()))\n",
    "print('Unique Trinity \"genes\"')\n",
    "print(len(fhet_merged_table.Gene.unique()))\n",
    "print('Unique Trinity \"transcripts\" (contigs) ')\n",
    "print(len(fhet_merged_table.TrinityContig.unique()))\n",
    "# fragments\n",
    "#fhet_merged_table.to_csv(gene_out)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
