{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. After tximport in R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Script: full_model.Rmd\n",
    "* Input: \"~/Documents/UCDavis/Whitehead/RNAseq_15killifish/salmon_denovo_by_species/\"\n",
    "* Output: \"~/Documents/UCDavis/Whitehead/counts_matrices/*_counts.csv\" for each species "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Merge annotations for each species, connecting Trinity contigs/genes to annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using dammit's `GFF3Parser` function \n",
    "1. Digests the gff3 file for each species (output from dammit, downloaded from the farm cluster)\n",
    "2. Sorts each contig by E-value score\n",
    "3. Assigns the lowest E-value score for each contig \n",
    "4. Separately, for each contig, saves gene names from the NCBI annotated F. heteroclitus genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "# requires dammit env\n",
    "# source activate dammit\n",
    "from dammit.fileio.gff3 import GFF3Parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['F_heteroclitusMDPP_counts.csv', 'F_parvapinis_counts.csv', 'L_parva_counts.csv', 'F_notatus_counts.csv', 'F_heteroclitusMDPL_counts.csv', 'F_grandis_counts.csv', 'F_zebrinus_counts.csv', 'A_xenica_counts.csv', 'F_olivaceus_counts.csv', 'F_catanatus_counts.csv', 'F_sciadicus_counts.csv', 'F_rathbuni_counts.csv', 'F_chrysotus_counts.csv', 'F_diaphanus_counts.csv', 'L_goodei_counts.csv', 'F_nottii_counts.csv', 'F_similis_counts.csv']\n"
     ]
    }
   ],
   "source": [
    "counts_matrices = \"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices/\"\n",
    "counts_files = os.listdir(counts_matrices)\n",
    "gene_out_dir = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_name_16Oct2018_filtnew/\"\n",
    "print(counts_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "========\n",
      "F_heteroclitusMDPP\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(496133, 10)\n",
      "Contigs filtered:\n",
      "(174264, 10)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/johnsolk/anaconda3/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet gene names (one name per contig):\n",
      "22641\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20558\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20558\n",
      "Unique NCBI protein ID\n",
      "20558\n",
      "Unique Trinity \"genes\"\n",
      "27387\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "66797\n",
      "Collaposed genes\n",
      "(27387, 18)\n",
      "Collaposed proteins\n",
      "(18626, 12)\n",
      "========\n",
      "F_parvapinis\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(279009, 9)\n",
      "Contigs filtered:\n",
      "(82915, 9)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20212\n",
      "Unique Fhet gene names (one name per contig):\n",
      "18293\n",
      "Unique Fhet annotations, contigs with expression\n",
      "18293\n",
      "Unique NCBI protein ID\n",
      "18293\n",
      "Unique Trinity \"genes\"\n",
      "21139\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "55189\n",
      "Collaposed genes\n",
      "(21139, 17)\n",
      "Collaposed proteins\n",
      "(16307, 11)\n",
      "========\n",
      "L_parva\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(275950, 10)\n",
      "Contigs filtered:\n",
      "(84392, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22218\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20629\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20629\n",
      "Unique NCBI protein ID\n",
      "20629\n",
      "Unique Trinity \"genes\"\n",
      "24381\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "65126\n",
      "Collaposed genes\n",
      "(24381, 18)\n",
      "Collaposed proteins\n",
      "(18425, 12)\n",
      "========\n",
      "F_notatus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(286353, 10)\n",
      "Contigs filtered:\n",
      "(99871, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22060\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20441\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20441\n",
      "Unique NCBI protein ID\n",
      "20441\n",
      "Unique Trinity \"genes\"\n",
      "26018\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "73212\n",
      "Collaposed genes\n",
      "(26018, 18)\n",
      "Collaposed proteins\n",
      "(18138, 12)\n",
      "========\n",
      "F_heteroclitusMDPL\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(392026, 10)\n",
      "Contigs filtered:\n",
      "(138702, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22499\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20606\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20606\n",
      "Unique NCBI protein ID\n",
      "20606\n",
      "Unique Trinity \"genes\"\n",
      "28257\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "68140\n",
      "Collaposed genes\n",
      "(28257, 18)\n",
      "Collaposed proteins\n",
      "(18809, 12)\n",
      "========\n",
      "F_grandis\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(608505, 10)\n",
      "Contigs filtered:\n",
      "(218430, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "23805\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22105\n",
      "Unique Fhet annotations, contigs with expression\n",
      "22105\n",
      "Unique NCBI protein ID\n",
      "22105\n",
      "Unique Trinity \"genes\"\n",
      "31788\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "82409\n",
      "Collaposed genes\n",
      "(31788, 18)\n",
      "Collaposed proteins\n",
      "(19999, 12)\n",
      "========\n",
      "F_zebrinus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(194115, 5)\n",
      "Contigs filtered:\n",
      "(74775, 5)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20076\n",
      "Unique Fhet gene names (one name per contig):\n",
      "18433\n",
      "Unique Fhet annotations, contigs with expression\n",
      "18433\n",
      "Unique NCBI protein ID\n",
      "18433\n",
      "Unique Trinity \"genes\"\n",
      "21945\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "49386\n",
      "Collaposed genes\n",
      "(21945, 13)\n",
      "Collaposed proteins\n",
      "(16743, 7)\n",
      "========\n",
      "A_xenica\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(248903, 10)\n",
      "Contigs filtered:\n",
      "(83882, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21414\n",
      "Unique Fhet gene names (one name per contig):\n",
      "19592\n",
      "Unique Fhet annotations, contigs with expression\n",
      "19592\n",
      "Unique NCBI protein ID\n",
      "19592\n",
      "Unique Trinity \"genes\"\n",
      "23407\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "64889\n",
      "Collaposed genes\n",
      "(23407, 18)\n",
      "Collaposed proteins\n",
      "(17512, 12)\n",
      "========\n",
      "F_olivaceus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(248630, 9)\n",
      "Contigs filtered:\n",
      "(76420, 9)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21208\n",
      "Unique Fhet gene names (one name per contig):\n",
      "19329\n",
      "Unique Fhet annotations, contigs with expression\n",
      "19329\n",
      "Unique NCBI protein ID\n",
      "19329\n",
      "Unique Trinity \"genes\"\n",
      "22166\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "60861\n",
      "Collaposed genes\n",
      "(22166, 17)\n",
      "Collaposed proteins\n",
      "(17166, 11)\n",
      "========\n",
      "F_catanatus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(283212, 8)\n",
      "Contigs filtered:\n",
      "(117481, 8)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22013\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20462\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20462\n",
      "Unique NCBI protein ID\n",
      "20462\n",
      "Unique Trinity \"genes\"\n",
      "26778\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "69579\n",
      "Collaposed genes\n",
      "(26778, 16)\n",
      "Collaposed proteins\n",
      "(18255, 10)\n",
      "========\n",
      "F_sciadicus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(178995, 5)\n",
      "Contigs filtered:\n",
      "(75575, 5)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20050\n",
      "Unique Fhet gene names (one name per contig):\n",
      "18622\n",
      "Unique Fhet annotations, contigs with expression\n",
      "18622\n",
      "Unique NCBI protein ID\n",
      "18622\n",
      "Unique Trinity \"genes\"\n",
      "22301\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "48443\n",
      "Collaposed genes\n",
      "(22301, 13)\n",
      "Collaposed proteins\n",
      "(16748, 7)\n",
      "========\n",
      "F_rathbuni\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(364231, 10)\n",
      "Contigs filtered:\n",
      "(144579, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22270\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20759\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20759\n",
      "Unique NCBI protein ID\n",
      "20759\n",
      "Unique Trinity \"genes\"\n",
      "25566\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "72736\n",
      "Collaposed genes\n",
      "(25566, 18)\n",
      "Collaposed proteins\n",
      "(18382, 12)\n",
      "========\n",
      "F_chrysotus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(269787, 9)\n",
      "Contigs filtered:\n",
      "(94962, 9)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21988\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20363\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20363\n",
      "Unique NCBI protein ID\n",
      "20363\n",
      "Unique Trinity \"genes\"\n",
      "25644\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "71777\n",
      "Collaposed genes\n",
      "(25644, 17)\n",
      "Collaposed proteins\n",
      "(18166, 11)\n",
      "========\n",
      "F_diaphanus\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(288318, 7)\n",
      "Contigs filtered:\n",
      "(112089, 7)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21264\n",
      "Unique Fhet gene names (one name per contig):\n",
      "19464\n",
      "Unique Fhet annotations, contigs with expression\n",
      "19464\n",
      "Unique NCBI protein ID\n",
      "19464\n",
      "Unique Trinity \"genes\"\n",
      "23529\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "54959\n",
      "Collaposed genes\n",
      "(23529, 15)\n",
      "Collaposed proteins\n",
      "(17588, 9)\n",
      "========\n",
      "L_goodei\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(266317, 10)\n",
      "Contigs filtered:\n",
      "(81555, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "22238\n",
      "Unique Fhet gene names (one name per contig):\n",
      "20275\n",
      "Unique Fhet annotations, contigs with expression\n",
      "20275\n",
      "Unique NCBI protein ID\n",
      "20275\n",
      "Unique Trinity \"genes\"\n",
      "25259\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "64771\n",
      "Collaposed genes\n",
      "(25259, 18)\n",
      "Collaposed proteins\n",
      "(18186, 12)\n",
      "========\n",
      "F_nottii\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(123565, 3)\n",
      "Contigs filtered:\n",
      "(59992, 3)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "17997\n",
      "Unique Fhet gene names (one name per contig):\n",
      "16674\n",
      "Unique Fhet annotations, contigs with expression\n",
      "16674\n",
      "Unique NCBI protein ID\n",
      "16674\n",
      "Unique Trinity \"genes\"\n",
      "21117\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "35895\n",
      "Collaposed genes\n",
      "(21117, 11)\n",
      "Collaposed proteins\n",
      "(15504, 5)\n",
      "========\n",
      "F_similis\n",
      "========\n",
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(392860, 10)\n",
      "Contigs filtered:\n",
      "(93199, 10)\n",
      "Unique Fhet gene names (one name per contig):\n",
      "21905\n",
      "Unique Fhet gene names (one name per contig):\n",
      "19484\n",
      "Unique Fhet annotations, contigs with expression\n",
      "19484\n",
      "Unique NCBI protein ID\n",
      "19484\n",
      "Unique Trinity \"genes\"\n",
      "23816\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "63078\n",
      "Collaposed genes\n",
      "(23816, 18)\n",
      "Collaposed proteins\n",
      "(17506, 12)\n"
     ]
    }
   ],
   "source": [
    "# 10/15/2018\n",
    "# keep only F. het\n",
    "# then pick one per contig\n",
    "# if a contig does not have an F. het annotation, \n",
    "# keep track of this? but then drop it\n",
    "for counts_file in counts_files:\n",
    "    if counts_file != \".DS_Store\":\n",
    "        species = counts_file.split(\"_\")[0]+\"_\"+counts_file.split(\"_\")[1]\n",
    "        print(\"========\")\n",
    "        print(species)\n",
    "        print(\"========\")\n",
    "        gene_out = gene_out_dir + species + \"_gene_counts_annotations_filt.csv\"\n",
    "        table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices/\"+counts_file)\n",
    "        print('Number of Trinity \"genes\" (this is how we summarized expression):')\n",
    "        print(table.shape)\n",
    "        table = table.rename(columns={'Unnamed: 0': 'Gene'})\n",
    "        # if countsvalue is >5 in any column, then keep\n",
    "        table_filt = table[(table.iloc[:,1:] > 5).any(1)]\n",
    "        print('Contigs filtered:')\n",
    "        print(table_filt.shape)\n",
    "        name = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"+species+\".trinity_out.Trinity.fasta.dammit.gff3\"\n",
    "        conversion_contig = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_23June2018/\"+species+\"_contig_gene.csv\"\n",
    "        conversion_dammit = \"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"+species+\".trinity_out.Trinity.fasta.dammit.namemap.csv\"\n",
    "        annotations = GFF3Parser(filename=name).read()\n",
    "        annotations = annotations.dropna(subset=['Name'])\n",
    "        # keeps track of how long the annotation is\n",
    "        annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)\n",
    "        conversions_dammit = pd.read_csv(conversion_dammit)\n",
    "        conversions_contig = pd.read_csv(conversion_contig)\n",
    "        conversions_dammit['Name'], conversions_dammit['info'] = conversions_dammit['original'].str.split(' ', 1).str\n",
    "        conversions_dammit = conversions_dammit[['Name','renamed']]\n",
    "        conversions_dammit.columns = ['Name','seqid']\n",
    "        coversions_contig = conversions_contig[['Name','Gene']]\n",
    "        # merge filtered counts with Trinity contig and gene ID\n",
    "        merged_table = pd.merge(table_filt,conversions_contig,on=\"Gene\",how='left')\n",
    "        # merge with dammit transcript ID\n",
    "        merged_table = pd.merge(merged_table,conversions_dammit,on=\"Name\",how='left')\n",
    "        # merge with annotation names\n",
    "        fhet = annotations[annotations['Name'].str.startswith(\"gi\")]\n",
    "        # pick best e-value match\n",
    "        fhet_filtered = fhet.sort_values(by=['seqid','score'],ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "        print(\"Unique Fhet gene names (one name per contig):\")\n",
    "        print(len(fhet_filtered.Name.unique()))\n",
    "        fhet_merged_table = pd.merge(merged_table,fhet_filtered,on='seqid',how='inner')\n",
    "        # gets rid of contigs without any F. het annotation\n",
    "        # we probably lose ~2000 because of low counts\n",
    "        # more \"Genes\" than annotations because the contigs are probably fragmented across a gene\n",
    "        print(\"Unique Fhet gene names (one name per contig):\")\n",
    "        print(len(fhet_merged_table.Name_y.unique()))\n",
    "        fhet_merged_table = fhet_merged_table.drop('Unnamed: 0', 1)\n",
    "        fhet_merged_table = fhet_merged_table.rename(columns = {'Name_x':'TrinityContig','Name_y':'Fhet_GeneName'}) \n",
    "        fhet_merged_table['split1'], fhet_merged_table['split2'],fhet_merged_table['split3'],fhet_merged_table['NCBIproteinID'],fhet_merged_table['NCBIproteinName'] = fhet_merged_table['Fhet_GeneName'].str.split('|', 5).str\n",
    "        fhet_merged_table = fhet_merged_table.drop('split1',1)\n",
    "        fhet_merged_table = fhet_merged_table.drop('split2',1)\n",
    "        fhet_merged_table = fhet_merged_table.drop('split3',1)\n",
    "        print(\"Unique Fhet annotations, contigs with expression\")\n",
    "        print(len(fhet_merged_table.Fhet_GeneName.unique()))\n",
    "        print('Unique NCBI protein ID')\n",
    "        print(len(fhet_merged_table.NCBIproteinID.unique()))\n",
    "        print('Unique Trinity \"genes\"')\n",
    "        print(len(fhet_merged_table.Gene.unique()))\n",
    "        print('Unique Trinity \"transcripts\" (contigs) ')\n",
    "        print(len(fhet_merged_table.TrinityContig.unique()))\n",
    "        fhet = fhet_merged_table.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "        print('Collaposed genes')\n",
    "        print(fhet.shape)\n",
    "        fhet_collapse = fhet.groupby(['NCBIproteinID'])[list(fhet.columns)[1:len(list(fhet.columns))-1]].agg('sum')\n",
    "        print('Collaposed proteins')\n",
    "        print(fhet_collapse.shape)\n",
    "        fhet_collapse.head()\n",
    "        fhet_collapse.to_csv(gene_out)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test one species, F. heteroclitus MDPP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Trinity \"genes\" (this is how we summarized expression):\n",
      "(496133, 10)\n"
     ]
    }
   ],
   "source": [
    "gene_out = gene_out_dir + \"F_heteroclitusMDPP\" + \"_gene_counts_annotations_filt.csv\"\n",
    "table = pd.read_csv(\"/Users/johnsolk/Documents/UCDavis/Whitehead/counts_matrices/\"+\"F_heteroclitusMDPP_counts.csv\")\n",
    "print('Number of Trinity \"genes\" (this is how we summarized expression):')\n",
    "print(table.shape)\n",
    "table = table.rename(columns={'Unnamed: 0': 'Gene'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Contigs filtered:\n",
      "(174264, 10)\n"
     ]
    }
   ],
   "source": [
    "# if countsvalue is >5 in any column, then keep\n",
    "table_filt = table[(table.iloc[:,1:] > 5).any(1)]\n",
    "print('Contigs filtered:')\n",
    "print(table_filt.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "name = \"/Users/johnsolk/Documents/UCDavis/Whitehead/gff_annotations/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.gff3\"\n",
    "conversion_contig = \"/Users/johnsolk/Documents/UCDavis/Whitehead/contig_gene_23June2018/\"+\"F_heteroclitusMDPP\"+\"_contig_gene.csv\"\n",
    "conversion_dammit = \"/Users/johnsolk/Documents/UCDavis/Whitehead/dammit_conversions/\"+\"F_heteroclitusMDPP\"+\".trinity_out.Trinity.fasta.dammit.namemap.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/johnsolk/anaconda3/lib/python3.6/site-packages/dammit/fileio/gff3.py:73: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used\n",
      "  dtype=dict(self.columns)):\n"
     ]
    }
   ],
   "source": [
    "annotations = GFF3Parser(filename=name).read()\n",
    "annotations = annotations.dropna(subset=['Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# keeps track of how long the annotation is\n",
    "annotations[\"length\"] = annotations[\"end\"].subtract(annotations[\"start\"], fill_value=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#pickonename = annotations.sort_values(by=['seqid', 'score'], ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','Note','database','Dbxref','start','end','length']]\n",
    "#pickonename = pickonename.dropna(axis=0,how=\"all\")\n",
    "#print('Number of contigs with annotations (one annotation/contig, sorted by E-value < 1e-05 and picked the lowest):')\n",
    "#print(pickonename.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "conversions_dammit = pd.read_csv(conversion_dammit)\n",
    "conversions_contig = pd.read_csv(conversion_contig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "conversions_dammit['Name'], conversions_dammit['info'] = conversions_dammit['original'].str.split(' ', 1).str\n",
    "conversions_dammit = conversions_dammit[['Name','renamed']]\n",
    "conversions_dammit.columns = ['Name','seqid']\n",
    "coversions_contig = conversions_contig[['Name','Gene']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# merge filtered counts with Trinity contig and gene ID\n",
    "merged_table = pd.merge(table_filt,conversions_contig,on=\"Gene\",how='left')\n",
    "# merge with dammit transcript ID\n",
    "merged_table = pd.merge(merged_table,conversions_dammit,on=\"Name\",how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Name</th>\n",
       "      <th>seqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN0_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>604146</td>\n",
       "      <td>TRINITY_DN0_c0_g1_i1</td>\n",
       "      <td>Transcript_604149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN1_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>604184</td>\n",
       "      <td>TRINITY_DN1_c0_g1_i1</td>\n",
       "      <td>Transcript_604187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10000_c1_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>631886</td>\n",
       "      <td>TRINITY_DN10000_c1_g1_i1</td>\n",
       "      <td>Transcript_631889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN100010_c1_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>189616</td>\n",
       "      <td>TRINITY_DN100010_c1_g1_i1</td>\n",
       "      <td>Transcript_189616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN100017_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>189637</td>\n",
       "      <td>TRINITY_DN100017_c0_g1_i1</td>\n",
       "      <td>Transcript_189637</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0       TRINITY_DN0_c0_g1                            0.0   \n",
       "1       TRINITY_DN1_c0_g1                            0.0   \n",
       "2   TRINITY_DN10000_c1_g1                            0.0   \n",
       "3  TRINITY_DN100010_c1_g1                            0.0   \n",
       "4  TRINITY_DN100017_c0_g1                            1.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            2.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            0.0                            0.0   \n",
       "4                            0.0                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            6.0                            0.0   \n",
       "1                           11.0                            0.0   \n",
       "2                           13.0                            1.0   \n",
       "3                            7.0                            0.0   \n",
       "4                           11.0                            0.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  0.0   \n",
       "1                            0.0                                  2.0   \n",
       "2                            0.0                                  4.0   \n",
       "3                            0.0                                  9.0   \n",
       "4                            0.0                                  1.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  0.0   \n",
       "1                                  0.0                                  0.0   \n",
       "2                                  2.0                                  0.0   \n",
       "3                                  1.0                                  0.0   \n",
       "4                                  1.0                                  0.0   \n",
       "\n",
       "   Unnamed: 0                       Name              seqid  \n",
       "0      604146       TRINITY_DN0_c0_g1_i1  Transcript_604149  \n",
       "1      604184       TRINITY_DN1_c0_g1_i1  Transcript_604187  \n",
       "2      631886   TRINITY_DN10000_c1_g1_i1  Transcript_631889  \n",
       "3      189616  TRINITY_DN100010_c1_g1_i1  Transcript_189616  \n",
       "4      189637  TRINITY_DN100017_c0_g1_i1  Transcript_189637  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### merge with annotation names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet gene names (one name per contig):\n",
      "22641\n"
     ]
    }
   ],
   "source": [
    "fhet = annotations[annotations['Name'].str.startswith(\"gi\")]\n",
    "fhet_filtered = fhet.sort_values(by=['seqid','score'],ascending=True).query('score < 1e-05').drop_duplicates(subset='seqid')[['seqid', 'Name','start','end','length']]\n",
    "print(\"Unique Fhet gene names (one name per contig):\")\n",
    "print(len(fhet_filtered.Name.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Transcript_100053</td>\n",
       "      <td>gi|831512517|ref|XP_012714591.1| PREDICTED: br...</td>\n",
       "      <td>0</td>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>Transcript_100263</td>\n",
       "      <td>gi|831554709|ref|XP_012729460.1| PREDICTED: mu...</td>\n",
       "      <td>0</td>\n",
       "      <td>122</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>Transcript_100278</td>\n",
       "      <td>gi|831477084|ref|XP_012712778.1| PREDICTED: ub...</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>Transcript_100279</td>\n",
       "      <td>gi|831477084|ref|XP_012712778.1| PREDICTED: ub...</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>Transcript_100291</td>\n",
       "      <td>gi|831575409|ref|XP_012736406.1| PREDICTED: ne...</td>\n",
       "      <td>0</td>\n",
       "      <td>88</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seqid                                               Name  \\\n",
       "34   Transcript_100053  gi|831512517|ref|XP_012714591.1| PREDICTED: br...   \n",
       "134  Transcript_100263  gi|831554709|ref|XP_012729460.1| PREDICTED: mu...   \n",
       "146  Transcript_100278  gi|831477084|ref|XP_012712778.1| PREDICTED: ub...   \n",
       "148  Transcript_100279  gi|831477084|ref|XP_012712778.1| PREDICTED: ub...   \n",
       "159  Transcript_100291  gi|831575409|ref|XP_012736406.1| PREDICTED: ne...   \n",
       "\n",
       "     start  end  length  \n",
       "34       0  124     124  \n",
       "134      0  122     122  \n",
       "146      0   70      70  \n",
       "148      0   70      70  \n",
       "159      0   88      88  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN100767_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>221533</td>\n",
       "      <td>TRINITY_DN100767_c0_g1_i1</td>\n",
       "      <td>Transcript_221533</td>\n",
       "      <td>gi|831525541|ref|XP_012719085.1| PREDICTED: zo...</td>\n",
       "      <td>9</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>605739</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10249_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>618554</td>\n",
       "      <td>TRINITY_DN10249_c0_g1_i1</td>\n",
       "      <td>Transcript_618557</td>\n",
       "      <td>gi|831577816|ref|XP_012737259.1| PREDICTED: ez...</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN102633_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>201922</td>\n",
       "      <td>TRINITY_DN102633_c0_g1_i1</td>\n",
       "      <td>Transcript_201922</td>\n",
       "      <td>gi|831498824|ref|XP_012709872.1| PREDICTED: co...</td>\n",
       "      <td>30</td>\n",
       "      <td>68</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN102877_c0_g1</td>\n",
       "      <td>125.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>143501</td>\n",
       "      <td>TRINITY_DN102877_c0_g1_i1</td>\n",
       "      <td>Transcript_143501</td>\n",
       "      <td>gi|831491323|ref|XP_012707217.1| PREDICTED: LY...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN100767_c0_g1                            0.0   \n",
       "1   TRINITY_DN10175_c0_g1                            1.0   \n",
       "2   TRINITY_DN10249_c0_g1                            0.0   \n",
       "3  TRINITY_DN102633_c0_g1                            0.0   \n",
       "4  TRINITY_DN102877_c0_g1                          125.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            2.0                            0.0   \n",
       "4                           79.0                           78.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            1.0                            0.0   \n",
       "3                            9.0                            2.0   \n",
       "4                           52.0                           55.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  6.0   \n",
       "1                            2.0                                  1.0   \n",
       "2                            0.0                                 17.0   \n",
       "3                            1.0                                  5.0   \n",
       "4                          105.0                                 40.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  1.0   \n",
       "1                                  2.0                                  9.0   \n",
       "2                                  0.0                                  0.0   \n",
       "3                                  0.0                                  0.0   \n",
       "4                                 92.0                                 72.0   \n",
       "\n",
       "   Unnamed: 0                     Name_x              seqid  \\\n",
       "0      221533  TRINITY_DN100767_c0_g1_i1  Transcript_221533   \n",
       "1      605739   TRINITY_DN10175_c0_g1_i1  Transcript_605742   \n",
       "2      618554   TRINITY_DN10249_c0_g1_i1  Transcript_618557   \n",
       "3      201922  TRINITY_DN102633_c0_g1_i1  Transcript_201922   \n",
       "4      143501  TRINITY_DN102877_c0_g1_i1  Transcript_143501   \n",
       "\n",
       "                                              Name_y  start  end  length  \n",
       "0  gi|831525541|ref|XP_012719085.1| PREDICTED: zo...      9   56      47  \n",
       "1  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...      0  220     220  \n",
       "2  gi|831577816|ref|XP_012737259.1| PREDICTED: ez...      0   45      45  \n",
       "3  gi|831498824|ref|XP_012709872.1| PREDICTED: co...     30   68      38  \n",
       "4  gi|831491323|ref|XP_012707217.1| PREDICTED: LY...      0   26      26  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table = pd.merge(merged_table,fhet_filtered,on='seqid',how='inner')\n",
    "fhet_merged_table.head()\n",
    "# gets rid of contigs without any F. het annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(66797, 17)"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet gene names (one name per contig):\n",
      "20558\n"
     ]
    }
   ],
   "source": [
    "# we probably lose ~2000 because of low counts\n",
    "# more \"Genes\" than annotations because the contigs are probably fragmented across a gene\n",
    "print(\"Unique Fhet gene names (one name per contig):\")\n",
    "print(len(fhet_merged_table.Name_y.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fhet_merged_table = fhet_merged_table.drop('Unnamed: 0', 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fhet_merged_table = fhet_merged_table.rename(columns = {'Name_x':'TrinityContig','Name_y':'Fhet_GeneName'})                                           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN100767_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TRINITY_DN100767_c0_g1_i1</td>\n",
       "      <td>Transcript_221533</td>\n",
       "      <td>gi|831525541|ref|XP_012719085.1| PREDICTED: zo...</td>\n",
       "      <td>9</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10249_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN10249_c0_g1_i1</td>\n",
       "      <td>Transcript_618557</td>\n",
       "      <td>gi|831577816|ref|XP_012737259.1| PREDICTED: ez...</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN102633_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN102633_c0_g1_i1</td>\n",
       "      <td>Transcript_201922</td>\n",
       "      <td>gi|831498824|ref|XP_012709872.1| PREDICTED: co...</td>\n",
       "      <td>30</td>\n",
       "      <td>68</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN102877_c0_g1</td>\n",
       "      <td>125.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>TRINITY_DN102877_c0_g1_i1</td>\n",
       "      <td>Transcript_143501</td>\n",
       "      <td>gi|831491323|ref|XP_012707217.1| PREDICTED: LY...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN100767_c0_g1                            0.0   \n",
       "1   TRINITY_DN10175_c0_g1                            1.0   \n",
       "2   TRINITY_DN10249_c0_g1                            0.0   \n",
       "3  TRINITY_DN102633_c0_g1                            0.0   \n",
       "4  TRINITY_DN102877_c0_g1                          125.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            2.0                            0.0   \n",
       "4                           79.0                           78.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            1.0                            0.0   \n",
       "3                            9.0                            2.0   \n",
       "4                           52.0                           55.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  6.0   \n",
       "1                            2.0                                  1.0   \n",
       "2                            0.0                                 17.0   \n",
       "3                            1.0                                  5.0   \n",
       "4                          105.0                                 40.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  1.0   \n",
       "1                                  2.0                                  9.0   \n",
       "2                                  0.0                                  0.0   \n",
       "3                                  0.0                                  0.0   \n",
       "4                                 92.0                                 72.0   \n",
       "\n",
       "               TrinityContig              seqid  \\\n",
       "0  TRINITY_DN100767_c0_g1_i1  Transcript_221533   \n",
       "1   TRINITY_DN10175_c0_g1_i1  Transcript_605742   \n",
       "2   TRINITY_DN10249_c0_g1_i1  Transcript_618557   \n",
       "3  TRINITY_DN102633_c0_g1_i1  Transcript_201922   \n",
       "4  TRINITY_DN102877_c0_g1_i1  Transcript_143501   \n",
       "\n",
       "                                       Fhet_GeneName  start  end  length  \n",
       "0  gi|831525541|ref|XP_012719085.1| PREDICTED: zo...      9   56      47  \n",
       "1  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...      0  220     220  \n",
       "2  gi|831577816|ref|XP_012737259.1| PREDICTED: ez...      0   45      45  \n",
       "3  gi|831498824|ref|XP_012709872.1| PREDICTED: co...     30   68      38  \n",
       "4  gi|831491323|ref|XP_012707217.1| PREDICTED: LY...      0   26      26  "
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fhet_merged_table['split1'], fhet_merged_table['split2'],fhet_merged_table['split3'],fhet_merged_table['NCBIproteinID'],fhet_merged_table['NCBIproteinName'] = fhet_merged_table['Fhet_GeneName'].str.split('|', 5).str\n",
    "fhet_merged_table = fhet_merged_table.drop('split1',1)\n",
    "fhet_merged_table = fhet_merged_table.drop('split2',1)\n",
    "fhet_merged_table = fhet_merged_table.drop('split3',1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th>NCBIproteinName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRINITY_DN100767_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TRINITY_DN100767_c0_g1_i1</td>\n",
       "      <td>Transcript_221533</td>\n",
       "      <td>gi|831525541|ref|XP_012719085.1| PREDICTED: zo...</td>\n",
       "      <td>9</td>\n",
       "      <td>56</td>\n",
       "      <td>47</td>\n",
       "      <td>XP_012719085.1</td>\n",
       "      <td>PREDICTED: zona pellucida sperm-binding prote...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TRINITY_DN10249_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN10249_c0_g1_i1</td>\n",
       "      <td>Transcript_618557</td>\n",
       "      <td>gi|831577816|ref|XP_012737259.1| PREDICTED: ez...</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>XP_012737259.1</td>\n",
       "      <td>PREDICTED: ezrin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TRINITY_DN102633_c0_g1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN102633_c0_g1_i1</td>\n",
       "      <td>Transcript_201922</td>\n",
       "      <td>gi|831498824|ref|XP_012709872.1| PREDICTED: co...</td>\n",
       "      <td>30</td>\n",
       "      <td>68</td>\n",
       "      <td>38</td>\n",
       "      <td>XP_012709872.1</td>\n",
       "      <td>PREDICTED: collagen alpha-1(XXVII) chain B-li...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TRINITY_DN102877_c0_g1</td>\n",
       "      <td>125.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>TRINITY_DN102877_c0_g1_i1</td>\n",
       "      <td>Transcript_143501</td>\n",
       "      <td>gi|831491323|ref|XP_012707217.1| PREDICTED: LY...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>XP_012707217.1</td>\n",
       "      <td>PREDICTED: LYR motif-containing protein 4 iso...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "0  TRINITY_DN100767_c0_g1                            0.0   \n",
       "1   TRINITY_DN10175_c0_g1                            1.0   \n",
       "2   TRINITY_DN10249_c0_g1                            0.0   \n",
       "3  TRINITY_DN102633_c0_g1                            0.0   \n",
       "4  TRINITY_DN102877_c0_g1                          125.0   \n",
       "\n",
       "   F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            0.0                            0.0   \n",
       "3                            2.0                            0.0   \n",
       "4                           79.0                           78.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "0                            0.0                            0.0   \n",
       "1                            0.0                            0.0   \n",
       "2                            1.0                            0.0   \n",
       "3                            9.0                            2.0   \n",
       "4                           52.0                           55.0   \n",
       "\n",
       "   F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "0                            0.0                                  6.0   \n",
       "1                            2.0                                  1.0   \n",
       "2                            0.0                                 17.0   \n",
       "3                            1.0                                  5.0   \n",
       "4                          105.0                                 40.0   \n",
       "\n",
       "   F_heteroclitusMDPP_transfer_2.quant  F_heteroclitusMDPP_transfer_3.quant  \\\n",
       "0                                  0.0                                  1.0   \n",
       "1                                  2.0                                  9.0   \n",
       "2                                  0.0                                  0.0   \n",
       "3                                  0.0                                  0.0   \n",
       "4                                 92.0                                 72.0   \n",
       "\n",
       "               TrinityContig              seqid  \\\n",
       "0  TRINITY_DN100767_c0_g1_i1  Transcript_221533   \n",
       "1   TRINITY_DN10175_c0_g1_i1  Transcript_605742   \n",
       "2   TRINITY_DN10249_c0_g1_i1  Transcript_618557   \n",
       "3  TRINITY_DN102633_c0_g1_i1  Transcript_201922   \n",
       "4  TRINITY_DN102877_c0_g1_i1  Transcript_143501   \n",
       "\n",
       "                                       Fhet_GeneName  start  end  length  \\\n",
       "0  gi|831525541|ref|XP_012719085.1| PREDICTED: zo...      9   56      47   \n",
       "1  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...      0  220     220   \n",
       "2  gi|831577816|ref|XP_012737259.1| PREDICTED: ez...      0   45      45   \n",
       "3  gi|831498824|ref|XP_012709872.1| PREDICTED: co...     30   68      38   \n",
       "4  gi|831491323|ref|XP_012707217.1| PREDICTED: LY...      0   26      26   \n",
       "\n",
       "    NCBIproteinID                                    NCBIproteinName  \n",
       "0  XP_012719085.1   PREDICTED: zona pellucida sperm-binding prote...  \n",
       "1  XP_012706852.1      PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "2  XP_012737259.1      PREDICTED: ezrin-like [Fundulus heteroclitus]  \n",
       "3  XP_012709872.1   PREDICTED: collagen alpha-1(XXVII) chain B-li...  \n",
       "4  XP_012707217.1   PREDICTED: LYR motif-containing protein 4 iso...  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(66797, 18)"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fhet_merged_table.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_3.quant</th>\n",
       "      <th>TrinityContig</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Fhet_GeneName</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th>NCBIproteinName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRINITY_DN10175_c0_g1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>TRINITY_DN10175_c0_g1_i1</td>\n",
       "      <td>Transcript_605742</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>220</td>\n",
       "      <td>220</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>TRINITY_DN146980_c0_g1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>TRINITY_DN146980_c0_g1_i1</td>\n",
       "      <td>Transcript_91068</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>497</td>\n",
       "      <td>497</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>TRINITY_DN170990_c2_g1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>TRINITY_DN170990_c2_g1_i1</td>\n",
       "      <td>Transcript_202201</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>293</td>\n",
       "      <td>293</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>TRINITY_DN171266_c0_g2</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TRINITY_DN171266_c0_g2_i1</td>\n",
       "      <td>Transcript_175700</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>309</td>\n",
       "      <td>309</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1367</th>\n",
       "      <td>TRINITY_DN179170_c2_g1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>TRINITY_DN179170_c2_g1_i1</td>\n",
       "      <td>Transcript_54728</td>\n",
       "      <td>gi|831490281|ref|XP_012706852.1| PREDICTED: ti...</td>\n",
       "      <td>0</td>\n",
       "      <td>218</td>\n",
       "      <td>218</td>\n",
       "      <td>XP_012706852.1</td>\n",
       "      <td>PREDICTED: titin-like [Fundulus heteroclitus]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Gene  F_heteroclitusMDPP_BW_1.quant  \\\n",
       "1      TRINITY_DN10175_c0_g1                            1.0   \n",
       "465   TRINITY_DN146980_c0_g1                            2.0   \n",
       "1016  TRINITY_DN170990_c2_g1                            3.0   \n",
       "1022  TRINITY_DN171266_c0_g2                            8.0   \n",
       "1367  TRINITY_DN179170_c2_g1                            4.0   \n",
       "\n",
       "      F_heteroclitusMDPP_BW_2.quant  F_heteroclitusMDPP_BW_3.quant  \\\n",
       "1                               0.0                            0.0   \n",
       "465                             2.0                            1.0   \n",
       "1016                            1.0                            2.0   \n",
       "1022                            0.0                            0.0   \n",
       "1367                            6.0                            0.0   \n",
       "\n",
       "      F_heteroclitusMDPP_FW_1.quant  F_heteroclitusMDPP_FW_2.quant  \\\n",
       "1                               0.0                            0.0   \n",
       "465                             3.0                            0.0   \n",
       "1016                            3.0                            3.0   \n",
       "1022                            0.0                            1.0   \n",
       "1367                            2.0                            2.0   \n",
       "\n",
       "      F_heteroclitusMDPP_FW_3.quant  F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "1                               2.0                                  1.0   \n",
       "465                             8.0                                  3.0   \n",
       "1016                            1.0                                  2.0   \n",
       "1022                            2.0                                  2.0   \n",
       "1367                            1.0                                  2.0   \n",
       "\n",
       "      F_heteroclitusMDPP_transfer_2.quant  \\\n",
       "1                                     2.0   \n",
       "465                                   6.0   \n",
       "1016                                  0.0   \n",
       "1022                                  2.5   \n",
       "1367                                  3.0   \n",
       "\n",
       "      F_heteroclitusMDPP_transfer_3.quant              TrinityContig  \\\n",
       "1                                     9.0   TRINITY_DN10175_c0_g1_i1   \n",
       "465                                  27.0  TRINITY_DN146980_c0_g1_i1   \n",
       "1016                                  7.0  TRINITY_DN170990_c2_g1_i1   \n",
       "1022                                  0.0  TRINITY_DN171266_c0_g2_i1   \n",
       "1367                                  3.0  TRINITY_DN179170_c2_g1_i1   \n",
       "\n",
       "                  seqid                                      Fhet_GeneName  \\\n",
       "1     Transcript_605742  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "465    Transcript_91068  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1016  Transcript_202201  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1022  Transcript_175700  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "1367   Transcript_54728  gi|831490281|ref|XP_012706852.1| PREDICTED: ti...   \n",
       "\n",
       "      start  end  length   NCBIproteinID  \\\n",
       "1         0  220     220  XP_012706852.1   \n",
       "465       0  497     497  XP_012706852.1   \n",
       "1016      0  293     293  XP_012706852.1   \n",
       "1022      0  309     309  XP_012706852.1   \n",
       "1367      0  218     218  XP_012706852.1   \n",
       "\n",
       "                                     NCBIproteinName  \n",
       "1      PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "465    PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1016   PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1022   PREDICTED: titin-like [Fundulus heteroclitus]  \n",
       "1367   PREDICTED: titin-like [Fundulus heteroclitus]  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "titin = fhet_merged_table.loc[fhet_merged_table['NCBIproteinID'] == 'XP_012706852.1']\n",
    "titin.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort by length descending, drop duplicate genes\n",
    "titin_gene = titin.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "#titin_gene.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>XP_012706852.1</th>\n",
       "      <td>9267.948719</td>\n",
       "      <td>5022.968313</td>\n",
       "      <td>2300.499999</td>\n",
       "      <td>7129.471635</td>\n",
       "      <td>2996.556833</td>\n",
       "      <td>4430.794692</td>\n",
       "      <td>3312.109758</td>\n",
       "      <td>7756.500001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    9267.948719                    5022.968313   \n",
       "\n",
       "                F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    2300.499999                    7129.471635   \n",
       "\n",
       "                F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012706852.1                    2996.556833                    4430.794692   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "NCBIproteinID                                         \n",
       "XP_012706852.1                          3312.109758   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_2.quant  \n",
       "NCBIproteinID                                        \n",
       "XP_012706852.1                          7756.500001  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now, collapse matrix by protein, sum expression for all fragments for each sample\n",
    "# colSum[2:10] for unique proteins 'NCBIproteinID'\n",
    "titin_collapse = titin_gene.groupby(['NCBIproteinID'])[list(titin_gene.columns)[1:9]].agg('sum')\n",
    "#titin_collapse = titin_gene.groupby(['NCBIproteinID']).sum()\n",
    "titin_collapse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(27387, 18)\n",
      "(18626, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>F_heteroclitusMDPP_BW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_BW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_2.quant</th>\n",
       "      <th>F_heteroclitusMDPP_FW_3.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_1.quant</th>\n",
       "      <th>F_heteroclitusMDPP_transfer_2.quant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBIproteinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>XP_012704699.1</th>\n",
       "      <td>277.294571</td>\n",
       "      <td>329.0</td>\n",
       "      <td>165.227518</td>\n",
       "      <td>250.000000</td>\n",
       "      <td>143.160386</td>\n",
       "      <td>170.000000</td>\n",
       "      <td>110.590974</td>\n",
       "      <td>213.904699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704700.1</th>\n",
       "      <td>373.000000</td>\n",
       "      <td>478.0</td>\n",
       "      <td>242.000001</td>\n",
       "      <td>409.000000</td>\n",
       "      <td>265.202322</td>\n",
       "      <td>334.000000</td>\n",
       "      <td>207.035645</td>\n",
       "      <td>505.690830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704702.1</th>\n",
       "      <td>664.999999</td>\n",
       "      <td>837.0</td>\n",
       "      <td>476.000000</td>\n",
       "      <td>823.000000</td>\n",
       "      <td>520.699682</td>\n",
       "      <td>695.999999</td>\n",
       "      <td>304.000000</td>\n",
       "      <td>526.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704705.1</th>\n",
       "      <td>892.841678</td>\n",
       "      <td>851.0</td>\n",
       "      <td>514.000000</td>\n",
       "      <td>727.999999</td>\n",
       "      <td>353.000000</td>\n",
       "      <td>727.000000</td>\n",
       "      <td>276.000000</td>\n",
       "      <td>731.000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XP_012704706.1</th>\n",
       "      <td>7.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                F_heteroclitusMDPP_BW_1.quant  F_heteroclitusMDPP_BW_2.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     277.294571                          329.0   \n",
       "XP_012704700.1                     373.000000                          478.0   \n",
       "XP_012704702.1                     664.999999                          837.0   \n",
       "XP_012704705.1                     892.841678                          851.0   \n",
       "XP_012704706.1                       7.000000                            5.0   \n",
       "\n",
       "                F_heteroclitusMDPP_BW_3.quant  F_heteroclitusMDPP_FW_1.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     165.227518                     250.000000   \n",
       "XP_012704700.1                     242.000001                     409.000000   \n",
       "XP_012704702.1                     476.000000                     823.000000   \n",
       "XP_012704705.1                     514.000000                     727.999999   \n",
       "XP_012704706.1                       3.000000                      10.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_FW_2.quant  F_heteroclitusMDPP_FW_3.quant  \\\n",
       "NCBIproteinID                                                                  \n",
       "XP_012704699.1                     143.160386                     170.000000   \n",
       "XP_012704700.1                     265.202322                     334.000000   \n",
       "XP_012704702.1                     520.699682                     695.999999   \n",
       "XP_012704705.1                     353.000000                     727.000000   \n",
       "XP_012704706.1                       0.000000                       1.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_1.quant  \\\n",
       "NCBIproteinID                                         \n",
       "XP_012704699.1                           110.590974   \n",
       "XP_012704700.1                           207.035645   \n",
       "XP_012704702.1                           304.000000   \n",
       "XP_012704705.1                           276.000000   \n",
       "XP_012704706.1                             4.000000   \n",
       "\n",
       "                F_heteroclitusMDPP_transfer_2.quant  \n",
       "NCBIproteinID                                        \n",
       "XP_012704699.1                           213.904699  \n",
       "XP_012704700.1                           505.690830  \n",
       "XP_012704702.1                           526.000000  \n",
       "XP_012704705.1                           731.000001  \n",
       "XP_012704706.1                             2.000000  "
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now do this on everything\n",
    "fhet = fhet_merged_table.sort_values(by=['Gene','length'],ascending=False).drop_duplicates(subset='Gene')\n",
    "print(fhet.shape)\n",
    "fhet_collapse = fhet.groupby(['NCBIproteinID'])[list(fhet.columns)[1:9]].agg('sum')\n",
    "print(fhet_collapse.shape)\n",
    "fhet_collapse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique Fhet annotations, contigs with expression\n",
      "20877\n",
      "Unique NCBI protein ID\n",
      "20877\n",
      "Unique Trinity \"genes\"\n",
      "27387\n",
      "Unique Trinity \"transcripts\" (contigs) \n",
      "66797\n"
     ]
    }
   ],
   "source": [
    "print(\"Unique Fhet annotations, contigs with expression\")\n",
    "print(len(fhet_merged_table.Fhet_GeneName.unique()))\n",
    "print('Unique NCBI protein ID')\n",
    "print(len(fhet_merged_table.NCBIproteinID.unique()))\n",
    "print('Unique Trinity \"genes\"')\n",
    "print(len(fhet_merged_table.Gene.unique()))\n",
    "print('Unique Trinity \"transcripts\" (contigs) ')\n",
    "print(len(fhet_merged_table.TrinityContig.unique()))\n",
    "# fragments\n",
    "#fhet_merged_table.to_csv(gene_out)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
